---
title: "Spat_microbiome"
author: "Mary K. English"
date: "11/10/2021"
output: html_document
---
\

Load packages.
```{r}
library(dada2)
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")
library(vegan)

```

#
Get rid of these bad family 29 files:
lane1-s051-index--CAGATAGT-CGTACTCA-29L1
lane1-s052-index--CAGATAGT-CTACGCAG-29L2
lane1-s053-index--CAGATAGT-GGAGACTA-29L3
lane1-s054-index--CAGATAGT-GTCGCTCG-29L4
lane1-s055-index--CAGATAGT-GTCGTAGT-29L5
lane1-s056-index--CAGATAGT-TAGCAGAC-29L6
lane1-s057-index--CAGATAGT-TCATAGAC-29L7
lane1-s058-index--CAGATAGT-TCGCTATA-29L8
lane1-s059-index--ACTCGCTA-AACTCTCG-29S1
lane1-s060-index--ACTCGCTA-ACTATGTC-29S2
lane1-s061-index--ACTCGCTA-AGTAGCGT-29S3
lane1-s062-index--ACTCGCTA-CAGTGAGT-29S4
lane1-s063-index--ACTCGCTA-CGTACTCA-29S5
lane1-s064-index--ACTCGCTA-CTACGCAG-29S6
lane1-s065-index--ACTCGCTA-GGAGACTA-29S7
lane1-s066-index--ACTCGCTA-GTCGCTCG-29S8
lane1-s067-index--CTCGATGA-AACTCTCG-29L1
lane1-s068-index--CTCGATGA-ACTATGTC-29L2
lane1-s069-index--CTCGATGA-AGTAGCGT-29L3
lane1-s070-index--CTCGATGA-CAGTGAGT-29L4
lane1-s071-index--CTCGATGA-CGTACTCA-29L5
lane1-s072-index--CTCGATGA-CTACGCAG-29L6
lane1-s073-index--CTCGATGA-GGAGACTA-29L7
lane1-s074-index--CTCGATGA-GTCGCTCG-29L8
move into your directory
#
```{bash}
cp /nfs2/hts/miseq/211102_M01498_0851_000000000-K357Y/L1/*.gz ./
```

Change names in bash first
```{bash}
mv lane1-s003-index--CGTACGAT-AACTCTCG-8L1_S3_R1_001.fastq.gz 8L1_R1.fastq.gz
```
etc. Also change family "8" to family "3"
```{r, message = FALSE}
library(dada2)
path <- "~/Documents/Oyster_Project/size_microbiome/fastq_files/"
```
\
Check to make sure your files are there with the list.files() function. You should see a number of files that end in ".fastq".
\
```{r, message = FALSE, results = 'hide'}
list.files(path)
```
\
From that directory, select the forward and reverse reads.\
\
```{r}
fnFs_size <- sort(list.files(path, pattern="_R1.fastq", full.names = TRUE))
fnRs_size <- sort(list.files(path, pattern="_R2.fastq", full.names = TRUE))
```
\
Extract the sample names from the file names and store the sample names in a separate object. Everything after the first underscore will become a sample name.
\
```{r}
sample.names_size <- sapply(strsplit(basename(fnFs_size), "_"), `[`, 1)
```
\
Visualize the quality of your forward and reverse reads.
Note when quality starts to rapidly deteriorate, and write down somewhere the point on the x-axis ("Cycle") where that happens
\
```{r}
plotQualityProfile(fnFs_size[1:2])
# These look really good. 30 to 240
plotQualityProfile(fnRs_size[1:2])
# these are bad. Maybe 30 to 200?
```
\
Distinguish the reads that will be quality-controlled from our original .fastq files .\
```{r}
filtFs_size <- file.path(path, "filtered", paste0(sample.names_size, "_F_filt.fastq.gz"))
filtRs_size <- file.path(path, "filtered", paste0(sample.names_size, "_R_filt.fastq.gz"))
names(filtFs_size) <- sample.names_size
names(filtRs_size) <- sample.names_size
```
\
dada2 has built-in filtering parameters that will probably be fine for your samples.\
\
truncLen(f,r) changes based on your read qualities from the plots above. The default is 0, meaning dada2 won't truncate your reads even if they badly deteriorate at the end. 
\
```{r}
out_size <- filterAndTrim(fnFs_size, filtFs_size, fnRs_size, filtRs_size,
                     truncLen=c(240,200), # Where in the read to truncate (fwd,rev).
                     maxN=0, # tbh not sure about this
                     maxEE=c(2,2), # The number of expected errors allowed in a read.
                     truncQ=2, # Once the read quality goes below 2, the read will truncate.
                     rm.phix=TRUE, # This is to get rid of phiX genome reads, if used in your run.
                     compress=TRUE, # Make the output file small.
                     multithread=TRUE) # On Windows set multithread=FALSE
```
\
3 columns: sample IDs ending in ".fastq", reads.in, and reads.out. reads.out tells you how many of the reads.in made it past the filtering step.
\
```{r}
head(out_size)
```
\
Have dada2 learn the error rates for reads (every time one base is incorrectly substituted for another during sequencing). Error rates for forward and reverse sequences are different.
\
```{r}
errF_size <- learnErrors(filtFs_size, multithread=TRUE)
errR_size <- learnErrors(filtRs_size, multithread=TRUE)
```
\
Visualize the error rates.
\
```{r}
plotErrors(errF_size, nominalQ=TRUE)
plotErrors(errR_size, nominalQ=TRUE)
```
\
\
Use filtered sequences and error rates to make an object that's the correct "class" (type) for dada2 to work with. One for forward and one for reverse.
\
```{r}
dadaFs_size <- dada(filtFs_size, err=errF_size, multithread=TRUE)
dadaRs_size <- dada(filtRs_size, err=errR_size, multithread=TRUE)
```
\
Then inspect the object to see how many "true sequence variants" there are in all the sequences in a sample (in this case, sample 1). This is called "denoising".
\
```{r}
dadaFs_size[[1]]
dadaRs_size[[1]]
```
\
Now it's time to merge the forward and reverse reads for each sequence to give full, denoised sequence.
\
```{r}
mergers_size <- mergePairs(dadaFs_size, filtFs_size, dadaRs_size, filtRs_size, verbose=TRUE)
```
\
This should be a data frame. Each row corresponds to a sample. Columns:\
1. sequence: the actual string of base pairs. This column might print "above" the rest of the data frame because its contents are so long.\
2. abundance: the number of times the sequence was detected.\
3&4. forward and reverse. each of these is an index number that tells you which forward read was paired with which reverse. They will probably have some differences.\
5. nmatch. the number of base pairs that overlap in the overlapregion.\
6. nmismatch. the number of base pairs that don't overlap in the overlap region (low).\
7. nindel. number of insertions or deletions in the overlap region (low).\
8. prefer. tells you which sequence (forward = 1, reverse = 2) was used as the "template" for the overlap region.\
9. accept. tells you if there was a strong enough overlap to join the forward and reverse sequences.\
\
\
Now it's time to make a table that will hold all this data. Your sequence table will be called "seqtab".
\
```{r}
seqtab_size <- makeSequenceTable(mergers_size)
```
\
Get the dimensions of your table and write it down:
\
```{r}
dim(seqtab_size)
# 154 x 3380
```
\
View lengths of the sequences, which should be around 252 bp for 16S reads.
```{r}
table(nchar(getSequences(seqtab_size)))
```
\
The top row of the output will be the length of the sequence in base pairs, and the bottom row is the number of times each 
sequence of that length appeared. All the bottom row numbers should sum to the number of sequences from the dim() function.
\
If something is way off -- like one sequence length that's way different from the others -- it's probably a primer error and you should get rid of it. In general, filter all sequence lengths to between 250 and 256 bp.\
```{r}
seqtab2_size <- seqtab_size[,nchar(colnames(seqtab_size)) %in% 250:256]
seqtab_size <- seqtab2_size
```
\
Remove chimeras, which are 2 existing sequences mistakenly combined into a new sequence.\
```{r}
seqtab_size.nochim <- removeBimeraDenovo(seqtab_size, method="consensus", multithread=TRUE, verbose=TRUE)
```
\
49 bimeras.
\
Run the dim() command on your new chimera-free sequence table to see how many chimeras there were.
\
```{r}
dim(seqtab_size.nochim)
# 154 x 3259
#write.csv(seqtab_size.nochim, file = "~/Documents/Oyster_Project/size_microbiome/seqtab_size.nochim.csv")
```
\
To get the percent of genuine sequences that weren't chimeras:
\
```{r}
sum(seqtab_size.nochim)/sum(seqtab_size)
```
\
99.8%
\
Keep track of how many sequences you lose at each "filtering" step:
\
```{r}
getN <- function(x) sum(getUniques(x))
track_size <- cbind(out_size, sapply(dadaFs_size, getN), sapply(dadaRs_size, getN), sapply(mergers_size, getN), rowSums(seqtab_size.nochim))
colnames(track_size) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track_size) <- sample.names_size
write.csv(track_size, file = "~/Documents/Oyster_Project/size_microbiome/track_size.csv")
```
\
See how many samples have been lost, from the original "input" to the most recent "nonchim":
\
```{r}
head(track_size)
```
\
Make a taxa table. To do so, have the silva 16S database stored on your computer. The file is "silva_nr_v132_train_set.fa.gz", and it's from this site:\
https://zenodo.org/record/1172783 \
\
```{r}
fn <- "~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz"
file.exists(fn)
taxa_size <- assignTaxonomy(seqtab_size.nochim, "~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz", multithread=TRUE, tryRC = TRUE)
write.csv(taxa_size, file = "~/Documents/Oyster_Project/size_microbiome/reviewer_edits/taxa_size.csv")
```
\
For phyloseq
\
\
First, load packages.
```{r}
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")
library(vegan)
library(lessR)
```
\
\
Turn ASV table, which is most likely a matrix, into a data frame so it's in the right format for phyloseq to work with.
```{r}
seqtab_size.nochim.df <- as.data.frame(seqtab_size.nochim)
```

Dealing with asv table. Each row is a sample. We want to remove samples with less than 1000 reads.
```{r}
seqtab_size.nochim.df <- t(seqtab_size.nochim.df)
# create asv count and taxa matrices
min(rowSums(seqtab_size.nochim.df)) # 0
write.csv(seqtab_size.nochim.df, file = "~/Documents/Oyster_Project/size_microbiome/reviewer_edits/seqtab_size.nochim.df.csv")
```
\
Identify samples (rows) with row sums below 1000 manually in excel.
Remove rows:
```{r}
#create a copy for manipulation
seqtab_size.nochim.df_1000 <- seqtab_size.nochim.df
# 40L1 = 78
# 25S4 = 41
# 25S8 = 45
# 29S5 = 58
# 69L6 = 99
# 69L8 = 100
# 74L1 = 109
seqtab_size.nochim.df_1000 <- ((seqtab_size.nochim.df[-c(41,45,58,78,99,100,109),]))
min(rowSums(seqtab_size.nochim.df_1000)) # 1235
```

Now assign taxonomy to sequence table.
```{r}
# must be a matrix
seqtab_size.nochim.df_1000 <- as.matrix(seqtab_size.nochim.df_1000)
taxa_size_1000 <- assignTaxonomy(seqtab_size.nochim.df_1000,"~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz", multithread=TRUE, tryRC = TRUE)
```

\
REMOVING CHLORPLASTS AND MITOCHONDRIA\
After removing chimeras and assigning taxonomy.\
```{r}
reduce_taxa <- apply(taxa_size_1000, 1, function(r) any(r %in% c("Chloroplast", "Mitochondria")))
```

remove from count table\
```{r}
dim(seqtab_size.nochim.df_1000) # 147 x 3259
seqtab_size.nochim.df_1000_reduced <- seqtab_size.nochim.df_1000[,!reduce_taxa]
```
See how many were removed\
```{r}
dim(seqtab_size.nochim.df_1000_reduced) # 147 x 3158
```
3259-3158 = 101
seqtab_size.nochim.df_1000_reduced is asv table.\
taxa_size_1000 is taxa table.
\
Now transpose. ASVs need to be rows.
```{r}
seqtab_size.nochim.df_1000_reduced <-t(seqtab_size.nochim.df_1000_reduced)
write.csv(seqtab_size.nochim.df_1000_reduced, file = "~/Documents/Oyster_Project/size_microbiome/asv_table_with_seqs.csv")
```

Now work on the taxonomy table.\
```{r}
dim(taxa_size_1000) # 3259 x 6.
taxa_size_1000_reduced <- taxa_size_1000[!reduce_taxa,]
asvs_size <- to("asv", nrow(taxa_size_1000_reduced))
dim(taxa_size_1000_reduced) # 3186 x 6
write.csv(taxa_size_1000_reduced, file = "~/Documents/Oyster_Project/size_microbiome/taxa_size_1000_reduced.csv")
```
your new taxa table is "taxa_size_reduced"

FILTER OUT LOW ABUNDANCE TAXA
Want to filter out (remove from ASV and taxa tables, and therefore seq names) all the taxa that represent < 0.1% of all reads (total number of reads: Sum(rowsums) = 1,892,480 reads among the remaining 147 samples that had >1000 reads).
0.001 * 1,892,480 = 1892
So, remove the asvs whose sums in all remaining samples combined are less than 1892.
```{r}
sum(rowSums(seqtab_size.nochim.df_1000_reduced))
seqtab_for_filter <- seqtab_size.nochim.df_1000_reduced
# back to data frame
seqtab_for_filter <- as.data.frame(seqtab_for_filter)
seqtab_low_filter <- seqtab_for_filter[c(rowSums(seqtab_for_filter)>1892),]
dim(seqtab_low_filter) # 96 taxa x 147 samples
write.csv(seqtab_low_filter, file = "~/Documents/Oyster_Project/size_microbiome/seqtab_low_filter.csv")
# make sure minimum is >1892
min(rowSums(seqtab_low_filter)) # 1913
# convert back to matrix 
seqtab_low_filter <- as.matrix(seqtab_low_filter)
seqtab_low_filter <- t(seqtab_low_filter)
#reassigning taxonomy on reduced ASV table
taxa_low_filter <- assignTaxonomy(seqtab_low_filter, "~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz", multithread=TRUE, tryRC = TRUE)
write.csv(taxa_low_filter, "~/Documents/Oyster_Project/size_microbiome/taxa_low_filter.csv")
```

Get strings.
```{r}
seqs_size <- colnames(seqtab_low_filter)
length(seqs_size) # 96
```

Change row names (AFTER getting fasta strings).
```{r}
asvs_low_size_filter <- to("asv", nrow(taxa_low_filter))
rownames(taxa_low_filter) <- asvs_low_size_filter
# make a taxa csv with asv names
write.csv(taxa_low_filter, file = "~/Documents/Oyster_Project/size_microbiome/taxa_low_filter_with_asvs.csv")
colnames(seqtab_low_filter) <- asvs_low_size_filter
# make asv csv with asv names
write.csv(seqtab_low_filter, file = "~/Documents/Oyster_Project/size_microbiome/seqtab_low_filter_with_asvs.csv")

```

Make a fasta file for your phyloseq object.
```{r}
seqs_size_strings <- DNAStringSet(seqs_size)
names(seqs_size_strings) <- asvs_low_size_filter
writeXStringSet(x = seqs_size_strings, filepath = "~/Documents/Oyster_Project/size_microbiome/seqs_size_strings.fa")
```
Your string set is called "seqs_size_strings.fa".\
\
Log into the files.cgrb infrastructure and drag your ".fa" file from above into your main directory.\
\
Use CyberDuck to move seqs_size_strings.fa to your files.cgrb directory.\
In terminal:\
{ssh -p 732 englimar@shell.cgrb.oregonstate.edu
SGE_Avail
qrsh -q makaira
mothur}
\
\textcolor{red}{
mothur > align.seqs(fasta=seqs_size_strings.fa, reference=/nfs5/Mueller_Lab/mk/16s_database/silva.nr_v132.align, flip=t, processors=46)
mothur > filter.seqs(fasta=seqs_size_strings.align, processors=46)
mothur > quit()}
\
\textcolor{blue}{
FastTreeMP -gtr -nt -log seqs_size_strings.filter.log seqs_size_strings.filter.fasta > seqs_size_strings.filter.tre
Re-root from midpoint:
/local/cluster/mueller/scripts/Genomics/stat/reroot.pl -midpoint < seqs_size_strings.filter.tre > seqs_size_strings.filter.midroot.tre
}

Download tree and strings_filter_fasta from cyberduck and put in directory with everything else.\
Read in and plot tree:
```{r}
library(ape)
spat_size_tree <- read.tree(file = "~/Documents/Oyster_Project/size_microbiome/seqs_size_strings.filter.midroot.tre")
plot(spat_size_tree)
```
Get reference fasta file:
```{r}
size_reduced_fasta_filtered <- readDNAStringSet("~/Documents/Oyster_Project/size_microbiome/seqs_size_strings.filter.fasta")
```

Build phyloseq object.
```{r}
spat_size_meta <- read.csv("~/Documents/Oyster_Project/size_microbiome/spat_size_meta.csv", sep=",", header=T, row.names=1)
# 25S4, 25S8,29S5,40L1,69L6,69L8,74L1. These 7 had less than 1000 filtered reads.
spat_size_meta_reduced <- spat_size_meta[-c(41,45,58,78,99,100,109),]
# 40L1 = 78
# 25S4 = 41
# 25S8 = 45
# 29S5 = 58
# 69L6 = 99
# 69L8 = 100
# 74L1 = 109
# Make Family and Size factors, for stats purposes
write.csv(spat_size_meta_reduced, file = "~/Documents/Oyster_Project/size_microbiome/spat_size_meta_reduced.csv")
spat_size_meta_reduced <- read.csv("~/Documents/Oyster_Project/size_microbiome/spat_size_meta_reduced.csv", sep = ",", header = T, row.names = 1)
spat_size_meta_reduced$Family <- as.factor(spat_size_meta_reduced$Family)
spat_size_meta_reduced$Size <- as.factor(spat_size_meta_reduced$Size)
#transpose sequence table if needed
seqtab_low_filter <- t(seqtab_low_filter)

seqtab_low_filter_with_asvs <- read_csv("Documents/Oyster_Project/size_microbiome/seqtab_low_filter_with_asvs.csv")
seqtab_low_filter_with_asvs <- as.matrix(seqtab_low_filter_with_asvs)
class(seqtab_low_filter_with_asvs) <- "numeric"
rownames(seqtab_low_filter_with_asvs) <- asvs_low_size_filter

taxa_low_filter_with_asvs <- read_csv("Documents/Oyster_Project/size_microbiome/taxa_low_filter_with_asvs.csv")
taxa_low_filter_with_asvs <- as.matrix(taxa_low_filter_with_asvs)
asvs_low_size_filter <- to("asv", nrow(taxa_low_filter))

rownames(taxa_low_filter_with_asvs_edit) <- asvs_low_size_filter
# remove first column
taxa_low_filter_with_asvs_edit <- taxa_low_filter_with_asvs[,-1]
rownames(taxa_low_filter_with_asvs_edit) <- asvs_low_size_filter
taxa_low_filter_with_asvs_edit <- as.matrix(taxa_low_filter_with_asvs_edit)

spat_size_phylo <- phyloseq(otu_table(seqtab_low_filter_with_asvs, taxa_are_rows = TRUE), 
                                   sample_data(spat_size_meta_reduced),
                                   tax_table(taxa_low_filter_with_asvs_edit),
                                   phy_tree(spat_size_tree),
                                   refseq(size_reduced_fasta_filtered))
```
Prune (keeping this in, but it shouldn't be necessary since we already pruned to 1000 reads).
```{r}
spat_size_phylo_sum <- prune_taxa(taxa_sums(spat_size_phylo) > 0, spat_size_phylo)
```
Alpha diversity:
```{r}
library(microbiome)
alpha_size_table <- microbiome::alpha(spat_size_phylo_sum, index = "all")
write.csv(alpha_size_table, file = "~/Documents/Oyster_Project/size_microbiome/alpha_size_table.csv")
#added columns with Size and Family
#reimport
alpha_size <- read.csv("~/Documents/Oyster_Project/size_microbiome/alpha_size_table.csv", sep = ",", header = T, row.names = 1)
alpha_size$Family <- as.factor(alpha_size$Family)
alpha_size$Size <- as.factor(alpha_size$Size)
t.test(chao1 ~ Size, data = alpha_size) # p < 0.00001
# considering weight
alpha_size_table_with_weight <- read.csv("~/Documents/Oyster_Project/size_microbiome/alpha_size_table_with_weight.csv", sep = ",", header = T, row.names = 1)
alpha_size_table_with_weight$Avg_wet_weight <- as.numeric(alpha_size_table_with_weight$Avg_wet_weight)
alpha_size_table_with_weight$Avg_dry_weight <- as.numeric(alpha_size_table_with_weight$Avg_dry_weight)
alpha_size_table_with_weight$Size <- as.factor(alpha_size_table_with_weight$Size)
alpha_size_table_with_weight$Family <- as.factor(alpha_size_table_with_weight$Family)
summary(lm(chao1 ~ Avg_wet_weight, data = alpha_size_table_with_weight))
summary(lm(chao1 ~ Avg_dry_weight, data = alpha_size_table_with_weight))
summary(lm(chao1 ~ Avg_wet_weight*Size*Family, data = alpha_size_table_with_weight))
summary(lm(chao1 ~ Avg_wet_weight*Size*Family, data = alpha_size_table_with_weight))
summary(lm(chao1 ~ Avg_dry_weight*Size, data = alpha_size_table_with_weight))
summary(lm(chao1 ~ Avg_wet_weight*Size, data = alpha_size_table_with_weight))
summary(lm(chao1 ~ Avg_wet_weight + Avg_dry_weight + Size + Family, data = alpha_size_table_with_weight))
summary(lm(chao1 ~ Avg_wet_weight*Avg_dry_weight*Size*Family, data = alpha_size_table_with_weight))
```
Plotting alpha diversity.
```{r}
alpha_size_plot <- ggplot(alpha_size, aes(x =Size, y = chao1))+
  geom_boxplot(aes(group = Size))+
  geom_label(x=1.5, y=60, label = "p(t.test)<0.0001", size = 3)+
  geom_point(aes(color = Family), position=position_dodge(width = 0.5))+
  labs(title = "Alpha diversity of microbiome of differentially sized spat",
       y = "Chao1",
       x = "Spat size",
       caption = "Taxa representing >0.01% of all reads")+
  scale_color_manual(values = mkePalette_10)

ggsave(filename = "alpha_size_plot.png", plot = alpha_size_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")

```

Calculate weighted unifrac beta diversity. Do UNIFRAC separate for use in adonis2.
```{r}
spat_size_phylo_sum_wunidist <- UniFrac(spat_size_phylo_sum, weighted=TRUE, normalized=TRUE, parallel=TRUE, fast=TRUE)

# Size
spat_size_phylo_sum_wunidist_adonis_size <- adonis2(spat_size_phylo_sum_wunidist ~ Size, data = spat_size_meta_reduced)
spat_size_phylo_sum_wunidist_adonis_size
# p = 0.001, R2 = 0.05032
# Family
spat_size_phylo_sum_wunidist_adonis_family <- adonis2(spat_size_phylo_sum_wunidist ~ Family, data = spat_size_meta_reduced)
spat_size_phylo_sum_wunidist_adonis_family
# p = 0.001, R2 = 0.14566
spat_size_phylo_sum_wunidist_adonis_family_size <- adonis2(spat_size_phylo_sum_wunidist ~ Size + Family, data = spat_size_meta_reduced)
spat_size_phylo_sum_wunidist_adonis_family_size
spat_size_phylo_sum_wunidist_adonis_family_and_size <- adonis2(spat_size_phylo_sum_wunidist ~ Size * Family, data = spat_size_meta_reduced)
spat_size_phylo_sum_wunidist_adonis_family_and_size
```

Plot beta diversity.
```{r}
spat_size_phylo_sum_wuni <- ordinate(spat_size_phylo_sum, "PCoA", "unifrac", weighted=TRUE)
mkePalette_10 <- c("#FDE725FF", "#B8DE29FF", "#73D055FF", "#3CBB75FF", "#20A387FF", "#238A8DFF","#2D708EFF","#39568CFF", "#453781FF", "#481567FF")
spat_size_phylo_sum_wuni_plot <- plot_ordination(spat_size_phylo_sum, spat_size_phylo_sum_wuni, color = "Family", shape = "Size")+
  geom_point(size = 3)+
  scale_color_manual(values = mkePalette_10)+
  scale_fill_manual(values = mkePalette_10)+
  theme(panel.background = element_rect(fill = "gray80"))+
  labs(title = "Beta diversity in microbime of differentially sized spat",
       subtitle = "Weighted UniFrac",
       caption = "Taxa representing >0.01% of all reads")
spat_size_phylo_sum_wuni_plot
ggsave(filename = "spat_size_phylo_sum_wuni_plot.png", plot = spat_size_phylo_sum_wuni_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
# Scaling points by read depth (track_size$nonchim)

```
Making another beta diversity plot to look at just size.
```{r}
spat_size_phylo_sum_wuni_plot_size_only <- plot_ordination(spat_size_phylo_sum, spat_size_phylo_sum_wuni, color = "Size")+#, label = "Family")+
  geom_point(size = 3)+
  scale_color_manual(values = c("forestgreen", "steelblue"))+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(title = "Beta diversity in microbime of differentially sized spat",
       subtitle = "Weighted UniFrac",
       caption = "Taxa representing >0.01% of all reads")#,
#       caption = "Label = Family")
ggsave(filename = "spat_size_phylo_sum_wuni_plot_size_only.png", plot = spat_size_phylo_sum_wuni_plot_size_only, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```
DESEQ2

```{r}
library(DESeq2)
deseq_spat_size_size <- phyloseq_to_deseq2(spat_size_phylo_sum, ~ Size)
```
Going to add 1 to every single cell in asv table to avoid any genes being 0.
```{r}
seqtab_low_filter_no_0 <- seqtab_low_filter+1

spat_size_phylo_no_0 <- phyloseq(otu_table(seqtab_low_filter_no_0, taxa_are_rows = TRUE), 
                                   sample_data(spat_size_meta_reduced),
                                   tax_table(taxa_low_filter),
                                   phy_tree(spat_size_tree),
                                   refseq(size_reduced_fasta_filtered))
spat_size_phylo_sum_no_0 <- prune_taxa(taxa_sums(spat_size_phylo_no_0) > 0, spat_size_phylo_no_0)

deseq_spat_size_no_0_size <- phyloseq_to_deseq2(spat_size_phylo_sum_no_0, ~ Size)
deseq_obj_spat_size_no_0_size <- DESeq(deseq_spat_size_no_0_size, test = "Wald", fitType = "parametric")
deseq_results_obj_spat_size_no_0_size <- results(deseq_obj_spat_size_no_0_size, cooksCutoff = FALSE)
# make significant results table at p < 0.01
sigtab_deseq_results_obj_spat_size_no_0_size <- deseq_results_obj_spat_size_no_0_size[which(deseq_results_obj_spat_size_no_0_size$padj < 0.01),]
write.csv(sigtab_deseq_results_obj_spat_size_no_0_size, file = "~/Documents/Oyster_Project/size_microbiome/sigtab_deseq_results_size.csv")
```
Verify that i'm interpreting the log-fold change correctly

```{r}
deseq_check_asv63 <- read.csv(file = "~/Documents/Oyster_Project/size_microbiome/asv_63.csv", sep = ",", row.names = 4, header = TRUE)
deseq_check_asv63$Family <- as.factor(deseq_check_asv63$Family)
ggplot(deseq_check_asv63)+
  geom_boxplot(aes(x = Size, y = Avg_pct_reads, fill = Size, group = Size))+
  facet_wrap(~Family)
#too small to see on some - these should be sub-setted and grouped by y variation

deseq_check_asv63_high <- subset(deseq_check_asv63, 
                                 Family == 3 | Family == 22 | Family == 29 | Family == 25)
deseq_check_asv63_low <- subset(deseq_check_asv63, 
                                Family != 3 & Family != 22 & Family != 29 & Family != 25 & Family != 94)

ggplot(deseq_check_asv63_high)+
  geom_boxplot(aes(x = Size, y = Pct_samples_reads, fill = Size, group = Size))+
  labs(title = "Abundance of ASV63 as percent of each sample's reads", y = "% of each sample's reads")+
  scale_fill_manual(values = c("orange", "steelblue"))+
  facet_wrap(~Family)

ggplot(deseq_check_asv63_low)+
  geom_boxplot(aes(x = Size, y = Pct_samples_reads, fill = Size, group = Size))+
    labs(title = "Abundance of ASV63 as percent of each sample's reads", y = "% of each sample's reads")+
  scale_fill_manual(values = c("orange", "steelblue"))+
  facet_wrap(~Family)
t.test(Pct_samples_reads ~ Size, data = deseq_check_asv63)
lm(Pct_samples_reads ~ Family, data = deseq_check_asv63)
summary(lm(Pct_samples_reads ~ Family, data = deseq_check_asv63))
summary(lm(Pct_samples_reads ~ Size, data = deseq_check_asv63))
glm(Pct_samples_reads ~ Size, data = deseq_check_asv63)
glm(Pct_samples_reads ~ Family, data = deseq_check_asv63)
glm(Pct_samples_reads ~ Size+Family, data = deseq_check_asv63)
```
```{r}
asv_63_avg <- read.csv(file = "~/Documents/Oyster_Project/size_microbiome/asv_63.csv", header = TRUE, sep = ",")
asv_63_avg$Family <- as.factor(asv_63_avg$Family)
ggplot(asv_63_avg)+
  geom_point(aes(x = Size, y = Average_pct_reads, color = Size))+
  facet_wrap(~Family)
asv_63_avg_high <- subset(asv_63_avg, 
                          Family == 3 | Family == 22 | Family == 29 | Family == 25)
ggplot(asv_63_avg_high)+
  geom_point(aes(x = Size, y = Average_pct_reads, color = Size), size = 3, show.legend = FALSE)+
  labs(title = "Avg. abundance of ASV63 as percent of each sample's reads", 
       y = "% of each sample's reads",
       caption = "by family")+
  scale_color_manual(values = c("orange", "steelblue"))+
  facet_wrap(~Family)
asv_63_avg_low <- subset(asv_63_avg, 
                         Family != 3 & Family != 22 & Family != 29 & Family != 25)
ggplot(asv_63_avg_low)+
  geom_point(aes(x = Size, y = Average_pct_reads, color = Size), size = 3, show.legend = FALSE)+
  labs(title = "Avg. abundance of ASV63 as percent of each sample's reads", 
       y = "% of each sample's reads",
       caption = "by family")+
  scale_color_manual(values = c("orange", "steelblue"))+
  facet_wrap(~Family)
```


Re-assessment of data
4. Make bar graph with top Y taxa by family.
5. After weighing, develop GLM with taxon(s) and size
Trying to develop a GLM for wet and dry weight, and other characteristics.
```{r}
deseq_check_asv63 <- subset(deseq_check_asv63, Family != 94)
deseq_check_asv63$Avg_wet_weight <- as.numeric(deseq_check_asv63$Avg_wet_weight)
deseq_check_asv63$Avg_dry_weight <- as.numeric(deseq_check_asv63$Avg_dry_weight)
#t.test(Avg_pct_reads ~ Size, data = deseq_check_asv63)
#lm(Avg_pct_reads ~ Family, data = deseq_check_asv63)
#summary(lm(Avg_pct_reads ~ Family, data = deseq_check_asv63))
#summary(lm(Avg_pct_reads ~ Size, data = deseq_check_asv63))
#glm(Avg_pct_reads ~ Size, data = deseq_check_asv63)
#glm(Avg_pct_reads ~ Family, data = deseq_check_asv63)
summary(lm(Avg_pct_reads ~ Avg_wet_weight, data = deseq_check_asv63))
# not signif
summary(lm(Avg_pct_reads ~ Avg_dry_weight, data = deseq_check_asv63))
# not signif
summary(lm(Avg_pct_reads ~ Avg_wet_weight, data = deseq_check_asv63))
```
Goal: write a for-loop that ultimately provides, for each of 96 asvs,  (Percent_ASVx_reads_in_each_sample ~ Avg_dry_weight)
Try it with largest 96 ASVs first? 
ASV table: seqtab_low_filter
Taxa table: taxa_low_filter
Metadata table: spat_size_meta_reduced


```{r}
percentages <- read_csv("Documents/Oyster_Project/size_microbiome/percentages.csv", 
+     col_types = cols(Family = col_character(), 
+         Sample = col_character()))
percentages$Avg_dry_weight <- as.numeric(percentages$Avg_dry_weight)
percentages$Avg_wet_weight <- as.numeric(percentages$Avg_wet_weight)
percentages$Size <- as.factor(percentages$Size)

#prints the p-value
p_val <- summary(lm(percentages$asv01 ~ Avg_dry_weight, data = percentages))$coefficients[8]
```
For COLUMN in percentages, COLUMN_p <- summary(lm(percentages$COLUMN ~ Size, data = percentages))$coefficients[8]. Append COLUMN and COLUMN_p to a dataframe.
Here goes:
```{r}
p_val_asv01 <- summary(lm(asv01~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv02 <- summary(lm(asv02~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv03 <- summary(lm(asv03~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv04 <- summary(lm(asv04~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv05 <- summary(lm(asv05~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv06 <- summary(lm(asv06~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv07 <- summary(lm(asv07~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv08 <- summary(lm(asv08~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv09 <- summary(lm(asv09~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv10 <- summary(lm(asv10~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv11 <- summary(lm(asv11~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv12 <- summary(lm(asv12~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv13 <- summary(lm(asv13~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv14 <- summary(lm(asv14~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv15 <- summary(lm(asv15~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv16 <- summary(lm(asv16~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv17 <- summary(lm(asv17~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv18 <- summary(lm(asv18~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv19 <- summary(lm(asv19~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv20 <- summary(lm(asv20~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv21 <- summary(lm(asv21~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv22 <- summary(lm(asv22~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv23 <- summary(lm(asv23~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv24 <- summary(lm(asv24~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv25 <- summary(lm(asv25~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv26 <- summary(lm(asv26~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv27 <- summary(lm(asv27~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv28 <- summary(lm(asv28~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv29 <- summary(lm(asv29~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv30 <- summary(lm(asv30~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv31 <- summary(lm(asv31~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv32 <- summary(lm(asv32~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv33 <- summary(lm(asv33~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv34 <- summary(lm(asv34~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv35 <- summary(lm(asv35~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv36 <- summary(lm(asv36~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv37 <- summary(lm(asv37~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv38 <- summary(lm(asv38~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv39 <- summary(lm(asv39~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv40 <- summary(lm(asv40~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv41 <- summary(lm(asv41~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv42 <- summary(lm(asv42~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv43 <- summary(lm(asv43~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv44 <- summary(lm(asv44~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv45 <- summary(lm(asv45~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv46 <- summary(lm(asv46~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv47 <- summary(lm(asv47~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv48 <- summary(lm(asv48~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv49 <- summary(lm(asv49~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv50 <- summary(lm(asv50~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv51 <- summary(lm(asv51~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv52 <- summary(lm(asv52~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv53 <- summary(lm(asv53~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv54 <- summary(lm(asv54~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv55 <- summary(lm(asv55~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv56 <- summary(lm(asv56~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv57 <- summary(lm(asv57~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv58 <- summary(lm(asv58~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv59 <- summary(lm(asv59~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv60 <- summary(lm(asv60~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv61 <- summary(lm(asv61~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv62 <- summary(lm(asv62~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv63 <- summary(lm(asv63~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv64 <- summary(lm(asv64~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv65 <- summary(lm(asv65~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv66 <- summary(lm(asv66~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv67 <- summary(lm(asv67~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv68 <- summary(lm(asv68~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv69 <- summary(lm(asv69~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv70 <- summary(lm(asv70~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv71 <- summary(lm(asv71~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv72 <- summary(lm(asv72~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv73 <- summary(lm(asv73~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv74 <- summary(lm(asv74~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv75 <- summary(lm(asv75~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv76 <- summary(lm(asv76~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv77 <- summary(lm(asv77~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv78 <- summary(lm(asv78~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv79 <- summary(lm(asv79~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv80 <- summary(lm(asv80~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv81 <- summary(lm(asv81~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv82 <- summary(lm(asv82~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv83 <- summary(lm(asv83~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv84 <- summary(lm(asv84~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv85 <- summary(lm(asv85~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv86 <- summary(lm(asv86~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv87 <- summary(lm(asv87~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv88 <- summary(lm(asv88~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv89 <- summary(lm(asv89~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv90 <- summary(lm(asv90~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv91 <- summary(lm(asv91~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv92 <- summary(lm(asv92~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv93 <- summary(lm(asv93~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv94 <- summary(lm(asv94~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv95 <- summary(lm(asv95~ Avg_dry_weight, data = percentages))$coefficients[8]
p_val_asv96 <- summary(lm(asv96~ Avg_dry_weight, data = percentages))$coefficients[8]

p_val_list <- ls(pattern = "p_val_")
write.csv(p_val_list, file = "~/Documents/Oyster_Project/size_microbiome/p_val_list.csv")
as.list(ls(pattern = "p_val_"))
paste('asv01',p_val_asv01)
paste('asv02',p_val_asv02)
paste('asv03',p_val_asv03)
paste('asv04',p_val_asv04)
paste('asv05',p_val_asv05)
paste('asv06',p_val_asv06)
paste('asv07',p_val_asv07)
paste('asv08',p_val_asv08)
paste('asv09',p_val_asv09)
paste('asv10',p_val_asv10)
paste('asv11',p_val_asv11)
paste('asv12',p_val_asv12)
paste('asv13',p_val_asv13)
paste('asv14',p_val_asv14)
paste('asv15',p_val_asv15)
paste('asv16',p_val_asv16)
paste('asv17',p_val_asv17)
paste('asv18',p_val_asv18)
paste('asv19',p_val_asv19)
paste('asv20',p_val_asv20)
paste('asv21',p_val_asv21)
paste('asv22',p_val_asv22)
paste('asv23',p_val_asv23)
paste('asv24',p_val_asv24)
paste('asv25',p_val_asv25)
paste('asv26',p_val_asv26)
paste('asv27',p_val_asv27)
paste('asv28',p_val_asv28)
paste('asv29',p_val_asv29)
paste('asv30',p_val_asv30)
paste('asv31',p_val_asv31)
paste('asv32',p_val_asv32)
paste('asv33',p_val_asv33)
paste('asv34',p_val_asv34)
paste('asv35',p_val_asv35)
paste('asv36',p_val_asv36)
paste('asv37',p_val_asv37)
paste('asv38',p_val_asv38)
paste('asv39',p_val_asv39)
paste('asv40',p_val_asv40)
paste('asv41',p_val_asv41)
paste('asv42',p_val_asv42)
paste('asv43',p_val_asv43)
paste('asv44',p_val_asv44)
paste('asv45',p_val_asv45)
paste('asv46',p_val_asv46)
paste('asv47',p_val_asv47)
paste('asv48',p_val_asv48)
paste('asv49',p_val_asv49)
paste('asv50',p_val_asv50)
paste('asv51',p_val_asv51)
paste('asv52',p_val_asv52)
paste('asv53',p_val_asv53)
paste('asv54',p_val_asv54)
paste('asv55',p_val_asv55)
paste('asv56',p_val_asv56)
paste('asv57',p_val_asv57)
paste('asv58',p_val_asv58)
paste('asv59',p_val_asv59)
paste('asv60',p_val_asv60)
paste('asv61',p_val_asv61)
paste('asv62',p_val_asv62)
paste('asv63',p_val_asv63)
paste('asv64',p_val_asv64)
paste('asv65',p_val_asv65)
paste('asv66',p_val_asv66)
paste('asv67',p_val_asv67)
paste('asv68',p_val_asv68)
paste('asv69',p_val_asv69)
paste('asv70',p_val_asv70)
paste('asv71',p_val_asv71)
paste('asv72',p_val_asv72)
paste('asv73',p_val_asv73)
paste('asv74',p_val_asv74)
paste('asv75',p_val_asv75)
paste('asv76',p_val_asv76)
paste('asv77',p_val_asv77)
paste('asv78',p_val_asv78)
paste('asv79',p_val_asv79)
paste('asv80',p_val_asv80)
paste('asv81',p_val_asv81)
paste('asv82',p_val_asv82)
paste('asv83',p_val_asv83)
paste('asv84',p_val_asv84)
paste('asv85',p_val_asv85)
paste('asv86',p_val_asv86)
paste('asv87',p_val_asv87)
paste('asv88',p_val_asv88)
paste('asv89',p_val_asv89)
paste('asv90',p_val_asv90)
paste('asv91',p_val_asv91)
paste('asv92',p_val_asv92)
paste('asv93',p_val_asv93)
paste('asv94',p_val_asv94)
paste('asv95',p_val_asv95)
paste('asv96',p_val_asv96)
```
```{r}
p.adjust(p_val_asv01, method = 'BH', n = 96)
p.adjust(p_val_asv02, method = 'BH', n = 96)
p.adjust(p_val_asv03, method = 'BH', n = 96)
p.adjust(p_val_asv04, method = 'BH', n = 96)
p.adjust(p_val_asv05, method = 'BH', n = 96)
p.adjust(p_val_asv06, method = 'BH', n = 96)
p.adjust(p_val_asv07, method = 'BH', n = 96)
p.adjust(p_val_asv08, method = 'BH', n = 96)
p.adjust(p_val_asv09, method = 'BH', n = 96)
p.adjust(p_val_asv10, method = 'BH', n = 96)
p.adjust(p_val_asv11, method = 'BH', n = 96)
p.adjust(p_val_asv12, method = 'BH', n = 96)
p.adjust(p_val_asv13, method = 'BH', n = 96)
p.adjust(p_val_asv14, method = 'BH', n = 96)
p.adjust(p_val_asv15, method = 'BH', n = 96)
p.adjust(p_val_asv16, method = 'BH', n = 96)
p.adjust(p_val_asv17, method = 'BH', n = 96)
p.adjust(p_val_asv18, method = 'BH', n = 96)
p.adjust(p_val_asv19, method = 'BH', n = 96)
p.adjust(p_val_asv20, method = 'BH', n = 96)
p.adjust(p_val_asv21, method = 'BH', n = 96)
p.adjust(p_val_asv22, method = 'BH', n = 96)
p.adjust(p_val_asv23, method = 'BH', n = 96)
p.adjust(p_val_asv24, method = 'BH', n = 96)
p.adjust(p_val_asv25, method = 'BH', n = 96)
p.adjust(p_val_asv26, method = 'BH', n = 96)
p.adjust(p_val_asv27, method = 'BH', n = 96)
p.adjust(p_val_asv28, method = 'BH', n = 96)
p.adjust(p_val_asv29, method = 'BH', n = 96)
p.adjust(p_val_asv30, method = 'BH', n = 96)
p.adjust(p_val_asv31, method = 'BH', n = 96)
p.adjust(p_val_asv32, method = 'BH', n = 96)
p.adjust(p_val_asv33, method = 'BH', n = 96)
p.adjust(p_val_asv34, method = 'BH', n = 96)
p.adjust(p_val_asv35, method = 'BH', n = 96)
p.adjust(p_val_asv36, method = 'BH', n = 96)
p.adjust(p_val_asv37, method = 'BH', n = 96)
p.adjust(p_val_asv38, method = 'BH', n = 96)
p.adjust(p_val_asv39, method = 'BH', n = 96)
p.adjust(p_val_asv40, method = 'BH', n = 96)
p.adjust(p_val_asv41, method = 'BH', n = 96)
p.adjust(p_val_asv42, method = 'BH', n = 96)
p.adjust(p_val_asv43, method = 'BH', n = 96)
p.adjust(p_val_asv44, method = 'BH', n = 96)
p.adjust(p_val_asv45, method = 'BH', n = 96)
p.adjust(p_val_asv46, method = 'BH', n = 96)
p.adjust(p_val_asv47, method = 'BH', n = 96)
p.adjust(p_val_asv48, method = 'BH', n = 96)
p.adjust(p_val_asv49, method = 'BH', n = 96)
p.adjust(p_val_asv50, method = 'BH', n = 96)
p.adjust(p_val_asv51, method = 'BH', n = 96)
p.adjust(p_val_asv52, method = 'BH', n = 96)
p.adjust(p_val_asv53, method = 'BH', n = 96)
p.adjust(p_val_asv54, method = 'BH', n = 96)
p.adjust(p_val_asv55, method = 'BH', n = 96)
p.adjust(p_val_asv56, method = 'BH', n = 96)
p.adjust(p_val_asv57, method = 'BH', n = 96)
p.adjust(p_val_asv58, method = 'BH', n = 96)
p.adjust(p_val_asv59, method = 'BH', n = 96)
p.adjust(p_val_asv60, method = 'BH', n = 96)
p.adjust(p_val_asv61, method = 'BH', n = 96)
p.adjust(p_val_asv62, method = 'BH', n = 96)
p.adjust(p_val_asv63, method = 'BH', n = 96)
p.adjust(p_val_asv64, method = 'BH', n = 96)
p.adjust(p_val_asv65, method = 'BH', n = 96)
p.adjust(p_val_asv66, method = 'BH', n = 96)
p.adjust(p_val_asv67, method = 'BH', n = 96)
p.adjust(p_val_asv68, method = 'BH', n = 96)
p.adjust(p_val_asv69, method = 'BH', n = 96)
p.adjust(p_val_asv70, method = 'BH', n = 96)
p.adjust(p_val_asv71, method = 'BH', n = 96)
p.adjust(p_val_asv72, method = 'BH', n = 96)
p.adjust(p_val_asv73, method = 'BH', n = 96)
p.adjust(p_val_asv74, method = 'BH', n = 96)
p.adjust(p_val_asv75, method = 'BH', n = 96)
p.adjust(p_val_asv76, method = 'BH', n = 96)
p.adjust(p_val_asv77, method = 'BH', n = 96)
p.adjust(p_val_asv78, method = 'BH', n = 96)
p.adjust(p_val_asv79, method = 'BH', n = 96)
p.adjust(p_val_asv80, method = 'BH', n = 96)
p.adjust(p_val_asv81, method = 'BH', n = 96)
p.adjust(p_val_asv82, method = 'BH', n = 96)
p.adjust(p_val_asv83, method = 'BH', n = 96)
p.adjust(p_val_asv84, method = 'BH', n = 96)
p.adjust(p_val_asv85, method = 'BH', n = 96)
p.adjust(p_val_asv86, method = 'BH', n = 96)
p.adjust(p_val_asv87, method = 'BH', n = 96)
p.adjust(p_val_asv88, method = 'BH', n = 96)
p.adjust(p_val_asv89, method = 'BH', n = 96)
p.adjust(p_val_asv90, method = 'BH', n = 96)
p.adjust(p_val_asv91, method = 'BH', n = 96)
p.adjust(p_val_asv92, method = 'BH', n = 96)
p.adjust(p_val_asv93, method = 'BH', n = 96)
p.adjust(p_val_asv94, method = 'BH', n = 96)
p.adjust(p_val_asv95, method = 'BH', n = 96)
p.adjust(p_val_asv96, method = 'BH', n = 96)
```

This gives 9 ASVs that are significant (p.adj < 0.05) with respect to dry weight:
asv89
asv53
asv60
asv09
asv35
asv76
asv69
asv27
asv50

```{r}
percentages_dry_sig <- percentages[,c(1:4,94,58,65,14,40,81,74,32,55)]
write.csv(percentages_dry_sig, file = "~/Documents/Oyster_Project/size_microbiome/percentages_dry_sig.csv")
ggplot(percentages_dry_sig, aes(x = Avg_dry_weight, y = asv89))+
  geom_point()
percentages_dry_sig_edits$Size <- as.factor(percentages_dry_sig_edits$Size)
percentages_dry_sig_edits$Family <- as.factor(percentages_dry_sig_edits$Family)
ggplot(percentages_dry_sig_edits, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV))+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = FALSE)+
  scale_color_manual(values = mkePalette)+
  facet_wrap(.~ASV, nrow = 3)
  
```
ASV09:
```{r}
percentages_dry_sig_edits_09 <- subset(percentages_dry_sig_edits, ASV == "asv09")
asv09_plot <- ggplot(percentages_dry_sig_edits_09, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "gold2", size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "gold2")+
#  geom_label(label = asv09_label, x = 0.3, y = 20)+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = expression(paste("Relationship of ASV09 (", italic("Allivibrio"), ") and spat dry weight")), 
       caption = expression(paste("y = 13x+0.76, ", R^2, ""[adj], "= 0.125")))
ggsave(filename = "asv09_plot.png", plot = asv09_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
# paste(("y=13x + 0.76,"), expression(R^2))
```
ASV27:
```{r}
summary(lm(asv27~ Avg_dry_weight, data = percentages)) # 1.63 - 2.70x, r2 = .095
"#B8DE29FF"
percentages_dry_sig_edits_27 <- subset(percentages_dry_sig_edits, ASV == "asv27")

asv27_plot <- ggplot(percentages_dry_sig_edits_27, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#B8DE29FF", size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#B8DE29FF")+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = "Relationship of ASV27 (Saprospiraceae) and spat dry weight", 
       caption = expression(paste("y = -2.70x + 1.63, ", R^2, ""[adj], "= 0.095")))
ggsave(filename = "asv27_plot.png", plot = asv27_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

ASV35:
```{r}
summary(lm(asv35~ Avg_dry_weight, data = percentages)) # -2.21 + 1.04, r2 = .122
percentages_dry_sig_edits_35 <- subset(percentages_dry_sig_edits, ASV == "asv35")

asv35_plot <- ggplot(percentages_dry_sig_edits_35, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#73D055FF", size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#73D055FF")+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = expression(paste("Relationship of ASV35 (", gamma, "-proteobacteria MBAE14) and spat dry weight")), 
       caption = expression(paste("y = -2.21x + 1.04, ", R^2, ""[adj], "= 0.122")))
ggsave(filename = "asv35_plot.png", plot = asv35_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

ASV50:
```{r}
summary(lm(asv50~ Avg_dry_weight, data = percentages)) # -1.23x + 0.588, r2 = ..0881
percentages_dry_sig_edits_50 <- subset(percentages_dry_sig_edits, ASV == "asv50")

asv50_plot <- ggplot(percentages_dry_sig_edits_50, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#3CBB75FF", size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#3CBB75FF")+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = expression(paste("Relationship of ASV50 (", italic("Aquibacter"), ") and spat dry weight")), 
       caption = expression(paste("y = -1.23x + 0.588, ", R^2, ""[adj], "= 0.0881")))
ggsave(filename = "asv50_plot.png", plot = asv50_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

asv53:
```{r}
summary(lm(asv53~ Avg_dry_weight, data = percentages)) # -1.28x + 0.591, r2 = .151
percentages_dry_sig_edits_53 <- subset(percentages_dry_sig_edits, ASV == "asv53")

asv53_plot <- ggplot(percentages_dry_sig_edits_53, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#20A387FF", size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#20A387FF")+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = expression(paste("Relationship of ASV53 (", italic("Ekhidna"), ") and spat dry weight")), 
       caption = expression(paste("y = -1.28x + 0.591, ", R^2, ""[adj], "= 0.151")))
ggsave(filename = "asv53_plot.png", plot = asv53_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

asv60:
```{r}
"#2D708EFF"
summary(lm(asv60~ Avg_dry_weight, data = percentages)) # -1.65x + 0.702, r2 = .139
percentages_dry_sig_edits_60 <- subset(percentages_dry_sig_edits, ASV == "asv60")

asv60_plot <- ggplot(percentages_dry_sig_edits_60, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#2D708EFF", size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#2D708EFF")+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = expression(paste("Relationship of ASV60 (", italic("Sulfitobacter"), ") and spat dry weight")), 
       caption = expression(paste("y = -1.65x + 0.702, ", R^2, ""[adj], "= 0.139")))
ggsave(filename = "asv60_plot.png", plot = asv60_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

asv69:
```{r}
summary(lm(asv69~ Avg_dry_weight, data = percentages)) # -0.576x + 0.342, r2 = .0973
percentages_dry_sig_edits_69 <- subset(percentages_dry_sig_edits, ASV == "asv69")

asv69_plot <- ggplot(percentages_dry_sig_edits_69, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#39568CFF", size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#39568CFF")+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = expression(paste("Relationship of ASV69 (", italic("Fabibacter"), ") and spat dry weight")), 
       caption = expression(paste("y = -0.576x + 0.342, ", R^2, ""[adj], "= 0.0973")))
ggsave(filename = "asv69_plot.png", plot = asv69_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

asv76:
```{r}
summary(lm(asv76~ Avg_dry_weight, data = percentages)) # -0.715x + 0.352, r2 = .103
percentages_dry_sig_edits_76 <- subset(percentages_dry_sig_edits, ASV == "asv76")

asv76_plot <- ggplot(percentages_dry_sig_edits_76, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#453781FF" , size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#453781FF" )+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = expression(paste("Relationship of ASV76 (", italic("Sphingorhabdus"), ") and spat dry weight")), 
       caption = expression(paste("y = -0.715x + 0.352, ", R^2, ""[adj], "= 0.103")))
ggsave(filename = "asv76_plot.png", plot = asv76_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

asv89:
```{r}

summary(lm(asv89~ Avg_dry_weight, data = percentages)) # 0.757x + 0.0222, r2 = .157
percentages_dry_sig_edits_89 <- subset(percentages_dry_sig_edits, ASV == "asv89")

asv89_plot <- ggplot(percentages_dry_sig_edits_89, aes(x = Avg_dry_weight, y = Percent_samples_reads))+
  geom_point(aes(color = ASV), color = "#481567FF" , size = 2.5)+
  geom_smooth(method = "lm", mapping = aes(color = ASV), se = TRUE, color = "#481567FF" )+
  ylim(0,2.5)+
  theme(panel.background = element_rect(fill = "gray85"))+
  labs(x = "Average dry weight of spat sample (g)", 
       y = "Percent of sample's reads mapping to ASV", 
       title = "Relationship of ASV89 (Oligoflexaceae) and spat dry weight", 
       caption = expression(paste("y = 0.757x + 0.0222, ", R^2, ""[adj], "= 0.157")))
ggsave(filename = "asv89_plot.png", plot = asv89_plot, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

Doing a linear model to see if spat size and read depth are correlated. Added a size variable to the track_size file.
```{r}
track_size <- read_csv("Documents/Oyster_Project/size_microbiome/track_size.csv", 
+     col_types = cols(size = col_factor(levels = c("Small", 
+         "Large"))))
# Size as factor
track_size$size <- as.factor(track_size$size)
track_size$family <- as.factor(track_size$family)
summary(lm(nonchim ~ size, data = track_size))
# Continuous dry weight
track_size$dry_wt <- as.numeric(track_size$dry_wt)
summary(lm(nonchim ~ dry_wt, data = track_size))
# trying to include family as a random effect in a GLMM model.
# paper: https://cran.r-project.org/web/packages/lme4/vignettes/lmer.pdf
library(lme4)
library(afex)
summary(lmer(nonchim ~ size + (1|family), track_size))
# Fixed effect: size
qqnorm(resid(lmer(nonchim ~ size + (1|family), track_size)))
qqline(resid(lmer(nonchim ~ size + (1|family), track_size)))
# The residuals do NOT fall along this line, so they are not normally distributed.
hist(track_size$nonchim, breaks = 80)
```
Graph with nonchim reads on the y-axis and size on the x-axis. One with dry weight, one with size factor (boxplot)
```{r}
# Boxplot first
spat_size_by_size <- ggplot(track_size, aes(x =size, y = nonchim))+
  geom_boxplot(aes(group = size), outlier.shape = NA)+
#  geom_label(x=1.5, y=60, label = "p(t.test)<0.0001", size = 3)+
  geom_point(aes(color = family), position=position_dodge(width = 0.5))+
  labs(title = "Read depth by spat size",
       y = "Filtered reads",
       x = "Spat size",
       caption = "Taxa representing >0.01% of all reads",
       color = "Family")+
  scale_color_manual(values = mkePalette_10)
ggsave(filename = "spat_size_by_size.png", plot = spat_size_by_size, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")

```
By dry weight:
```{r}
spat_size_dry_wt_nonchim <- ggplot(track_size, aes(x =dry_wt, y = nonchim))+
#  geom_boxplot(aes(group = size), outlier.colour = NULL)+
#  geom_label(x=1.5, y=60, label = "p(t.test)<0.0001", size = 3)+
  geom_point(aes(color = family), position=position_dodge(width = 0.5))+
  labs(title = "Read depth by spat size",
       y = "Filtered reads",
       x = "Spat size (dry wt. g)",
       caption = "Taxa representing >0.01% of all reads",
       color = "Family")+
  xlim(0,0.65)+
  scale_color_manual(values = mkePalette_10)
ggsave(filename = "spat_size_dry_wt_nonchim.png", plot = spat_size_dry_wt_nonchim, path = "~/Documents/Oyster_Project/size_microbiome/", scale = 2.5, width = 75, height = 50, units = "mm")
```

SETTING UPPER READ LIMIT OF 15K READS
```{r}
seqtab_size.nochim.df_1000_rare <- rrarefy(seqtab_size.nochim.df_1000, sample=max(15000))
rowSums(seqtab_size.nochim.df_1000_rare)
max(rowSums(seqtab_size.nochim.df_1000_rare)) # 15,000
min(rowSums(seqtab_size.nochim.df_1000_rare)) # 1235
dim(seqtab_size.nochim.df_1000_rare) # 147 x 3259
```
Now all the 147 samples have read depth between 1235 and 15,000 reads.
Assign taxonomy.
```{r}
taxa_size_rare <- assignTaxonomy(seqtab_size.nochim.df_1000_rare, "~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz", multithread=TRUE, tryRC = TRUE)
```
Remove mitochondria and chloroplasts.
```{r}
reduce_taxa <- apply(taxa_size_rare, 1, function(r) any(r %in% c("Chloroplast", "Mitochondria")))
seqtab_size.nochim.df_1000_rare_reduced <- seqtab_size.nochim.df_1000_rare[,!reduce_taxa]
dim(seqtab_size.nochim.df_1000_rare_reduced) # 147 x 3181
write.csv(seqtab_size.nochim.df_1000_rare_reduced, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/seqtab_size.nochim.df_1000_rare_reduced.csv")

taxa_size_1000_rare_reduced <- taxa_size_rare[!reduce_taxa,]
dim(taxa_size_1000_rare_reduced)
asvs_size_rare <- to("asv", nrow(taxa_size_1000_rare_reduced))
dim(taxa_size_1000_rare_reduced) # 3181 x 6
write.csv(taxa_size_1000_rare_reduced, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/taxa_size_1000_reduced.csv")
seqtab_size.nochim.df_1000_rare_reduced_with_asvs <- t(seqtab_size.nochim.df_1000_rare_reduced)
rownames(seqtab_size.nochim.df_1000_rare_reduced_with_asvs) <- asvs_size_rare
write.csv(seqtab_size.nochim.df_1000_rare_reduced_with_asvs, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/seqtab_size.nochim.df_1000_rare_reduced_with_asvs.csv")

```
Need to cut at the taxa that represent > 0.1% of all reads in all 147 samples.

```{r}
min(rowSums(seqtab_size.nochim.df_1000_rare_reduced)) # 1097
max(rowSums(seqtab_size.nochim.df_1000_rare_reduced)) # 14989
sum(rowSums(seqtab_size.nochim.df_1000_rare_reduced)) # 1,409,665
1409665*0.001 # 1410
seqtab_size.nochim.df_1000_rare_reduced <- t(as.data.frame(seqtab_size.nochim.df_1000_rare_reduced))
top0.1 <- seqtab_size.nochim.df_1000_rare_reduced[-which(rowSums(seqtab_size.nochim.df_1000_rare_reduced[sapply(seqtab_size.nochim.df_1000_rare_reduced, is.numeric)]) < 1410), ]
min(colSums(top0.1)) # 1082
max(colSums(top0.1)) # 13972
dim(top0.1) # 101 taxa, 147 samples.
asvs_size_rare_top <- to("asv", nrow(top0.1))
top0.1_with_asvs <- t(top0.1)
rownames(top0.1_with_asvs) <- asvs_size_rare_top
correctednames <- c("3L1", "3L2", "3L3", "3L4", "3L5","3L6","3L7","3L8","3S1","3S2", "3S3", "3S4", "3S5", "3S6", "3S7", "3S8")
colnames(top0.1_with_asvs)[100:115] <- correctednames

write.csv(top0.1_with_asvs, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/top0.1_with_asvs.csv")

#reassign taxonomy
top0.1 <- t(top0.1)
taxa_size_rare_top0.1 <- assignTaxonomy(top0.1, "~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz", multithread=TRUE, tryRC = TRUE)
dim(taxa_size_rare_top0.1) # 101 x 6
seqs_size_rare_top <- rownames(taxa_size_rare_top0.1) # FASTA sequences
taxa_size_rare_top0.1_with_asvs <- taxa_size_rare_top0.1
rownames(taxa_size_rare_top0.1_with_asvs) <- asvs_size_rare_top
write.csv(taxa_size_rare_top0.1_with_asvs, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/taxa_size_rare_top0.1_with_asvs.csv")

seqs_size_strings <- DNAStringSet(seqs_size)
seqs_size_top <- DNAStringSet(seqs_size_rare_top)

names(seqs_size_top) <- asvs_size_rare_top
writeXStringSet(x = seqs_size_top, filepath = "~/Documents/Oyster_Project/size_microbiome/rarefied/seqs_size_top.fa")
```
Align sequences and make tree
\textcolor{blue}{ssh -p 732 englimar@shell.cgrb.oregonstate.edu
SGE_Avail
qrsh -q makaira
mothur}
\
\textcolor{red}{
mothur > align.seqs(fasta=seqs_size_top.fa, reference=/dfs/Mueller_Lab/mk/16s_database/silva.nr_v132.align, flip=t, processors=46)
mothur > filter.seqs(fasta=seqs_size_top.align, processors=46)
mothur > quit()}
\
\textcolor{blue}{
FastTreeMP -gtr -nt -log seqs_size_top.filter.log seqs_size_top.filter.fasta > seqs_size_top.filter.tre
Re-root from midpoint:
/local/cluster/mueller/scripts/Genomics/stat/reroot.pl -midpoint < seqs_size_top.filter.tre > seqs_size_top.filter.midroot.tre

```{r}
library(ape)
spat_rare_size_tree <- read.tree(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/seqs_size_top.filter.midroot.tre")
plot(spat_rare_size_tree)
```
Read in the sequences from mothur.
```{r}
size_rare_top_fasta_filtered <- readDNAStringSet("~/Documents/Oyster_Project/size_microbiome/rarefied/seqs_size_top.filter.fasta")
```

Build phyloseq object.
```{r}
spat_size_meta_reduced <- read.csv("~/Documents/Oyster_Project/size_microbiome/spat_size_meta_reduced.csv", sep = ",", header = T, row.names = 1)
spat_size_meta_reduced$Family <- as.factor(spat_size_meta_reduced$Family)
spat_size_meta_reduced$Size <- as.factor(spat_size_meta_reduced$Size)

top0.1_with_asvs <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/top0.1_with_asvs.csv", sep = ",", header = T, row.names = 1)
rownames(top0.1_with_asvs)
taxa_size_rare_top0.1_with_asvs <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/taxa_size_rare_top0.1_with_asvs.csv", sep = ",", header = T, row.names = 1)
#list_of_samples <- as.list(colnames(top0.1_with_asvs))
write.csv(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/samplelist.csv", x = list_of_samples)
sample_list <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/samplelist.csv")
colnames(top0.1_with_asvs) <- sample_list$Name

spat_size_phylo_rare <- phyloseq(otu_table(top0.1_with_asvs, taxa_are_rows = TRUE), 
                                   sample_data(spat_size_meta_reduced),
                                   tax_table(as.matrix(taxa_size_rare_top0.1_with_asvs)),
                                   phy_tree(spat_rare_size_tree),
                                   refseq(size_rare_top_fasta_filtered))
```

Prune (keeping this in, but it shouldn't be necessary since we already pruned to 1000 reads).
```{r}
spat_size_phylo_rare_sum <- prune_taxa(taxa_sums(spat_size_phylo_rare) > 0, spat_size_phylo_rare)
```
Alpha diversity:
```{r}
library(microbiome)
alpha_size_rare_table <- microbiome::alpha(spat_size_phylo_rare_sum, index = "all")
write.csv(alpha_size_rare_table, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/alpha_size_rare_table.csv")
#added columns with Size, Family, and all weights
#reimport
alpha_size <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/alpha_size_rare_table.csv", sep = ",", header = T, row.names = 1)
alpha_size$Family <- as.factor(alpha_size$Family)
alpha_size$Size <- as.factor(alpha_size$Size)

qqnorm(resid(lm(chao1 ~ Size, data = alpha_size))) # fuck yeah that shit is 100
qqline(resid(lm(chao1 ~ Size, data = alpha_size))) 
qqnorm(resid(lm(chao1 ~ Family, data = alpha_size))) # yes
qqline(resid(lm(chao1 ~ Family, data = alpha_size))) 
qqnorm(resid(lmer(chao1 ~ Size + (1|Family), alpha_size))) # god damn that looks good, w/ family as random
qqline(resid(lmer(chao1 ~ Size + (1|Family), alpha_size)))
qqnorm(resid(lm(chao1 ~ Avg_org_wt, data = alpha_size)))
qqnorm(resid(lm(chao1 ~ Avg_dry_wt, data = alpha_size)))
# All this data is perfectly normally distributed, but also gonna inspect visually
ggplot(alpha_size, aes(x = Avg_dry_wt, y = chao1))+
  geom_point()

# a little heavy to the left
ggplot(alpha_size, aes(x = Avg_org_wt, y = chao1))+
  geom_point()
# again a little heavy to the left
t.test(chao1 ~ Size, data = alpha_size) # p < 0.0001

# considering weight
summary(lm(chao1 ~ Avg_wet_wt, data = alpha_size)) # p = 0.02319
summary(lm(chao1 ~ Avg_dry_wt, data = alpha_size)) # p = 0.0377
summary(lm(chao1 ~ Avg_org_wt, data = alpha_size)) # p = 0.0358

summary(lmer(chao1 ~ Size + (1|Family), alpha_size))
```
Plotting alpha diversity.
```{r}
alpha_size_plot <- ggplot(alpha_size, aes(x =Size, y = chao1))+
  geom_boxplot(aes(group = Size))+
  geom_label(x=1.5, y=60, label = "p(t.test)<0.0001", size = 3)+
  geom_point(aes(color = Family), position=position_dodge(width = 0.5))+
  labs(title = "Microbiome alpha diversity in differentially sized spat",
       y = "Chao1",
       x = "Spat size",
       caption = "Taxa representing >0.01% of all reads\nRarefied data (read depth capped at 15,000 per sample)")+
  scale_color_manual(values = mkePalette_10)

ggsave(filename = "alpha_size_plot.png", plot = alpha_size_plot, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/", scale = 2.5, width = 75, height = 50, units = "mm")
# Shannon index just to have it
summary(lmer(diversity_shannon ~ Size + (1|Family), alpha_size)) # p < 0.0001
t.test(diversity_shannon ~ Size, data = alpha_size) # p < 0.0005
alpha_size_plot_shannon <- ggplot(alpha_size, aes(x =Size, y = diversity_shannon))+
  geom_boxplot(aes(group = Size))+
  geom_label(x=1.5, y=60, label = "p(t.test)<0.0005", size = 3)+
  geom_point(aes(color = Family), position=position_dodge(width = 0.5), show.legend = FALSE)+
  labs(title = "Microbiome alpha diversity in differentially sized spat",
       y = "Shannon diversity index",
       x = "Spat size",
       caption = "Taxa representing >0.01% of all reads\nRarefied data (read depth capped at 15,000 per sample)")+
  scale_color_manual(values = mkePalette_10)
ggsave(filename = "alpha_size_plot_shannon.png", plot = alpha_size_plot_shannon, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/", scale = 2.5, width = 75, height = 50, units = "mm")
```
Differential abundance analysis with ANCOM-BC
```{r}
ancom_out <- ancombc(phyloseq = spat_size_phylo_rare_sum, formula = "Size", 
              p_adj_method = "BH", zero_cut = 0.90,
              group = "Size", struc_zero = FALSE, neg_lb = FALSE, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)
res <- ancom_out$res
res_qval <- res$q_val
write.csv(res_qval, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/qvalues.csv")
res_pval <- res$p_val
write.csv(res_pval, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/pvalues.csv")
res_de <- res$diff_abn
write.csv(res_de, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/de.csv")
# Changing the method of p_adj
ancom_out_2 <- ancombc(phyloseq = spat_size_phylo_rare_sum, formula = "Size",
              p_adj_method = "BH", zero_cut = 0.90,
              group = "Family", struc_zero = TRUE, neg_lb = TRUE, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = FALSE)
res2 <- ancom_out_2$res
res2_qval <- res2$q_val
res2_de <- res2$diff_abn
colnames <- c("p_adj", "DA")
diff_abund <- cbind(res2_qval, res2_de)
colnames(diff_abund) <- colnames
write.csv(diff_abund, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/diff_abund.csv")
```
38/94 are differentially abundant with p < 0.05, but I don't know why there are only 94 and not 101 (as there are 101 ASVs).

Going to try to make a heat map with phylogenetic tree!
```{r}
qvals_in_order <- res2_qval[order(res2_qval$SizeSmall),]
res2_lowq <- rownames(res2$q_val[,res2_qval$SizeSmall])[base::order(res$q_val[,res2_qval$SizeSmall])]
matrix <- as.matrix(data.frame(otu_table(spat_size_phylo_rare_sum)))
rownames(matrix) <- as.character(tax_table(spat_size_phylo_rare_sum)[, "Genus"])
metadata_sub <- data.frame(sample_data(spat_size_phylo_rare_sum))
annotation_color <- data.frame(Size = as.factor(metadata_sub$Size),
                               Spat_Family = as.factor(metadata_sub$Family))
rownames(annotation_color) <- rownames(metadata_sub)
annotation_row <- data.frame(Family = as.factor(tax_table(spat_size_phylo_rare_sum)[,"Family"]))
#rownames(annotation_row) <- rownames(matrix)
```

Beta diversity
```{r}
spat_size_phylo_rare_sum_wuni <- ordinate(spat_size_phylo_rare_sum, "PCoA", "unifrac", weighted=TRUE)
mkePalette_10 <- c("#FDE725FF", "#B8DE29FF", "#73D055FF", "#3CBB75FF", "#20A387FF", "#238A8DFF","#2D708EFF","#39568CFF", "#453781FF", "#481567FF")
spat_size_phylo_rare_sum_wuni_plot <- plot_ordination(spat_size_phylo_rare_sum, spat_size_phylo_rare_sum_wuni, color = "Family", shape = "Size")+
  geom_point(size = 3)+
  scale_color_manual(values = mkePalette_10)+
  scale_fill_manual(values = mkePalette_10)+
#  stat_ellipse(type = "norm")+
  theme(panel.background = element_rect(fill = "gray80"))+
  labs(title = "Beta diversity in microbime of differentially sized spat",
       subtitle = "Weighted UniFrac",
       caption = "Taxa representing >0.01% of all reads\nRarefied data (read depth capped at 15,000 per sample)")

spat_size_phylo_rare_sum_wuni_plot
ggsave(filename = "spat_size_phylo_rare_sum_wuni_plot.png", plot = spat_size_phylo_rare_sum_wuni_plot, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/", scale = 2.5, width = 100, height = 50, units = "mm")
```

Size only
```{r}
spat_size_phylo_rare_sum_family <- plot_ordination(spat_size_phylo_rare_sum, spat_size_phylo_rare_sum_wuni, color = "Size")+
  geom_point(size = 3)+
  scale_color_manual(values = c("green4", "steelblue"))+
  scale_fill_manual(values = c("green4", "steelblue"))+
  theme(panel.background = element_rect(fill = "gray80"))+
  stat_ellipse(type = "norm", linetype = 2)+
  labs(title = "Beta diversity in microbime of differentially sized spat",
       subtitle = "Weighted UniFrac",
       caption = "Taxa representing >0.01% of all reads\nRarefied data (read depth capped at 15,000 per sample)")

ggsave(filename = "spat_size_phylo_rare_sum_wuni_family_plot.png", plot = spat_size_phylo_rare_sum_family, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/", scale = 2.5, width = 100, height = 50, units = "mm")

```

Beta stats
```{r}
spat_size_phylo_rare_sum_wunidist <- UniFrac(spat_size_phylo_rare_sum, weighted=TRUE, normalized=TRUE, parallel=TRUE, fast=TRUE)

spat_size_phylo_rare_sum_wunidist_adonis_size <- adonis2(spat_size_phylo_rare_sum_wunidist ~ Size, data = spat_size_meta_reduced)
spat_size_phylo_rare_sum_wunidist_adonis_size
# p = 0.001, R2 = 0.05292
# Family
spat_size_phylo_rare_sum_wunidist_adonis_family <- adonis2(spat_size_phylo_rare_sum_wunidist ~ Family, data = spat_size_meta_reduced)
spat_size_phylo_rare_sum_wunidist_adonis_family
# p = 0.001, R2 = 0.14191
spat_size_phylo_rare_sum_wunidist_adonis_family_size <- adonis2(spat_size_phylo_rare_sum_wunidist ~ Size + Family, data = spat_size_meta_reduced)
spat_size_phylo_rare_sum_wunidist_adonis_family_size
spat_size_phylo_rare_sum_wunidist_adonis_family_and_size <- adonis2(spat_size_phylo_rare_sum_wunidist ~ Size * Family, data = spat_size_meta_reduced)
spat_size_phylo_rare_sum_wunidist_adonis_family_and_size
adonis2(spat_size_phylo_rare_sum_wunidist ~ Family, data = spat_size_meta_reduced, strata = spat_size_meta_reduced$Size)

# Attempting to make Family a random variable
perm <- how(nperm = 999)
setBlocks(perm) <- with(spat_size_meta_reduced, Family)

spat_size_phylo_rare_sum_wunidist_adonis_size_fam_random <- adonis2(spat_size_phylo_rare_sum_wunidist ~ Size, data = spat_size_meta_reduced, permutations = perm)

spat_size_phylo_rare_sum_wunidist_adonis_size_fam_random
```
Trying DESEQ2 I guess
```{r}
seqtab_low_filter_no_0 <- seqtab_low_filter+1
top0.1_with_asvs_no_0 <- top0.1_with_asvs + 1

spat_size_phylo_rare_no_0 <- phyloseq(otu_table(top0.1_with_asvs_no_0, taxa_are_rows = TRUE), 
                                   sample_data(spat_size_meta_reduced),
                                   tax_table(as.matrix(taxa_size_rare_top0.1_with_asvs)),
                                   phy_tree(spat_rare_size_tree),
                                   refseq(size_rare_top_fasta_filtered))
spat_size_phylo_rare_sum_no_0 <- prune_taxa(taxa_sums(spat_size_phylo_rare_no_0) > 0, spat_size_phylo_rare_no_0)

deseq_spat_size_rare_no_0_size <- phyloseq_to_deseq2(spat_size_phylo_rare_no_0, ~ Size )
deseq_obj_spat_size_rare_no_0_size <- DESeq(deseq_spat_size_rare_no_0_size, test = "Wald", fitType = "local")
deseq_results_obj_spat_size_rare_no_0_size <- results(deseq_obj_spat_size_rare_no_0_size, cooksCutoff = FALSE, pAdjustMethod = "BH")
# make significant results table at p < 0.01
deseq_results_spat_rare_no_0 <- deseq_results_obj_spat_size_rare_no_0_size[which(deseq_results_obj_spat_size_rare_no_0_size$padj < 0.01),]
write.csv(deseq_results_spat_rare_no_0, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/deseq_results_spat_rare_no_0.csv")
```

Considering trimming from 5k-15k reads, instead of 1k-15k based on committee meeting.

Make rarefaction curve to determine a good rarefaction cutoff. Chimeras removed and reads trimmed to between 250 and 256 bp.
```{r}
# full size
rarenames <- rownames(spat_size_meta_reduced)
names
rarecurve(otu_table(spat_size_phylo_rare), step = 50, label = FALSE)
# zoomed in, no labels
rarecurve(otu_table(spat_size_phylo_rare), step = 50, xlim = c(0,15000), label = FALSE)
rarecurve(otu_table(spat_size_phylo_rare), step = 50, xlim = c(0,5000), label = FALSE)
```
Going to move forward with 5000 < x < 15000.
Start:
sequence table: seqtab_size.nochim.df
taxa table: taxa_size
```{r}
# set upper limit of 15,000
seqtab_size.nochim.df.5000.rare <- rrarefy(seqtab_size.nochim.df, sample=max(15000))
rowSums(seqtab_size.nochim.df.5000.rare)
seqtab_size.nochim.df.5000.rare <- as.data.frame(seqtab_size.nochim.df.5000.rare)
seqtab_size.nochim.df.5000.rare_edit <- seqtab_size.nochim.df.5000.rare %>%
  filter(rowSums(across(where(is.numeric))) >= 5000)
rowSums(seqtab_size.nochim.df.5000.rare_edit)
min(rowSums(seqtab_size.nochim.df.5000.rare_edit)) # 5,060
max(rowSums(seqtab_size.nochim.df.5000.rare_edit)) # 15,000
```
New sequence table: seqtab_size.nochim.df.5000.rare_edit.
Remove less abundant taxa.
```{r}
sum(colSums(seqtab_size.nochim.df.5000.rare_edit))
```
1416643 x 0.001 = 1416
Remove ASVs with reads less than 1416 across all samples.
```{r}
seqtab_size.nochim.df.5000_0.01 <-  seqtab_size.nochim.df.5000.rare_edit[,c(colSums(seqtab_size.nochim.df.5000.rare_edit)>1416)]

dim(seqtab_size.nochim.df.5000_0.01) # 128 samples and 104 taxa
seqtab_size.nochim.df.5000_0.01 <- t(seqtab_size.nochim.df.5000_0.01)
write.csv(seqtab_size.nochim.df.5000_0.01, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqtab_5000_0.01.csv")
# make sure minimum is >1416
min(rowSums(seqtab_size.nochim.df.5000_0.01)) # 1443
# convert back to matrix 
seqtab_size.nochim.df.5000_0.01 <- as.matrix(seqtab_size.nochim.df.5000_0.01)
seqtab_size.nochim.df.5000_0.01 <- t(seqtab_size.nochim.df.5000_0.01)
#reassigning taxonomy on reduced ASV table
taxa_size_5000 <- assignTaxonomy(seqtab_size.nochim.df.5000_0.01, "~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz", multithread=TRUE, tryRC = TRUE)
write.csv(taxa_size_5000, "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/taxa_size_5000.csv")
```
Remove mitochondria and chloroplasts. Format tables.
```{r}
seqtab_size.nochim.df.5000_0.01 <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqtab_5000_0.01.csv", header = T, check.names = FALSE, row.names = 1)
taxa_size_5000 <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/taxa_size_5000.csv", header = T)
reduce_taxa <- apply(taxa_size_5000, 1, function(r) any(r %in% c("Chloroplast", "Mitochondria")))
seqtab_size.nochim.df.5000_0.01_reduced <- seqtab_size.nochim.df.5000_0.01[,!reduce_taxa]
dim(seqtab_size.nochim.df.5000_0.01_reduced) # 147 x 97; 7 were removed
write.csv(seqtab_size.nochim.df.5000_0.01_reduced, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqtab_size.nochim.df.5000_0.01_reduced.csv")
seqtab_size.nochim.df.5000_0.01_reduced <- t(seqtab_size.nochim.df.5000_0.01_reduced)
seqtab_size.nochim.df.5000_0.01_reduced_with_asvs <- seqtab_size.nochim.df.5000_0.01_reduced
rownames(seqtab_size.nochim.df.5000_0.01_reduced_with_asvs) <- asvs_5000
write.csv(seqtab_size.nochim.df.5000_0.01_reduced_with_asvs, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqtab_size.nochim.df.5000_0.01_reduced_with_asvs.csv")
taxa_size_5000_reduced <- taxa_size_5000[!reduce_taxa,]
dim(taxa_size_5000_reduced) # 97 x 6
asvs_5000 <- to("asv", nrow(taxa_size_5000_reduced))
write.csv(taxa_size_5000_reduced, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/taxa_size_5000_reduced.csv")
taxa_size_5000_reduced_with_asvs <- taxa_size_5000_reduced
rownames(taxa_size_5000_reduced_with_asvs) <- asvs_5000
write.csv(taxa_size_5000_reduced_with_asvs, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/taxa_size_5000_reduced_with_asvs.csv")
```
Make strings
```{r}
seqs_5000 <- colnames(seqtab_size.nochim.df.5000_0.01_reduced)
seqs_5000_strings <- DNAStringSet(seqs_5000)
names(seqs_5000_strings) <- asvs_5000
writeXStringSet(x = seqs_5000_strings, filepath = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqs_5000_strings.fa")
```

To the makaira
\textcolor{red}{
mothur > align.seqs(fasta=seqs_5000_strings.fa, reference=/dfs/Mueller_Lab/mk/16s_database/silva.nr_v132.align, flip=t, processors=16)
mothur > filter.seqs(fasta=seqs_5000_strings.align, processors=16)
mothur > quit()}
\
\textcolor{blue}{
FastTreeMP -gtr -nt -log seqs_5000_strings.filter.log seqs_5000_strings.filter.fasta > seqs_5000_strings.filter.tre
Re-root from midpoint:
/local/cluster/mueller/scripts/Genomics/stat/reroot.pl -midpoint < seqs_5000_strings.filter.tre > seqs_5000_strings.filter.midroot.tre
}
Read in tree
```{r}
spat_5000_tree <- read.tree(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqs_5000_strings.filter.midroot.tre")
plot(spat_5000_tree)
```
Read in the sequences from mothur.
```{r}
size_5000_fasta_filtered <- readDNAStringSet("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqs_5000_strings.filter.fasta")
```
Edit metadata table, and add read depth as a variable.

Removed samples: "22L5", "24L1", "24L7", "24S7", "24S8", "25L5", "25S5", "29L1", "69L5", "69S8", "74S5", "3S7", "94L1", "94L3", "94L4", "95L1", "95L5", "95S3", "95S5"
```{r}
seqtab_size.nochim.df.5000.rare
getN <- function(x) sum(getUniques(x))
track_size_5000 <- cbind(out_size, sapply(dadaFs_size, getN), sapply(dadaRs_size, getN), sapply(mergers_size, getN), rowSums(seqtab_size.nochim), colSums(seqtab_size.nochim.df.5000_0.01_reduced))
write.csv(colSums(seqtab_size.nochim.df.5000_0.01_reduced), "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/nonplastid.csv")
write.csv(rowSums(seqtab_size.nochim.df.5000.rare), file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/rowsums.csv")
colnames(track_size_5000) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim_reads")
rownames(track_size_5000) <- sample.names_size
write.csv(track_size_5000, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/track_size_5000.csv")
# manually remove rows
spat_size_meta_reduced_reads <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_meta_reduced_reads.csv", sep = ",", header = T, row.names = 1)
spat_size_meta_reduced_reads$Family <- as.factor(spat_size_meta_reduced_reads$Family)
spat_size_meta_reduced_reads$Size <- as.factor(spat_size_meta_reduced_reads$Size)
spat_size_meta_reduced_5000 <- filter(spat_size_meta_reduced_reads, !identifier %in% c("22L5", "24L1", "24L7", "24S7", "24S8", "25L5", "25S5", "29L1", "69L5", "69S8", "74S5", "3S7", "94L1", "94L3", "94L4", "95L1", "95L5", "95S3", "95S5"))
write.csv(spat_size_meta_reduced_5000, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_meta_reduced_5000.csv")
```
Edit ASV table to correct naming error with 8's and 3's.
```{r}
asvs_size5000_with_asvs <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqtab_size.nochim.df.5000_0.01_reduced_with_asvs.csv", sep = ",", header = T, row.names = 1)
colnames(asvs_size5000_with_asvs)
# correct names
seqtab_size.nochim.df.5000_0.01_reduced_with_asvs <- read.csv(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqtab_size.nochim.df.5000_0.01_reduced_with_asvs.csv", sep = ",", header = T, row.names = 1)

#colnames(asvs_size5000_with_asvs) <- spat_size_meta_reduced_5000$identifier # out of order
samples_with_x <- colnames(asvs_size5000_with_asvs)
write.csv(samples_with_x, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/samples_with_x.csv")
samples_without_x <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/samples_without_x.csv", sep = ",", header = T, row.names = 1)

colnames(asvs_size5000_with_asvs) <- samples_without_x$Name
#samples_without_x$Name == rownames(spat_size_meta_reduced_5000)

asvs_size5000_with_asvs_renamed <- asvs_size5000_with_asvs
colnames(asvs_size5000_with_asvs_renamed)
#rownames(spat_size_meta_reduced_5000) == colnames(asvs_size5000_with_asvs_renamed)
```
Craft phyloseq object
```{r}
taxa_size5000_with_asvs <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/taxa_size_5000_reduced_with_asvs.csv", sep = ",", header = T, row.names = 1)

spat_size_meta_reduced_5000 <- read.csv(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_meta_reduced_5000.csv", sep = ",", header = T, row.names = 1)
spat_size_meta_reduced_5000$Spat_family <- as.factor(spat_size_meta_reduced_5000$Spat_family)
spat_size_meta_reduced_5000$Size <- as.factor(spat_size_meta_reduced_5000$Size)
spat_size_meta_reduced_5000$order_for_heat_2 <- as.numeric(spat_size_meta_reduced_5000$order_for_heat_2)
spat_size_meta_reduced_5000$order_for_heat_3 <- as.character(spat_size_meta_reduced_5000$order_for_heat_3)

spat_5000_tree <- ape::read.tree(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqs_5000_strings.filter.midroot.tre", )

size_5000_fasta_filtered <- readDNAStringSet("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqs_5000_strings.filter.fasta")

asvs_size5000_with_asvs <- read.csv(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqtab_size.nochim.df.5000_0.01_reduced_with_asvs.csv", sep = ",", header = T, row.names = 1, check.names = FALSE)

spat_size_phylo_rare_5000 <- phyloseq(otu_table(asvs_size5000_with_asvs, taxa_are_rows = TRUE), 
                                   sample_data(spat_size_meta_reduced_5000),
                                   tax_table(as.matrix(taxa_size5000_with_asvs)),
                                   phy_tree(spat_5000_tree),
                                   refseq(size_5000_fasta_filtered))
```

Beta diversity! And make sure to size the dots by depth, both rarefied and not.
```{r}
spat_size_phylo_rare_5000_wuni <- ordinate(spat_size_phylo_rare_5000, "PCoA", "unifrac", weighted=TRUE)
# Beta diversity plot with dots corresponding to nonchimeric read depth
spat_size_phylo_rare_5000_size_depth_nonchim <- plot_ordination(spat_size_phylo_rare_5000, spat_size_phylo_rare_5000_wuni, color = "Size")+
  geom_point(mapping = aes(size = nonchim_reads))+
  scale_color_manual(values = c("green4", "steelblue"))+
  scale_fill_manual(values = c("green4", "steelblue"))+
  theme(panel.background = element_rect(fill = "gray80"))+
  stat_ellipse(type = "norm", linetype = 2, show.legend = FALSE)+
  labs(title = "Beta diversity in microbiome of differentially sized spat",
       subtitle = "Weighted UniFrac",
       caption = "Rarefied data (read depth 5,000 - 15,000 per sample)\nTaxa representing >0.01% of all reads",
       size = "Unrarefied read depth",
       color = "Spat size class")
ggsave(plot = spat_size_phylo_rare_5000_size_depth_nonchim, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_phylo_rare_5000_size_depth_nonchim.png", width = 200, height = 100, units = "mm")

# Beta diversity plot final
spat_size_phylo_rare_5000_size_rare_depth <- 
  plot_ordination(spat_size_phylo_rare_5000, spat_size_phylo_rare_5000_wuni, color = "Size")+
  geom_point(mapping = aes(fill = Size), size = 4, pch = 21, color = "black", show.legend = TRUE)+
  scale_color_manual(values = c("#FF6600", "steelblue3"))+
  scale_fill_manual(values = c("#FF6600", "steelblue3"))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(color = "gray90", size = 0.25),
        panel.grid.minor = element_line(size = 0),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot",
        legend.text = element_text(size = 10),
        legend.box.spacing = unit(1, "mm"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  stat_ellipse(type = "norm", linetype = 2, show.legend = FALSE)+
  #  geom_label(x=-0.36, y=-0.3, label = "stress = 0.16", size = 5, color = "black", fill = "white", label.size = 0)+
  labs(title = "Spat microbiome by size (Weighted UniFrac)",
       x = "Axis 1 (38.4%)",
       y = "Axis 2 (18.4%)")+
    #   size = "Rarefied read depth")+
    guides(color = guide_legend(override.aes = list(size = 5)))+
    geom_text(label = "p = 0.023", x = -0.47, y = -0.34, stat = "identity", check_overlap = TRUE, size = 3, show.legend = FALSE, color = "black")
ggsave(plot = spat_size_phylo_rare_5000_size_rare_depth, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_phylo_rare_5000_size_rare_depth.jpg", scale = 1.25, width = 200, height = 100, units = "mm")

spat_size_phylo_rare_5000_wunidist <- UniFrac(spat_size_phylo_rare_5000, weighted=TRUE, normalized=TRUE, parallel=TRUE, fast=TRUE)
spat_size_phylo_rare_5000_wunidist_adonis_size <- adonis2(spat_size_phylo_rare_5000_wunidist ~ Size, data = spat_size_meta_reduced_5000)
spat_size_phylo_rare_5000_wunidist_adonis_size
# Basic stats, trying to run adonis2 on a mixed effects model with the "perm" option.
perm <- how(nperm = 999)
setBlocks(perm) <- with(spat_size_meta_reduced_5000, Spat_family)
adonis2(spat_size_phylo_rare_5000_wunidist ~ Spat_family, data = spat_size_meta_reduced_5000) # p = 0.3
spat_size_phylo_rare_5000_wunidist_adonis_size_fam_random <- adonis2(spat_size_phylo_rare_5000_wunidist ~ Size, data = spat_size_meta_reduced_5000, permutations = perm)
adonis2(spat_size_phylo_rare_5000_wunidist ~ nonchim_reads, data = spat_size_meta_reduced_5000)
adonis2(spat_size_phylo_rare_5000_wunidist ~ Size * Spat_family, data = spat_size_meta_reduced_5000)
adonis2(spat_size_phylo_rare_5000_wunidist ~ Spat_family, data = spat_size_meta_reduced_5000) # p = 0.289
adonis2(spat_size_phylo_rare_5000_wunidist ~ Size, data = spat_size_meta_reduced_5000) # p = 0.007
#with family random
setBlocks(perm) <- with(spat_size_meta_reduced_5000, Spat_family)
adonis2(spat_size_phylo_rare_5000_wunidist ~ Size, data = spat_size_meta_reduced_5000, perm = perm)

adonis2(spat_size_phylo_rare_5000_wunidist ~ rare_reads, data = spat_size_meta_reduced_5000) # p = 0.2
qqnorm(resid(spat_size_phylo_rare_5000_wunidist_adonis_size_fam_random))

```
Making family a random effect, not a fixed effect. Mixed effects model using:
Size = fixed
Family = random
Intercept = fixed (UniFrac distance can't be < 0)
Slope = random
```{r}
adonis2(spat_size_phylo_rare_5000_wunidist ~ Size + (1|Spat_family), data = spat_size_meta_reduced_5000)
# doing dry weight; since family 94 is missing, removing that.
spat_size_meta_reduced_5000_no94 <- subset(spat_size_meta_reduced_5000, Spat_family != "94")
merge
spat_size_phylo_rare_5000_no94_wunidist <- UniFrac(spat_size_phylo_rare_5000_no94, weighted=TRUE, normalized = TRUE, parallel=TRUE, fast=TRUE)
spat_size_phylo_rare_5000_no94 <- subset_samples(spat_size_phylo_rare_5000, Spat_family != "94")
metano94 <- subset(spat_size_meta_reduced_5000, Spat_family != "94")
spat_size_phylo_rare_5000_no94_wunidist <- UniFrac(spat_size_phylo_rare_5000_no94, weighted=TRUE, normalized = TRUE, parallel=TRUE, fast=TRUE)
adonis2(spat_size_phylo_rare_5000_no94_wunidist ~ Avg_dry_weight + (0 + Avg_dry_weight | Spat_family), data = metano94)
adonis2(spat_size_phylo_rare_5000_no94_wunidist ~ Size, data = metano94)
metano94$Avg_dry_weight

```


Beta diversity plot where the points are species (colored by Order) instead of samples. This helps you tell which taxa are responsible for the variance.
```{r}
plot_ordination(spat_size_phylo_rare_5000, spat_size_phylo_rare_5000_wuni, type = 'species')+
  geom_point(mapping = aes(color = Order))+
#  geom_text(aes(label = Order), hjust=0, vjust = 0)+
  theme(panel.background = element_rect(fill = "gray80"), legend.key.height = unit(0.5, "mm"), legend.key.size = unit(1, "mm"))+
  guides(col = guide_legend(ncol = 1))+
  stat_ellipse(type = "norm", linetype = 2, show.legend = FALSE)+
  labs(title = "Beta diversity in microbime of differentially sized spat: by taxa",
       subtitle = "Weighted UniFrac",
       caption = "Rarefied data (read depth 5,000 - 15,000 per sample)\nTaxa representing >0.01% of all reads",
       size = "Rarefied read depth",
       color = "Order")
```
Constrained PCoA so that the beta diversity plots can be better separated.
Unconstrained ordination: for a single data matrix.
Constrained: for two or more data sets.
CCA and dbRDA

```{r}
# Calculate the weighted unifrac distance
wunidist_1 <- phyloseq::distance(spat_size_phylo_rare_5000, method = "wunifrac")
# Make data frame
sampdata_1 <- data.frame(sample_data(spat_size_phylo_rare_5000))
# Use adonis on the metadata + the distance data
adonis2(wunidist_1 ~ Size, data = sampdata_1) # p = 0.001
adonis2(wunidist_1 ~ Spat_family, data = sampdata_1) p = 0.001
# Making family random
perm <- how(nperm = 999)
setBlocks(perm) <- with(sampdata_1, Spat_family)
adonis2(wunidist_1 ~ Size, data = sampdata_1, permutations = perm) # p = 0.001

# Beta dispersion
permutest(betadisper(wunidist_1, sampdata_1$Spat_family)) # p = 0.08
permutest(betadisper(wunidist_1, sampdata_1$Size)) # p = 0.348
permutest(betadisper(wunidist_1, sampdata_1$Size), permutations = perm) # 0.304

# trying it with a different method from vegan manual
perm <- 
```
How to interpret this beta dispersion result?
Betadisper tests homogeneity of the dispersion among the samples by a certain grouping (Family, or Size). Basically asks the question "Is dispersion among groups homogeneous, or are there differences by group?". A low p-value for this question would indicate that the dispersion is different for at least one group.
For permutest(betadisper(wunidist_1, sampdata_1$Family)) # p = 0.098, beta dispersion is the same among families (but close to not being the same).
For permutest(betadisper(wunidist_1, sampdata_1$Size)) # p = 0.348, beta dispersion is the same among size class.
For permutest(betadisper(wunidist_1, sampdata_1$Size), permutations = perm) # 0.304, beta dispersion is the same among size class when family is constrained as a random variable. So, either way, this means that I should be able to use adonis2 without any artifacts.

Constrained analysis plots:
```{r}
wunidist_1 <- phyloseq::distance(spat_size_phylo_rare_5000, method =  "wunifrac")
cap_ord_spat <-ordinate(spat_size_phylo_rare_5000, method="CAP", distance=wunidist_1, formula = ~Size)
# has families by color, size by shape.
plot_ordination(spat_size_phylo_rare_5000, cap_ord_spat, color = "Spat_family", shape = "Size", title = "Spat microbiome: CAP | Size") +
  theme_classic()+
  scale_color_manual(values = mkePalette_10)+
  geom_point(size = 3) +
  theme(legend.position = "bottom")
# just looking at size.
# Don't understand this. What am I constricting?
cap_ord_spat <- ordinate(spat_size_phylo_rare_5000, method="CAP", distance=wunidist_1, formula = ~Size)
plot_ordination(spat_size_phylo_rare_5000, cap_ord_spat, color = "Size", title = "Spat microbiome: Constrained analysis of Principle Coordinates | Size") +
  theme_classic()+
  scale_color_manual(values = c("springgreen4", "steelblue4"))+
  geom_point(size = 3) +
  labs(caption = "Family constrained")+
  theme(legend.position = "bottom")

plot(dbrda(formula = wunidist_1 ~ Size + Family, data = sampdata_1))

```

Alpha diversity with the same two variables (rarefied and non-rarefied read depth)
```{r}
library(microbiome)
alpha_rare_size <- microbiome::alpha(spat_size_phylo_rare_5000, index = "all")
write.csv(alpha_rare_size, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/alpha_rare_size.csv")
```
Added read depth, family, size
```{r}
# Need to remove all of sample 74; the weight data for organics was super weird. Removed organic weights from all 74's samples.
#Manually add family, size, 3 types of weights, and 2 types of reads.
alpha_rare_size_meta <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/alpha_rare_size_with_meta.csv", sep = ",", header = T, row.names = 1)
alpha_rare_size_meta$Avg_wet_wt <- as.numeric(alpha_rare_size_meta$Avg_wet_wt)
alpha_rare_size_meta$Avg_dry_wt <- as.numeric(alpha_rare_size_meta$Avg_dry_wt)
alpha_rare_size_meta$Avg_org_wt <- as.numeric(alpha_rare_size_meta$Avg_org_wt)
alpha_rare_size_meta$Nonchim_reads <- as.numeric(alpha_rare_size_meta$Nonchim_reads)
alpha_rare_size_meta$Rare_reads <- as.numeric(alpha_rare_size_meta$Rare_reads)
alpha_rare_size_meta$Size <- as.factor(alpha_rare_size_meta$Size)
alpha_rare_size_meta$Spat_family <- as.factor(alpha_rare_size_meta$Spat_family)
```

Looking at alpha diversity data to make sure it's normally distributed and stats-ready.
```{r}
qqnorm(resid(lm(chao1 ~ Size, data = alpha_rare_size_meta))) # yes
qqline(resid(lm(chao1 ~ Size, data = alpha_rare_size_meta))) 
qqnorm(resid(lm(chao1 ~ Family, data = alpha_rare_size_meta))) # yes
qqline(resid(lm(chao1 ~ Family, data = alpha_rare_size_meta))) 
qqnorm(resid(lmer(chao1 ~ Size + (1|Family), alpha_rare_size_meta))) # looks good, w/ family as random
qqline(resid(lmer(chao1 ~ Size + (1|Family), alpha_rare_size_meta)))
qqnorm(resid(lm(chao1 ~ Avg_org_wt, data = alpha_rare_size_meta)))
qqnorm(resid(lm(chao1 ~ Avg_dry_wt, data = alpha_rare_size_meta)))
# All this data is perfectly normally distributed, but also gonna inspect visually
# Distribution of chao1 with dry weight
ggplot(alpha_rare_size_meta, aes(x = Avg_dry_wt, y = chao1))+
  geom_point()
# Distribution of chao1 with organic weight
# a little heavy to the left
ggplot(alpha_rare_size_meta, aes(x = Avg_org_wt, y = chao1))+
  geom_point()
# again a little heavy to the left
# t.test between chao1 and size
t.test(diversity_shannon ~ Size, data = alpha_rare_size_meta) # p = 0.02155
library(lmerTest)
# considering weight
summary(lm(chao1 ~ Avg_wet_wt, data = alpha_rare_size_meta)) # p = 0.0005
summary(lm(chao1 ~ Avg_dry_wt, data = alpha_rare_size_meta)) # p = 0.0009
summary(lm(chao1 ~ Avg_dry_wt*Rare_reads, data = alpha_rare_size_meta)) # p(interactive) = 0.05
summary(lm(chao1 ~ Avg_org_wt, data = alpha_rare_size_meta)) # p = 0.005
summary(lm(chao1 ~ Avg_org_wt*Rare_reads, data = alpha_rare_size_meta)) # p(interactive) = .05
# mixed effects model with family as random
summary(lmer(diversity_shannon ~ Size + (1|Spat_family), alpha_rare_size_meta)) # p(size) < 0.0001
# QUESTIONS - running into scaling issues
summary(lmer(chao1 ~ Nonchim_reads + (1|Spat_family), alpha_rare_size_meta)) # big effect here
summary(lmer(diversity_shannon ~ Nonchim_reads + (1|Spat_family), alpha_rare_size_meta)) # p =0.019
summary(lmer(diversity_shannon ~ Rare_reads + (1|Spat_family), alpha_rare_size_meta)) # no effect, p = 0.386
summary(lm(diversity_shannon ~ Rare_reads, alpha_rare_size_meta)) # 0.386
summary(lm(diversity_shannon ~ Nonchim_reads, alpha_rare_size_meta)) # .0189
summary(lmer(chao1 ~ Rare_reads + (1|Spat_family), alpha_rare_size_meta)) # p < 0.0001
summary(lm(chao1 ~ Rare_reads, alpha_rare_size_meta)) # big effect
summary(lm(chao1 ~ Nonchim_reads, alpha_rare_size_meta)) # big effect
# rescale?
alpha_rare_size_meta$chao1_400 <- (alpha_rare_size_meta$chao1)*400
summary(lmer(chao1_400 ~ Rare_reads + (1|Spat_family), alpha_rare_size_meta))

# Is there an effect of replicate number? Is there an effect on the number of core ASVs?
summary(lmer(diversity_shannon ~ Reps + (1|Spat_family), alpha_rare_size_meta))
# with family held constant, no -- p = 0.5
# without family held constant, no -- p = 0.5
summary(lm(diversity_shannon ~ Reps, alpha_rare_size_meta))
# What about core ASVs?
summary(lm(CoreASVs ~ Reps, alpha_rare_size_meta))
# huge effect. What if we change it to a factor?
summary(lm(CoreASVs ~ as.factor(Reps), alpha_rare_size_meta)) # still really big
summary(lmer(CoreASVs ~ Reps + (1|Spat_family), alpha_rare_size_meta)) # still big
# visual
ggplot(alpha_rare_size_meta, aes(x = Reps, y = CoreASVs_n1))+
  geom_point(mapping = aes(color = Spat_family, shape = Size), position = position_jitter())

newdf <- alpha_rare_size_meta[c("Spat_family", "Size", "Reps", "CoreASVs_n1", "CoreASVs_75")]
newdf_2 <- subset(newdf, rownames(newdf) %in% c("22L1", "22S1", "24L2", "24S1", "25L1", "25S1", "29L2", "29S1", "40L2", "40S1", "69L1", "69S1", "74L2", "74S1", "3L1", "3S1", "94L2", "94S1", "95L2", "95S1"))
#alpha_with_core_lines <- 
  ggplot(newdf_2, aes(x = Reps))+
 geom_point(mapping = aes(y = CoreASVs_n1, shape = Size), color = "black", size = 5, show.legend = TRUE, position = position_jitter(seed = 1))+
    geom_point(mapping = aes(y = CoreASVs_75, shape = Size), color = "red", size = 5, show.legend = TRUE, position = position_jitter(seed = 2))+
#  geom_point(mapping = aes(color = Spat_family, shape = Size), position = position_jitter(seed = 1), size = 4)+
#  geom_smooth(mapping = aes(linetype = Size), color = "black", se = FALSE, size = 0.5)+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot")+
  labs(y = "Number of core ASVs", x = "Number of replicates",
       title = "Core ASVs in spat microbiome by size class and families",
       color = "Spat family", 
       caption = "p(lm(CoreASVs ~ Reps)) < 0.0001\np(lmer(CoreASVs ~ Reps + (1|Spat_family))) = 0.002")+
  viridis::scale_color_viridis(discrete = TRUE)
ggsave(filename = "alpha_with_core_lines.png", plot = alpha_with_core_lines, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/", scale = 2.5, width = 75, height = 75, units = "mm")
```


Alpha diversity, 5-15k reads, where dot size corresponds to rarefied read depth (5000-15000)
```{r}
alpha_plot_shannon <- ggplot(alpha_rare_size_meta, aes(x =Size, y =diversity_shannon))+
  geom_boxplot(aes(group = Size), outlier.shape = NA)+
  geom_label(x=1.5, y=3.5, label = "p(t.test)=0.02", size = 3)+
  geom_point(mapping = aes(color = Spat_family, size = Rare_reads), position=position_dodge(width = 0.5), show.legend = TRUE)+
#  geom_point(mapping = aes(color = Spat_family, size = Rare_reads), pch = 21, color = "black", show.legend = TRUE, position=position_dodge(width = 0.5, preserve = "single"))+
   labs(title = "Microbiome alpha diversity in differentially sized spat",
       y = "Shannon diversity index",
       x = "Spat size",
       caption = "Rarefied data (read depth 5,000 - 15,000 per sample)\nTaxa representing >0.01% of all reads",
       size = "Rarefied read depth", color = "Spat family")+
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(size = 0),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot")+
  viridis::scale_color_viridis(discrete = TRUE)

ggsave(plot = alpha_plot_shannon, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/alpha_plot_shannon.png", device = "png", scale = 1.5, width = 100, height = 100, units = "mm")
```
Looking at shannon, finals.
```{r}
summary(lm(dominance_simpson ~ Size, data = alpha_rare_size_meta)) # p = 0.529
t.test(dominance_simpson ~ Size, data = alpha_rare_size_meta) # p =0.524
summary(lm(diversity_shannon ~ Size, data = alpha_rare_size_meta)) # p = 0.02378
summary(lm(diversity_shannon ~ Spat_family, data = alpha_rare_size_meta)) # p = 0.321
summary(lm(diversity_shannon ~ Spat_family*Size, data = alpha_rare_size_meta))
summary(lmer(diversity_shannon ~ Size + (1|Spat_family), data = alpha_rare_size_meta)) # p = 0.0237
t.test(diversity_shannon ~ Size, data = alpha_rare_size_meta) # p = 0.022
```


Alpha diversity plot, 5-15k reads, sized by NON-rarefied read depth (5000+)
```{r}
alpha_plot_shannon_nonchim <- ggplot(alpha_rare_size_meta, aes(x =Size, y =diversity_shannon))+
  geom_boxplot(aes(group = Size), outlier.shape = NA)+
  geom_label(x=1.5, y=3.5, label = "p(t.test)=0.02", size = 3)+
  geom_point(mapping = aes(color = Spat_family, size = Nonchim_reads), position=position_dodge(width = 0.5), show.legend = TRUE)+
  labs(title = "Microbiome alpha diversity in differentially sized spat",
       y = "Shannon diversity index",
       x = "Spat size",
       caption = "Rarefied data (read depth 5,000 - 15,000 per sample)\nTaxa representing >0.01% of all reads",
       size = "Unrarefied read depth", color = "Spat family")+
  guides(size = guide_legend(order = 1), color = guide_legend (order = 2))+
  scale_color_manual(values = mkePalette_10)
ggsave(plot = alpha_plot_shannon_nonchim, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/alpha_plot_shannon_nonchim.png", device = "png", scale = 1.5, width = 100, height = 100, units = "mm")
```
Evenness.
```{r}
ggplot(alpha_rare_size_meta, aes(x =Size, y = evenness_simpson))+
  geom_boxplot(aes(group = Size), outlier.shape = NA)+
  geom_label(x=1.5, y=3.5, label = "p(t.test)=0.02", size = 3)+
  geom_point(mapping = aes(color = Spat_family, size = Rare_reads), position=position_dodge(width = 0.5), show.legend = TRUE)+
   labs(title = "Microbiome alpha diversity in differentially sized spat",
       x = "Spat size",
       caption = "Rarefied data (read depth 5,000 - 15,000 per sample)\nTaxa representing >0.01% of all reads",
       size = "Rarefied read depth", color = "Spat family")+
  scale_color_manual(values = mkePalette_10)

t.test(evenness_simpson ~ Size, data = alpha_rare_size_meta)
```

Bar plots that are colored by order. (But, need to rename Family to Spat_family, to avoid taxonomy confusion).
```{r}
spat_size_meta_reduced_5000 <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_meta_reduced_5000.csv", sep = ",", header = T, row.names = 1)
#spat_size_meta_reduced_5000 <- filter(spat_size_meta_reduced_reads, !identifier %in% c("22L5", "24L1", "24L7", "24S7", "24S8", "25L5", "25S5", "29L1", "69L5", "69S8", "74S5", "3S7", "94L1", "94L3", "94L4", "95L1", "95L5", "95S3", "95S5"))
colnames(spat_size_meta_reduced_5000)
#names(spat_size_meta_reduced_5000)[names(spat_size_meta_reduced_5000) == "Family" ] <- "Spat_family"
spat_size_meta_reduced_5000$Spat_family <- as.factor(spat_size_meta_reduced_5000$Spat_family)
# adding replicate (1-8)
spat_size_meta_reduced_5000$Replicate <- as.factor(spat_size_meta_reduced_5000$Replicate)

spat_size_phylo_rare_5000_familyrename <- phyloseq(otu_table(asvs_size5000_with_asvs_renamed, taxa_are_rows = TRUE), 
                                   sample_data(spat_size_meta_reduced_5000),
                                   tax_table(as.matrix(taxa_size5000_with_asvs)),
                                   phy_tree(spat_5000_tree),
                                   refseq(size_5000_fasta_filtered))

#Subset family? and then facet by size
# example with family 22
subset_22 <- subset_samples(spat_size_phylo_rare_5000_familyrename, Spat_family == "22")
# this function converts counts to percents
tform <- transform_sample_counts(subset_22, function(x) x / sum(x))

testplot <- plot_bar(tform, x = "identifier", fill = "Order")

plot_22 <- testplot + facet_grid(.~ Size, scales = "free")
ggsave(plot = plot_22, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/plot_22.png", width = 200, height = 200, units = "mm")

# All small samples
subset_small <- subset_samples(spat_size_phylo_rare_5000_familyrename, Size == "Small")
tform <- transform_sample_counts(subset_small, function(x) x / sum(x))
testplot3 <- plot_bar(tform, x = "identifier", fill = "Order")
final_small_plot <- testplot3 + 
  facet_grid(.~Spat_family, scales = "free")+
  labs(x = "Sample", y = "Relative abundance within sample",
       title = "Spat microbiome by family: Small spat")+
  theme(legend.position = "bottom", legend.spacing.y = unit(1,"mm"), legend.key.height = unit(2,"mm"))

ggsave(plot = final_small_plot, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/final_small_plot.png", width = 250, height = 200, units = "mm")

tform2 <- transform_sample_counts(spat_size_phylo_rare_5000_familyrename, function(x) x / sum(x))
plot_all <- plot_bar(tform2, x = "Replicate", fill = "Order")
plot_final <- plot_all +
  facet_grid(Size~Spat_family, scales = "free")+
  labs(title = "Spat microbiome by family and size")+
  theme(legend.position = "right", legend.spacing.y = unit(1,"mm"), legend.key.height = unit(1,"mm"), 
        axis.text.x = element_text(angle = 0))+
  guides(fill = guide_legend(ncol = 1))
ggsave(plot = plot_final, filename = plot_final, "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/plot_final.png", width = 300, height = 200, units = "mm")
```
Going to start trying to get some results with ANCOM BC.
```{r}
ancom_rare_out <- ancombc(phyloseq = spat_size_phylo_rare_5000_familyrename, formula = "Size", 
              p_adj_method = "BH", zero_cut = 0.90,
              group = "Size", struc_zero = FALSE, neg_lb = FALSE, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

rare_res <- ancom_rare_out$res
rare_res_qval <- rare_res$q_val
write.csv(rare_res_qval, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/qvalues.csv")
rare_res_pval <- rare_res$p_val
write.csv(rare_res_pval, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/pvalues.csv")
rare_res_da <- rare_res$diff_abn
write.csv(rare_res_da, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/dabund.csv")
rare_diff_abund <- cbind(rare_res_qval, rare_res_da)
colnames <- c("P_adj", "Diff_abund")
colnames(rare_diff_abund) <- colnames
write.csv(rare_diff_abund, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/diff_abund.csv")
```
Logistic regression on the 104 taxa, with categorical size variable.
```{r}
tform_logit <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
percent_otu <- as.data.frame(otu_table(tform_logit))
write.csv(t(percent_otu), file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/percent_otus.csv")
# added in all the metadata manually
percent_otus_with_meta <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/percent_otus_with_meta.csv", sep = ",", header = T, row.names = 1)
```

```{r}
# incorrect usage due to binary independent variable
glm_p_val_asv01 <- summary(glm(asv01~Size, family = "binomial", data = percent_otus_with_meta))$coefficients[8]

ttest_p_asv01 <- t.test(asv01 ~ Size, data = percent_otus_with_meta)
```
Testing for normality
```{r}
percent_otu_t <- as.data.frame(t(percent_otu))
shapiro.test(log(percent_otu_t$asv01))
shapiro.test(percent_otu_t$asv02)
# hella abnormal
```
T.test

```{r}
test_phylo <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
test_asv_09 <- as.data.frame(subset(t(otu_table(test_phylo))[,"asv09"]))
test_meta <- as.data.frame(sample_data(test_phylo)$Size)
test_object <- cbind(test_asv_09, test_meta)
colnames(test_object) <- c("Pct_asv_09", "Size")
test_object$Size <- as.factor(test_object$Size)

# t.test
test_phylo <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
t_asv_01 <- t.test(asv01 ~ Size, data = percent_otus_with_meta, alternative = "two.sided")
summary(glm(formula = Pct_asv_09 ~ Size, family = "binomial", data = test_object))
t.test(Pct_asv_09 ~ Size, data = test_object)
test_asv_67 <- as.data.frame(subset(t(otu_table(test_phylo))[,"asv67"]))
test_object_67 <- cbind(test_asv_67, test_meta)
colnames(test_object_67) <- c("Pct_asv_67", "Size")
t.test(Pct_asv_67 ~ Size, data = test_object_67, alternative = "two.sided")

ttest_p_val_asv01 <- t.test(asv01~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv02 <- t.test(asv02~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv03 <- t.test(asv03~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv04 <- t.test(asv04~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv05 <- t.test(asv05~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv06 <- t.test(asv06~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv07 <- t.test(asv07~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv08 <- t.test(asv08~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv09 <- t.test(asv09~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv10 <- t.test(asv10~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv11 <- t.test(asv11~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv12 <- t.test(asv12~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv13 <- t.test(asv13~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv14 <- t.test(asv14~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv15 <- t.test(asv15~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv16 <- t.test(asv16~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv17 <- t.test(asv17~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv18 <- t.test(asv18~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv19 <- t.test(asv19~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv20 <- t.test(asv20~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv21 <- t.test(asv21~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv22 <- t.test(asv22~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv23 <- t.test(asv23~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv24 <- t.test(asv24~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv25 <- t.test(asv25~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv26 <- t.test(asv26~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv27 <- t.test(asv27~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv28 <- t.test(asv28~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv29 <- t.test(asv29~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv30 <- t.test(asv30~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv31 <- t.test(asv31~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv32 <- t.test(asv32~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv33 <- t.test(asv33~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv34 <- t.test(asv34~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv35 <- t.test(asv35~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv36 <- t.test(asv36~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv37 <- t.test(asv37~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv38 <- t.test(asv38~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv39 <- t.test(asv39~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv40 <- t.test(asv40~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv41 <- t.test(asv41~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv42 <- t.test(asv42~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv43 <- t.test(asv43~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv44 <- t.test(asv44~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv45 <- t.test(asv45~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv46 <- t.test(asv46~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv47 <- t.test(asv47~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv48 <- t.test(asv48~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv49 <- t.test(asv49~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv50 <- t.test(asv50~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv51 <- t.test(asv51~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv52 <- t.test(asv52~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv53 <- t.test(asv53~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv54 <- t.test(asv54~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv55 <- t.test(asv55~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv56 <- t.test(asv56~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv57 <- t.test(asv57~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv58 <- t.test(asv58~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv59 <- t.test(asv59~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv60 <- t.test(asv60~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv61 <- t.test(asv61~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv62 <- t.test(asv62~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv63 <- t.test(asv63~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv64 <- t.test(asv64~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv65 <- t.test(asv65~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv66 <- t.test(asv66~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv67 <- t.test(asv67~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv68 <- t.test(asv68~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv69 <- t.test(asv69~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv70 <- t.test(asv70~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv71 <- t.test(asv71~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv72 <- t.test(asv72~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv73 <- t.test(asv73~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv74 <- t.test(asv74~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv75 <- t.test(asv75~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv76 <- t.test(asv76~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv77 <- t.test(asv77~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv78 <- t.test(asv78~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv79 <- t.test(asv79~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv80 <- t.test(asv80~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv81 <- t.test(asv81~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv82 <- t.test(asv82~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv83 <- t.test(asv83~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv84 <- t.test(asv84~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv85 <- t.test(asv85~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv86 <- t.test(asv86~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv87 <- t.test(asv87~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv88 <- t.test(asv88~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv89 <- t.test(asv89~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv90 <- t.test(asv90~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv91 <- t.test(asv91~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv92 <- t.test(asv92~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv93 <- t.test(asv93~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv94 <- t.test(asv94~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv95 <- t.test(asv95~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv96 <- t.test(asv96~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value
ttest_p_val_asv97 <- t.test(asv97~Size, alternative = "two.sided", data = percent_otus_with_meta)$p.value

adjust_ttest_p_val_asv01<-p.adjust(ttest_p_val_asv01, method = "BH", n = 97)
adjust_ttest_p_val_asv02<-p.adjust(ttest_p_val_asv02, method = "BH", n = 97)
adjust_ttest_p_val_asv03<-p.adjust(ttest_p_val_asv03, method = "BH", n = 97)
adjust_ttest_p_val_asv04<-p.adjust(ttest_p_val_asv04, method = "BH", n = 97)
adjust_ttest_p_val_asv05<-p.adjust(ttest_p_val_asv05, method = "BH", n = 97)
adjust_ttest_p_val_asv06<-p.adjust(ttest_p_val_asv06, method = "BH", n = 97)
adjust_ttest_p_val_asv07<-p.adjust(ttest_p_val_asv07, method = "BH", n = 97)
adjust_ttest_p_val_asv08<-p.adjust(ttest_p_val_asv08, method = "BH", n = 97)
adjust_ttest_p_val_asv09<-p.adjust(ttest_p_val_asv09, method = "BH", n = 97)
adjust_ttest_p_val_asv10<-p.adjust(ttest_p_val_asv10, method = "BH", n = 97)
adjust_ttest_p_val_asv11<-p.adjust(ttest_p_val_asv11, method = "BH", n = 97)
adjust_ttest_p_val_asv12<-p.adjust(ttest_p_val_asv12, method = "BH", n = 97)
adjust_ttest_p_val_asv13<-p.adjust(ttest_p_val_asv13, method = "BH", n = 97)
adjust_ttest_p_val_asv14<-p.adjust(ttest_p_val_asv14, method = "BH", n = 97)
adjust_ttest_p_val_asv15<-p.adjust(ttest_p_val_asv15, method = "BH", n = 97)
adjust_ttest_p_val_asv16<-p.adjust(ttest_p_val_asv16, method = "BH", n = 97)
adjust_ttest_p_val_asv17<-p.adjust(ttest_p_val_asv17, method = "BH", n = 97)
adjust_ttest_p_val_asv18<-p.adjust(ttest_p_val_asv18, method = "BH", n = 97)
adjust_ttest_p_val_asv19<-p.adjust(ttest_p_val_asv19, method = "BH", n = 97)
adjust_ttest_p_val_asv20<-p.adjust(ttest_p_val_asv20, method = "BH", n = 97)
adjust_ttest_p_val_asv21<-p.adjust(ttest_p_val_asv21, method = "BH", n = 97)
adjust_ttest_p_val_asv22<-p.adjust(ttest_p_val_asv22, method = "BH", n = 97)
adjust_ttest_p_val_asv23<-p.adjust(ttest_p_val_asv23, method = "BH", n = 97)
adjust_ttest_p_val_asv24<-p.adjust(ttest_p_val_asv24, method = "BH", n = 97)
adjust_ttest_p_val_asv25<-p.adjust(ttest_p_val_asv25, method = "BH", n = 97)
adjust_ttest_p_val_asv26<-p.adjust(ttest_p_val_asv26, method = "BH", n = 97)
adjust_ttest_p_val_asv27<-p.adjust(ttest_p_val_asv27, method = "BH", n = 97)
adjust_ttest_p_val_asv28<-p.adjust(ttest_p_val_asv28, method = "BH", n = 97)
adjust_ttest_p_val_asv29<-p.adjust(ttest_p_val_asv29, method = "BH", n = 97)
adjust_ttest_p_val_asv30<-p.adjust(ttest_p_val_asv30, method = "BH", n = 97)
adjust_ttest_p_val_asv31<-p.adjust(ttest_p_val_asv31, method = "BH", n = 97)
adjust_ttest_p_val_asv32<-p.adjust(ttest_p_val_asv32, method = "BH", n = 97)
adjust_ttest_p_val_asv33<-p.adjust(ttest_p_val_asv33, method = "BH", n = 97)
adjust_ttest_p_val_asv34<-p.adjust(ttest_p_val_asv34, method = "BH", n = 97)
adjust_ttest_p_val_asv35<-p.adjust(ttest_p_val_asv35, method = "BH", n = 97)
adjust_ttest_p_val_asv36<-p.adjust(ttest_p_val_asv36, method = "BH", n = 97)
adjust_ttest_p_val_asv37<-p.adjust(ttest_p_val_asv37, method = "BH", n = 97)
adjust_ttest_p_val_asv38<-p.adjust(ttest_p_val_asv38, method = "BH", n = 97)
adjust_ttest_p_val_asv39<-p.adjust(ttest_p_val_asv39, method = "BH", n = 97)
adjust_ttest_p_val_asv40<-p.adjust(ttest_p_val_asv40, method = "BH", n = 97)
adjust_ttest_p_val_asv41<-p.adjust(ttest_p_val_asv41, method = "BH", n = 97)
adjust_ttest_p_val_asv42<-p.adjust(ttest_p_val_asv42, method = "BH", n = 97)
adjust_ttest_p_val_asv43<-p.adjust(ttest_p_val_asv43, method = "BH", n = 97)
adjust_ttest_p_val_asv44<-p.adjust(ttest_p_val_asv44, method = "BH", n = 97)
adjust_ttest_p_val_asv45<-p.adjust(ttest_p_val_asv45, method = "BH", n = 97)
adjust_ttest_p_val_asv46<-p.adjust(ttest_p_val_asv46, method = "BH", n = 97)
adjust_ttest_p_val_asv47<-p.adjust(ttest_p_val_asv47, method = "BH", n = 97)
adjust_ttest_p_val_asv48<-p.adjust(ttest_p_val_asv48, method = "BH", n = 97)
adjust_ttest_p_val_asv49<-p.adjust(ttest_p_val_asv49, method = "BH", n = 97)
adjust_ttest_p_val_asv50<-p.adjust(ttest_p_val_asv50, method = "BH", n = 97)
adjust_ttest_p_val_asv51<-p.adjust(ttest_p_val_asv51, method = "BH", n = 97)
adjust_ttest_p_val_asv52<-p.adjust(ttest_p_val_asv52, method = "BH", n = 97)
adjust_ttest_p_val_asv53<-p.adjust(ttest_p_val_asv53, method = "BH", n = 97)
adjust_ttest_p_val_asv54<-p.adjust(ttest_p_val_asv54, method = "BH", n = 97)
adjust_ttest_p_val_asv55<-p.adjust(ttest_p_val_asv55, method = "BH", n = 97)
adjust_ttest_p_val_asv56<-p.adjust(ttest_p_val_asv56, method = "BH", n = 97)
adjust_ttest_p_val_asv57<-p.adjust(ttest_p_val_asv57, method = "BH", n = 97)
adjust_ttest_p_val_asv58<-p.adjust(ttest_p_val_asv58, method = "BH", n = 97)
adjust_ttest_p_val_asv59<-p.adjust(ttest_p_val_asv59, method = "BH", n = 97)
adjust_ttest_p_val_asv60<-p.adjust(ttest_p_val_asv60, method = "BH", n = 97)
adjust_ttest_p_val_asv61<-p.adjust(ttest_p_val_asv61, method = "BH", n = 97)
adjust_ttest_p_val_asv62<-p.adjust(ttest_p_val_asv62, method = "BH", n = 97)
adjust_ttest_p_val_asv63<-p.adjust(ttest_p_val_asv63, method = "BH", n = 97)
adjust_ttest_p_val_asv64<-p.adjust(ttest_p_val_asv64, method = "BH", n = 97)
adjust_ttest_p_val_asv65<-p.adjust(ttest_p_val_asv65, method = "BH", n = 97)
adjust_ttest_p_val_asv66<-p.adjust(ttest_p_val_asv66, method = "BH", n = 97)
adjust_ttest_p_val_asv67<-p.adjust(ttest_p_val_asv67, method = "BH", n = 97)
adjust_ttest_p_val_asv68<-p.adjust(ttest_p_val_asv68, method = "BH", n = 97)
adjust_ttest_p_val_asv69<-p.adjust(ttest_p_val_asv69, method = "BH", n = 97)
adjust_ttest_p_val_asv70<-p.adjust(ttest_p_val_asv70, method = "BH", n = 97)
adjust_ttest_p_val_asv71<-p.adjust(ttest_p_val_asv71, method = "BH", n = 97)
adjust_ttest_p_val_asv72<-p.adjust(ttest_p_val_asv72, method = "BH", n = 97)
adjust_ttest_p_val_asv73<-p.adjust(ttest_p_val_asv73, method = "BH", n = 97)
adjust_ttest_p_val_asv74<-p.adjust(ttest_p_val_asv74, method = "BH", n = 97)
adjust_ttest_p_val_asv75<-p.adjust(ttest_p_val_asv75, method = "BH", n = 97)
adjust_ttest_p_val_asv76<-p.adjust(ttest_p_val_asv76, method = "BH", n = 97)
adjust_ttest_p_val_asv77<-p.adjust(ttest_p_val_asv77, method = "BH", n = 97)
adjust_ttest_p_val_asv78<-p.adjust(ttest_p_val_asv78, method = "BH", n = 97)
adjust_ttest_p_val_asv79<-p.adjust(ttest_p_val_asv79, method = "BH", n = 97)
adjust_ttest_p_val_asv80<-p.adjust(ttest_p_val_asv80, method = "BH", n = 97)
adjust_ttest_p_val_asv81<-p.adjust(ttest_p_val_asv81, method = "BH", n = 97)
adjust_ttest_p_val_asv82<-p.adjust(ttest_p_val_asv82, method = "BH", n = 97)
adjust_ttest_p_val_asv83<-p.adjust(ttest_p_val_asv83, method = "BH", n = 97)
adjust_ttest_p_val_asv84<-p.adjust(ttest_p_val_asv84, method = "BH", n = 97)
adjust_ttest_p_val_asv85<-p.adjust(ttest_p_val_asv85, method = "BH", n = 97)
adjust_ttest_p_val_asv86<-p.adjust(ttest_p_val_asv86, method = "BH", n = 97)
adjust_ttest_p_val_asv87<-p.adjust(ttest_p_val_asv87, method = "BH", n = 97)
adjust_ttest_p_val_asv88<-p.adjust(ttest_p_val_asv88, method = "BH", n = 97)
adjust_ttest_p_val_asv89<-p.adjust(ttest_p_val_asv89, method = "BH", n = 97)
adjust_ttest_p_val_asv90<-p.adjust(ttest_p_val_asv90, method = "BH", n = 97)
adjust_ttest_p_val_asv91<-p.adjust(ttest_p_val_asv91, method = "BH", n = 97)
adjust_ttest_p_val_asv92<-p.adjust(ttest_p_val_asv92, method = "BH", n = 97)
adjust_ttest_p_val_asv93<-p.adjust(ttest_p_val_asv93, method = "BH", n = 97)
adjust_ttest_p_val_asv94<-p.adjust(ttest_p_val_asv94, method = "BH", n = 97)
adjust_ttest_p_val_asv95<-p.adjust(ttest_p_val_asv95, method = "BH", n = 97)
adjust_ttest_p_val_asv96<-p.adjust(ttest_p_val_asv96, method = "BH", n = 97)
adjust_ttest_p_val_asv97<-p.adjust(ttest_p_val_asv97, method = "BH", n = 97)
paste('asv01', ttest_p_val_asv01, adjust_ttest_p_val_asv01)
paste('asv02', ttest_p_val_asv02, adjust_ttest_p_val_asv02)
paste('asv03', ttest_p_val_asv03, adjust_ttest_p_val_asv03)
paste('asv04', ttest_p_val_asv04, adjust_ttest_p_val_asv04)
paste('asv05', ttest_p_val_asv05, adjust_ttest_p_val_asv05)
paste('asv06', ttest_p_val_asv06, adjust_ttest_p_val_asv06)
paste('asv07', ttest_p_val_asv07, adjust_ttest_p_val_asv07)
paste('asv08', ttest_p_val_asv08, adjust_ttest_p_val_asv08)
paste('asv09', ttest_p_val_asv09, adjust_ttest_p_val_asv09)
paste('asv10', ttest_p_val_asv10, adjust_ttest_p_val_asv10)
paste('asv11', ttest_p_val_asv11, adjust_ttest_p_val_asv11)
paste('asv12', ttest_p_val_asv12, adjust_ttest_p_val_asv12)
paste('asv13', ttest_p_val_asv13, adjust_ttest_p_val_asv13)
paste('asv14', ttest_p_val_asv14, adjust_ttest_p_val_asv14)
paste('asv15', ttest_p_val_asv15, adjust_ttest_p_val_asv15)
paste('asv16', ttest_p_val_asv16, adjust_ttest_p_val_asv16)
paste('asv17', ttest_p_val_asv17, adjust_ttest_p_val_asv17)
paste('asv18', ttest_p_val_asv18, adjust_ttest_p_val_asv18)
paste('asv19', ttest_p_val_asv19, adjust_ttest_p_val_asv19)
paste('asv20', ttest_p_val_asv20, adjust_ttest_p_val_asv20)
paste('asv21', ttest_p_val_asv21, adjust_ttest_p_val_asv21)
paste('asv22', ttest_p_val_asv22, adjust_ttest_p_val_asv22)
paste('asv23', ttest_p_val_asv23, adjust_ttest_p_val_asv23)
paste('asv24', ttest_p_val_asv24, adjust_ttest_p_val_asv24)
paste('asv25', ttest_p_val_asv25, adjust_ttest_p_val_asv25)
paste('asv26', ttest_p_val_asv26, adjust_ttest_p_val_asv26)
paste('asv27', ttest_p_val_asv27, adjust_ttest_p_val_asv27)
paste('asv28', ttest_p_val_asv28, adjust_ttest_p_val_asv28)
paste('asv29', ttest_p_val_asv29, adjust_ttest_p_val_asv29)
paste('asv30', ttest_p_val_asv30, adjust_ttest_p_val_asv30)
paste('asv31', ttest_p_val_asv31, adjust_ttest_p_val_asv31)
paste('asv32', ttest_p_val_asv32, adjust_ttest_p_val_asv32)
paste('asv33', ttest_p_val_asv33, adjust_ttest_p_val_asv33)
paste('asv34', ttest_p_val_asv34, adjust_ttest_p_val_asv34)
paste('asv35', ttest_p_val_asv35, adjust_ttest_p_val_asv35)
paste('asv36', ttest_p_val_asv36, adjust_ttest_p_val_asv36)
paste('asv37', ttest_p_val_asv37, adjust_ttest_p_val_asv37)
paste('asv38', ttest_p_val_asv38, adjust_ttest_p_val_asv38)
paste('asv39', ttest_p_val_asv39, adjust_ttest_p_val_asv39)
paste('asv40', ttest_p_val_asv40, adjust_ttest_p_val_asv40)
paste('asv41', ttest_p_val_asv41, adjust_ttest_p_val_asv41)
paste('asv42', ttest_p_val_asv42, adjust_ttest_p_val_asv42)
paste('asv43', ttest_p_val_asv43, adjust_ttest_p_val_asv43)
paste('asv44', ttest_p_val_asv44, adjust_ttest_p_val_asv44)
paste('asv45', ttest_p_val_asv45, adjust_ttest_p_val_asv45)
paste('asv46', ttest_p_val_asv46, adjust_ttest_p_val_asv46)
paste('asv47', ttest_p_val_asv47, adjust_ttest_p_val_asv47)
paste('asv48', ttest_p_val_asv48, adjust_ttest_p_val_asv48)
paste('asv49', ttest_p_val_asv49, adjust_ttest_p_val_asv49)
paste('asv50', ttest_p_val_asv50, adjust_ttest_p_val_asv50)
paste('asv51', ttest_p_val_asv51, adjust_ttest_p_val_asv51)
paste('asv52', ttest_p_val_asv52, adjust_ttest_p_val_asv52)
paste('asv53', ttest_p_val_asv53, adjust_ttest_p_val_asv53)
paste('asv54', ttest_p_val_asv54, adjust_ttest_p_val_asv54)
paste('asv55', ttest_p_val_asv55, adjust_ttest_p_val_asv55)
paste('asv56', ttest_p_val_asv56, adjust_ttest_p_val_asv56)
paste('asv57', ttest_p_val_asv57, adjust_ttest_p_val_asv57)
paste('asv58', ttest_p_val_asv58, adjust_ttest_p_val_asv58)
paste('asv59', ttest_p_val_asv59, adjust_ttest_p_val_asv59)
paste('asv60', ttest_p_val_asv60, adjust_ttest_p_val_asv60)
paste('asv61', ttest_p_val_asv61, adjust_ttest_p_val_asv61)
paste('asv62', ttest_p_val_asv62, adjust_ttest_p_val_asv62)
paste('asv63', ttest_p_val_asv63, adjust_ttest_p_val_asv63)
paste('asv64', ttest_p_val_asv64, adjust_ttest_p_val_asv64)
paste('asv65', ttest_p_val_asv65, adjust_ttest_p_val_asv65)
paste('asv66', ttest_p_val_asv66, adjust_ttest_p_val_asv66)
paste('asv67', ttest_p_val_asv67, adjust_ttest_p_val_asv67)
paste('asv68', ttest_p_val_asv68, adjust_ttest_p_val_asv68)
paste('asv69', ttest_p_val_asv69, adjust_ttest_p_val_asv69)
paste('asv70', ttest_p_val_asv70, adjust_ttest_p_val_asv70)
paste('asv71', ttest_p_val_asv71, adjust_ttest_p_val_asv71)
paste('asv72', ttest_p_val_asv72, adjust_ttest_p_val_asv72)
paste('asv73', ttest_p_val_asv73, adjust_ttest_p_val_asv73)
paste('asv74', ttest_p_val_asv74, adjust_ttest_p_val_asv74)
paste('asv75', ttest_p_val_asv75, adjust_ttest_p_val_asv75)
paste('asv76', ttest_p_val_asv76, adjust_ttest_p_val_asv76)
paste('asv77', ttest_p_val_asv77, adjust_ttest_p_val_asv77)
paste('asv78', ttest_p_val_asv78, adjust_ttest_p_val_asv78)
paste('asv79', ttest_p_val_asv79, adjust_ttest_p_val_asv79)
paste('asv80', ttest_p_val_asv80, adjust_ttest_p_val_asv80)
paste('asv81', ttest_p_val_asv81, adjust_ttest_p_val_asv81)
paste('asv82', ttest_p_val_asv82, adjust_ttest_p_val_asv82)
paste('asv83', ttest_p_val_asv83, adjust_ttest_p_val_asv83)
paste('asv84', ttest_p_val_asv84, adjust_ttest_p_val_asv84)
paste('asv85', ttest_p_val_asv85, adjust_ttest_p_val_asv85)
paste('asv86', ttest_p_val_asv86, adjust_ttest_p_val_asv86)
paste('asv87', ttest_p_val_asv87, adjust_ttest_p_val_asv87)
paste('asv88', ttest_p_val_asv88, adjust_ttest_p_val_asv88)
paste('asv89', ttest_p_val_asv89, adjust_ttest_p_val_asv89)
paste('asv90', ttest_p_val_asv90, adjust_ttest_p_val_asv90)
paste('asv91', ttest_p_val_asv91, adjust_ttest_p_val_asv91)
paste('asv92', ttest_p_val_asv92, adjust_ttest_p_val_asv92)
paste('asv93', ttest_p_val_asv93, adjust_ttest_p_val_asv93)
paste('asv94', ttest_p_val_asv94, adjust_ttest_p_val_asv94)
paste('asv95', ttest_p_val_asv95, adjust_ttest_p_val_asv95)
paste('asv96', ttest_p_val_asv96, adjust_ttest_p_val_asv96)
paste('asv97', ttest_p_val_asv97, adjust_ttest_p_val_asv97)
```
Subset phyloseq with significant p-values.
```{r}
asvs_ttest <- subset(otu_table(spat_size_phylo_rare_5000)[c("asv53","asv60", "asv35", "asv69", "asv27", "asv76", "asv50", "asv57", "asv68", "asv79"),])
asvs_ttest_relabund <- otu_table(transform_sample_counts(spat_size_phylo_rare_5000, function(x) x/sum(x)), taxa_are_rows = TRUE)
asvs_ttest_relabund_subset <- subset(otu_table(asvs_ttest_relabund)[c("asv53","asv60", "asv35", "asv69", "asv27", "asv76", "asv50", "asv57", "asv68", "asv79"),])

phylo_ttest_subset <- merge_phyloseq(otu_table(asvs_ttest_relabund_subset, taxa_are_rows = TRUE), sample_data(spat_size_meta_reduced_5000), tax_table(as.matrix(taxa_size5000_with_asvs)), phy_tree(spat_5000_tree), refseq(size_5000_fasta_filtered))

#phylo_ttest_subset <- merge_phyloseq(otu_table(transform_sample_counts(spat_size_phylo_rare_5000, function(x) x/sum(x)), taxa_are_rows = TRUE), sample_data(spat_size_meta_reduced_5000), tax_table(as.matrix(taxa_size5000_with_asvs)), phy_tree(spat_5000_tree), refseq(size_5000_fasta_filtered))
```

Heat map

```{r}
#tform_heat <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))

plot_heatmap(phylo_ttest_subset, method = "Wunifrac", sample.label = "Replicate", taxa.label = NA)+
  labs(asv = "Relative abundance in sample")+
  facet_wrap(.~Spat_family+Size, scales = "free", nrow = 2)


plot_heatmap(phylo_ttest_subset, method = "Wunifrac", sample.label = "Replicate", taxa.label = "Family", sample.order = "order_for_heat_map")
```
Going to try with top 10 ASVs instead.
```{r}
#asvs_top_subset <- subset(otu_table(asvs_ttest_relabund)[c("asv01","asv02", "asv03", "asv04", "asv05", "asv06", "asv07", "asv08", "asv09", "asv10"),])
#phylo_top_subset <- merge_phyloseq(otu_table(asvs_top_subset, taxa_are_rows = TRUE), sample_data(spat_size_meta_reduced_5000), tax_table(as.matrix(taxa_size5000_with_asvs)), phy_tree(spat_5000_tree), refseq(size_5000_fasta_filtered))
library(pheatmap)
#pheatmap(otu_table(phylo_top_subset))
#hclust_asv <- hclust(dist(otu_table(phylo_top_subset)), method = "complete")
library(dendextend)
#as.dendrogram(hclust_asv) %>% plot(horiz = TRUE)
#pheatmap(otu_table(phylo_top_subset), cluster_rows = hclust_asv, cluster_cols = TRUE)

hclust_asv_da <- hclust(dist(otu_table(phylo_ttest_subset)), method = "complete")
as.dendrogram(hclust_asv_da) %>% plot(horiz = TRUE)
#pheatmap(otu_table(asvs_top_subset), cutree_cols = 10)

Heatmap(otu_table(phylo_ttest_subset), cluster_rows = hclust_asv_da, column_order = order(spat_size_meta_reduced_5000$order_for_heat_2, method = "radix"))#, column_labels = spat_size_meta_reduced_5000$order_for_heat_2)#, column_split = spat_size_meta_reduced_5000$Spat_family)
pheatmap(otu_table(phylo_ttest_subset), cluster_rows = hclust_asv_da, cluster_cols = FALSE)
dim(otu_table(phylo_ttest_subset))
```
Trying with gheatmap function.
```{r}
library(ggtree)
meta_subset <- data.frame(sample_data(phylo_ttest_subset))

plot(phy_tree(phylo_ttest_subset))
ttest_relabund_tree <- ggtree(phy_tree(phylo_ttest_subset))
#ggplot(data = data.frame(otu_table(phylo_ttest_subset)), aes(x = ))+
geom_tile()
heatmap <- gheatmap(p = ttest_relabund_tree, data = otu_table(phylo_ttest_subset), low = "dodgerblue", high = "red4")
# how to get facet to work?
heatmap +
  ggplot(data = sample_data(phylo_ttest_subset), aes(x = ))
# New plan: make separate small and large phyloseq objects
asvs_small_ttest_relabund_subset <- subset_samples(phylo_ttest_subset, Size == "Small")
asvs_large_ttest_relabund_subset <- subset_samples(phylo_ttest_subset, Size == "Large")
# Make small phylo object
phylo_ttest_subset_small <- merge_phyloseq(otu_table(asvs_small_ttest_relabund_subset, taxa_are_rows = TRUE), sample_data(spat_size_meta_reduced_5000), tax_table(as.matrix(taxa_size5000_with_asvs)), phy_tree(spat_5000_tree), refseq(size_5000_fasta_filtered))
# Make large phylo object
phylo_ttest_subset_large <- merge_phyloseq(otu_table(asvs_large_ttest_relabund_subset, taxa_are_rows = TRUE), sample_data(spat_size_meta_reduced_5000), tax_table(as.matrix(taxa_size5000_with_asvs)), phy_tree(spat_5000_tree), refseq(size_5000_fasta_filtered))
large_heatmap <- gheatmap(p = ttest_relabund_tree, data = otu_table(asvs_large_ttest_relabund_subset), low = "white", high = "green")
large_heatmap +
  scale_y_continuous(breaks = breakslist)
combo_heatmap <- gheatmap(annotated_tree, otu_table(phylo_ttest_subset))
library(RColorBrewer)
# Small
small_heatmap <- gheatmap(p = ttest_relabund_tree, data = otu_table(asvs_small_ttest_relabund_subset), low = "white", high = "blue")

facet_plot(small_heatmap, panel = "Test", data = data.frame(sample_data(phylo_ttest_subset_small)), geom = geom_tile())

write.csv(asvs_ttest_relabund_subset, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/asvs_relabund.csv")
# add taxa manually
asvs_ttest_relabund_subset_taxa <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/asvs_relabund_with_taxa.csv", sep = ",", header = T, row.names = 1)
colnames(asvs_ttest_relabund_subset_taxa) <- c(samples_without_x$Name, "Blank", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus")

breakslist <- seq(0,0.1, by = 0.001)
pheatmap(otu_table(asvs_small_ttest_relabund_subset), breaks = breakslist, col = colorRampPalette(c("white", "blue"))(100), cluster_cols = FALSE, annotation_names_row = TRUE, show_rownames = TRUE, labels_row = asvnames)
large_heatmap <- pheatmap(otu_table(asvs_large_ttest_relabund_subset), breaks = breakslist, col = colorRampPalette(c("white", "darkgreen"))(100), cluster_cols = FALSE,treeheight_row = 0)

Taxa <- c("ASV79 (Nitrosomonas)", "ASV68 (Woeseiaceae)", "ASV35 (Gammaproteobacteria MBAE14)", "ASV60 (Sulfitobacter)", "ASV76 (Sphingorhabdus)", "ASV57 (Persicirhabdus)", "ASV53 (Ekhidna)", "ASV27 (Saprospiraceae)", "ASV50 (Aquibacter)", "ASV69 (Fabibacter)")

tree<- ggtree(ttest_relabund_tree)
small_heatmap %>% insert_right(large_heatmap)
ttest_relabund_tree
asvs_relabund_with_taxa <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/asvs_relabund_with_taxa.csv")
colnames(asvs_relabund_with_taxa) <- c("ASV", samples_without_x$Name, "Blank", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Code")
otu_for_reorder <- data.frame(otu_table(phylo_ttest_subset), check.names = FALSE)
otu_order <- spat_size_meta_reduced_5000$identifier
otu_for_reorder_complete <- otu_for_reorder[,otu_order]
tax_table(phylo_ttest_subset) <- cbind(tax_table(phylo_ttest_subset), Taxa)
otu_for_reorder_phylo <- otu_table(otu_for_reorder_complete, taxa_are_rows = TRUE)
otu_table(phylo_ttest_subset) <- otu_for_reorder_phylo

tree_with_meta <- ggtree(phylo_ttest_subset, branch.length = "none") %<+% spat_size_meta_reduced_5000
annotated_tree <- tree_with_meta + geom_label(aes(label = Taxa), size = 2, hjust = 1, nudge_y = 0.25) + #geom_tiplab(hjust = 1, vjust = -.9, size = 3, aes(label = Taxa))+
  geom_tippoint(aes(color = Family), size = 2)+geom_tippoint(shape = 1, size = 2, color = "black", show.legend = TRUE)+
scale_color_viridis(discrete = TRUE, option = "C", end = 0.9, na.value = "yellow")

test_plot <- gheatmap(annotated_tree, otu_table(phylo_ttest_subset), colnames_angle = 90, font.size = 0, width = 4, offset = 0.1)+
  scale_fill_gradientn(colors = RColorBrewer::brewer.pal(3, "YlGnBu"), 
                       name = "Relative abundance\n(% of sample's reads)")+
  #solid line between large and small
  geom_segment(aes(x = 14.54, xend = 14.54, y = 0.5, yend = 10.5), size = 0.5)+
  # large family lines
  geom_segment(aes(x = 6.26, xend = 6.26, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 7.22, xend = 7.22, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 8.00, xend = 8.00, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 9.07, xend = 9.07, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 10.36, xend = 10.36, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 11.46, xend = 11.46, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 12.02, xend = 12.02, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 12.86, xend = 12.86, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 13.61, xend = 13.61, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  #small family lines
  geom_segment(aes(x = 15.61, xend = 15.61, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 16.6, xend = 16.6, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 17.34, xend = 17.34, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 18.47, xend = 18.47, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 19.53, xend = 19.53, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 20.77, xend = 20.77, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 21.9, xend = 21.9, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 23, xend = 23, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  geom_segment(aes(x = 24.27, xend = 24.27, y = 0.5, yend = 10.5), size = 0.25, color = "gray40", linetype = 2)+
  # Large family labels
    geom_label(label = "22", x = 5.6, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "24", x = 6.6, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "25", x = 7.5, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "29", x = 8.5, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "3", x = 9.7, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "40", x = 10.9, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "69", x = 11.65, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "74", x = 12.5, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "94", x = 13.2, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "95", x = 14.1, y = 0.05, label.size = 0, size = 3)+
  # Small family labels
  geom_label(label = "22", x = 15.1, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "24", x = 16, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "25", x = 17, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "29", x = 18, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "3", x = 19, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "40", x = 20.2, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "69", x = 21.4, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "74", x = 22.5, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "94", x = 23.6, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "95", x = 24.8, y = 0.05, label.size = 0, size = 3)+
  geom_label(label = "Large", x = 10, y = -0.35, size = 4)+
  geom_label(label = "Small", x = 20, y = -0.35, size = 4)+
  labs(title = "Differentially abundant taxa by size class")

  
ggsave(plot = test_plot, filename = "~/Desktop/test_plot.png", device = "png", scale = 1.5, width = 200, height = 100, units = "mm")

gheatmap(annotated_tree, otu_table(asvs_small_ttest_relabund_subset))+scale_fill_gradient(low = "white", high = "blue")

tree_with_meta_small <- ggtree(asvs_small_ttest_relabund_subset) %<+% spat_size_meta_reduced_5000

annot_small_tree <- tree_with_meta_small + geom_tiplab(hjust = 1, vjust = -.9) + geom_tippoint(aes(color = Family))
gheatmap(annot_small_tree, otu_table(asvs_small_ttest_relabund_subset)+scale_fill_gradient(low = "white", high = "blue"))
```
Pick a subset of ASVs (most abundant, or differentially abundant? We did those that were present in >25% of samples with > 100 reads for eelgrass).
Make 3 models: dry weight + 1 | Family, 1 with dry weight*family

##########

Going to look at alpha and beta diversity, with all taxa, for samples with 5000-15000 reads.

```{r, message = FALSE}
library(dada2)
path <- "~/Documents/Oyster_Project/size_microbiome/fastq_files/"
```
\
Check to make sure your files are there with the list.files() function. You should see a number of files that end in ".fastq".
\
```{r, message = FALSE, results = 'hide'}
list.files(path)
```
\
From that directory, select the forward and reverse reads.\
\
```{r}
fnFs_size <- sort(list.files(path, pattern="_R1.fastq", full.names = TRUE))
fnRs_size <- sort(list.files(path, pattern="_R2.fastq", full.names = TRUE))
```
\
Extract the sample names from the file names and store the sample names in a separate object. Everything after the first underscore will become a sample name.
\
```{r}
sample.names_size <- sapply(strsplit(basename(fnFs_size), "_"), `[`, 1)
```
\
Visualize the quality of your forward and reverse reads.
Note when quality starts to rapidly deteriorate, and write down somewhere the point on the x-axis ("Cycle") where that happens
\
```{r}
plotQualityProfile(fnFs_size[1:2])
# These look really good. 30 to 240
plotQualityProfile(fnRs_size[1:2])
# these are bad. Maybe 30 to 200?
```
\
Distinguish the reads that will be quality-controlled from our original .fastq files .\
```{r}

filtFs_size <- file.path(path, "filtered", paste0(sample.names_size, "_F_filt.fastq.gz"))
filtRs_size <- file.path(path, "filtered", paste0(sample.names_size, "_R_filt.fastq.gz"))
names(filtFs_size) <- sample.names_size
names(filtRs_size) <- sample.names_size
```
\
dada2 has built-in filtering parameters that will probably be fine for your samples.\
\
truncLen(f,r) changes based on your read qualities from the plots above. The default is 0, meaning dada2 won't truncate your reads even if they badly deteriorate at the end. 
\
```{r}
out_size <- filterAndTrim(fnFs_size, filtFs_size, fnRs_size, filtRs_size,
                     truncLen=c(240,200), # Where in the read to truncate (fwd,rev).
                     maxN=0, # tbh not sure about this
                     maxEE=c(2,2), # The number of expected errors allowed in a read.
                     truncQ=2, # Once the read quality goes below 2, the read will truncate.
                     rm.phix=TRUE, # This is to get rid of phiX genome reads, if used in your run.
                     compress=TRUE, # Make the output file small.
                     multithread=TRUE) # On Windows set multithread=FALSE
```
\
3 columns: sample IDs ending in ".fastq", reads.in, and reads.out. reads.out tells you how many of the reads.in made it past the filtering step.
\
```{r}
head(out_size)
```
\
Have dada2 learn the error rates for reads (every time one base is incorrectly substituted for another during sequencing). Error rates for forward and reverse sequences are different.
\
```{r}
errF_size <- learnErrors(filtFs_size, multithread=TRUE)
errR_size <- learnErrors(filtRs_size, multithread=TRUE)
```
\
```{r}
dadaFs_size <- dada(filtFs_size, err=errF_size, multithread=TRUE)
dadaRs_size <- dada(filtRs_size, err=errR_size, multithread=TRUE)
```
\
```{r}
dadaFs_size[[1]]
dadaRs_size[[1]]
```
\
```{r}
mergers_size <- mergePairs(dadaFs_size, filtFs_size, dadaRs_size, filtRs_size, verbose=TRUE)
```
\
```{r}
head(mergers_size[[1]])
```
\
```{r}
seqtab_size_for_diversity <- makeSequenceTable(mergers_size)
```
\
Get the dimensions of your table and write it down:
\
```{r}
dim(seqtab_size_for_diversity)
# 154 x 3380
```
\
View lengths of the sequences, which should be around 252 bp for 16S reads.
```{r}
table(nchar(getSequences(seqtab_size_for_diversity)))
```

\
```{r}
seqtab2_size_for_diversity <- seqtab_size_for_diversity[,nchar(colnames(seqtab_size_for_diversity)) %in% 250:256]
seqtab_size_for_diversity <- seqtab2_size_for_diversity
```
\
Remove chimeras, which are 2 existing sequences mistakenly combined into a new sequence.\
```{r}
seqtab_size_for_diversity.nochim <- removeBimeraDenovo(seqtab_size_for_diversity, method="consensus", multithread=TRUE, verbose=TRUE)
```
\
49 bimeras.
\
Run the dim() command on your new chimera-free sequence table to see how many chimeras there were.
\
```{r}
dim(seqtab_size_for_diversity.nochim)
# 154 x 3259
#write.csv(seqtab_size.nochim, file = "~/Documents/Oyster_Project/size_microbiome/seqtab_size.nochim.csv")
```
\
Keep track of how many sequences you lose at each "filtering" step:
\
```{r}
getN <- function(x) sum(getUniques(x))
track_size_for_diversity <- cbind(out_size, sapply(dadaFs_size, getN), sapply(dadaRs_size, getN), sapply(mergers_size, getN), rowSums(seqtab_size_for_diversity.nochim))
colnames(track_size_for_diversity) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track_size_for_diversity) <- sample.names_size
write.csv(track_size_for_diversity, file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/track_size_for_diversity.csv")
```
\
See how many samples have been lost, from the original "input" to the most recent "nonchim":
\
```{r}
head(track_size_for_diversity)
```
Remove the samples that have less than 5000 reads, and rarefy the samples that have over 15,000.
```{r}
library(vegan)
library(dplyr)
# first rarefy
seqtab_size_for_diversity.nochim_rare <- rrarefy(seqtab_size_for_diversity.nochim, sample=max(15000))
rowSums(seqtab_size_for_diversity.nochim)
max(rowSums(seqtab_size_for_diversity.nochim_rare)) # 15,000
min(rowSums(seqtab_size_for_diversity.nochim_rare)) # 0
# Now remove samples with less than 5000
dim(seqtab_size_for_diversity.nochim_rare) # 154 x 3259
seqtab_size_for_diversity.nochim_rare <- as.data.frame(seqtab_size_for_diversity.nochim_rare)
seqtab_size_for_diversity.nochim_rare_5000 <- seqtab_size_for_diversity.nochim_rare %>%
  filter(rowSums(across(where(is.numeric))) >= 5000)
min(rowSums(seqtab_size_for_diversity.nochim_rare_5000)) # 5060
max(rowSums(seqtab_size_for_diversity.nochim_rare_5000)) # 15000
```
\
```{r}
class(seqtab_size_for_diversity.nochim_rare_5000)
seqtab_size_for_diversity.nochim_rare_5000 <- as.matrix(seqtab_size_for_diversity.nochim_rare_5000)
taxa_for_diversity <- assignTaxonomy(seqtab_size_for_diversity.nochim_rare_5000, "~/Documents/Oyster_Project/sequencing_feb_2020/fastq_files/exp_18/silva_nr_v132_train_set.fa.gz", multithread=TRUE, tryRC = TRUE)
dim(taxa_for_diversity) # 3259 x 6
```
Remove mitochondria and chloroplasts from the current taxa table (taxa_for_diversity) and the asv table (seqtab_size_for_diversity.nochim)
```{r}
reduce_taxa <- apply(taxa_for_diversity, 1, function(r) any(r %in% c("Chloroplast", "Mitochondria")))
```
Now work on the taxonomy table.\
```{r}
library(lessR)
dim(taxa_for_diversity) # 3259 x 6.
taxa_for_diversity_reduced <- taxa_for_diversity[!reduce_taxa,]
asvs_for_diversity <- to("asv", nrow(taxa_for_diversity_reduced))
dim(taxa_for_diversity_reduced) # 3181 x 6
rownames(taxa_for_diversity_reduced) <- asvs_for_diversity
# accidentally overwrote this; will need to re-make
write.csv(taxa_for_diversity_reduced, file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/taxa_for_diversity_reduced_with_asvs.csv")
```

remove from count table\
```{r}
dim(seqtab_size_for_diversity.nochim_rare_5000) # 128 x 3259
seqtab_size_for_diversity.nochim_rare_5000_reduced <- seqtab_size_for_diversity.nochim_rare_5000[,!reduce_taxa]
```
See how many were removed\
```{r}
dim(seqtab_size_for_diversity.nochim_rare_5000_reduced) # 128 x 3181
```
\
Now transpose. ASVs need to be rows.
```{r}
seqtab_size_for_diversity.nochim_rare_5000_reduced <-t(seqtab_size_for_diversity.nochim_rare_5000_reduced)
write.csv(seqtab_size_for_diversity.nochim_rare_5000_reduced, file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/asv_table_with_seqs.csv")
```
Write strings
```{r}
seqs_size_diversity <- rownames(seqtab_size_for_diversity.nochim_rare_5000_reduced)
length(seqs_size_diversity) # 3181
```

Change row names (AFTER getting fasta strings).
```{r}
rownames(seqtab_size_for_diversity.nochim_rare_5000_reduced) <- asvs_for_diversity
write.csv(seqtab_size_for_diversity.nochim_rare_5000_reduced, file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/asv_table_with_asvs.csv")
```

Make a fasta file for your phyloseq object.
```{r}
library(Biostrings)
seqs_size_diversity_strings <- DNAStringSet(seqs_size_diversity)
names(seqs_size_diversity_strings) <- asvs_for_diversity
writeXStringSet(x = seqs_size_diversity_strings, filepath = "~/Documents/Oyster_Project/size_microbiome/for_diversity/seqs_size_diversity_strings.fa")
```
\
For phyloseq
\
First, load packages.
```{r}
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")
library(vegan)
library(lessR)
```
mothur > align.seqs(fasta=seqs_size_diversity_strings.fa, reference=/dfs/Mueller_Lab/mk/16s_database/silva.nr_v132.align, flip=t, processors=16)
mothur > filter.seqs(fasta=seqs_size_diversity_strings.align, processors=16)
mothur > quit()

FastTreeMP -gtr -nt -log seqs_size_diversity_strings.filter.log seqs_size_diversity_strings.filter.fasta > seqs_size_diversity_strings.filter.tre
Re-root from midpoint:
/local/cluster/mueller/scripts/Genomics/stat/reroot.pl -midpoint < seqs_size_diversity_strings.filter.tre > seqs_size_diversity_strings.filter.midroot.tre

Read in and plot tree:
```{r}
library(ape)
# have to unload ggtree, phyloseq, and microbiome in that order to get ape to recognize
spat_size_diversity_tree <- ape::read.tree(file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/seqs_size_diversity_strings.filter.midroot.tre")
plot(spat_size_diversity_tree)
```
Import parts of phyloseq object
```{r}
write.csv(seqtab_size_for_diversity.nochim_rare_5000_reduced, file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/asv_table_with_asvs.csv")
diversity_asvs <- read.csv("~/Documents/Oyster_Project/size_microbiome/for_diversity/asv_table_with_asvs.csv", check.names = FALSE, row.names = 1, header = T, sep = ",")
diversity_taxa <- read.csv("~/Documents/Oyster_Project/size_microbiome/for_diversity/taxa_for_diversity_reduced_with_asvs.csv", check.names = FALSE, row.names = 1, header = T, sep = ",")
diversity_meta <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_meta_reduced_5000.csv", check.names = FALSE, row.names = 1, header = T, sep = ",")
diversity_meta$Size <- as.factor(diversity_meta$Size)
diversity_meta$Spat_family <- as.factor(diversity_meta$Spat_family)
diversity_meta$Replicate <- as.factor(diversity_meta$Replicate)
diversity_meta$Avg_wet_weight <- as.numeric(diversity_meta$Avg_wet_weight)
diversity_meta$Avg_dry_weight <- as.numeric(diversity_meta$Avg_dry_weight)
diversity_meta$Avg_org_weight <- as.numeric(diversity_meta$Avg_org_weight)
size_diversity_fasta_filtered <- readDNAStringSet("~/Documents/Oyster_Project/size_microbiome/for_diversity/seqs_size_diversity_strings.filter.fasta")
```
Build phyloseq object for diversity
```{r}
diversity_size <- phyloseq(otu_table(diversity_asvs, taxa_are_rows = TRUE),
                           sample_data(diversity_meta),
                           tax_table(as.matrix(diversity_taxa)),
                           refseq(size_diversity_fasta_filtered),
                           phy_tree(spat_size_diversity_tree))
# keeping over 100 reads total
diversity_size_sum <- prune_taxa(taxa_sums(diversity_size) > 100, diversity_size)
# keeping samples that are present in at least 25% of reads
in25 <- phyloseq::genefilter_sample(diversity_size_sum, filterfun_sample(function(x) x >= 1), A = 0.25*nsamples(diversity_size_sum))
diversity_size_25 <- prune_taxa(in25, diversity_size_sum)
write.csv(otu_table(diversity_size_25), file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/asvs_in_25.csv")
```
This leaves us with 110 taxa. Note that they are out of order in the ASV table.
Microbiome analysis!
```{r}
alpha_diversity_25 <- microbiome::alpha(diversity_size_25, index = "all")
write.csv(alpha_diversity_25, file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/alpha_diversity_25.csv")
# manually added metadata
alpha_diversity_25_meta <- read.csv("~/Documents/Oyster_Project/size_microbiome/for_diversity/alpha_diversity_25_with_meta.csv", header = T, check.names = FALSE, sep = ",")
alpha_diversity_25_meta$Spat_family <- as.factor(alpha_diversity_25_meta$Spat_family)
alpha_diversity_25_meta$Size <- as.factor(alpha_diversity_25_meta$Size)

qqnorm(resid(lm(diversity_shannon ~ Size, data = alpha_diversity_25_meta))) # fuck yeah that shit is 100
qqnorm(resid(lm(diversity_shannon ~ Spat_family, data = alpha_diversity_25_meta))) # yes
qqnorm(resid(lmer(diversity_shannon ~ Size + (1|Spat_family), alpha_diversity_25_meta))) # god damn that looks good, w/ family as random
# All this data is perfectly normally distributed, but also gonna inspect visually
ggplot(alpha_diversity_25_meta, aes(x = log(Avg_dry_wt), y = diversity_shannon))+
  geom_point()
# a little heavy to the left
ggplot(alpha_diversity_25_meta, aes(x = Avg_dry_wt, y = diversity_shannon))+
  geom_point()

# testing for normality
ggdensity(alpha_diversity_25_meta$diversity_shannon)
shapiro.test(alpha_diversity_25_meta$diversity_shannon) #p = 0.06 just barely normal

t.test(diversity_shannon ~ Size, data = alpha_diversity_25_meta) # p = 0.0075

# considering weight
summary(lm(diversity_shannon ~ Avg_wet_wt, data = alpha_diversity_25_meta)) # p = 0.004
summary(lm(diversity_shannon ~ Avg_dry_wt, data = alpha_diversity_25_meta)) # p = 0.0065
summary(lm(diversity_shannon ~ Avg_org_wt, data = alpha_diversity_25_meta)) # p = 0.007

summary(lmer(diversity_shannon ~ Size + (1|Spat_family), alpha_diversity_25_meta)) # p = 0.008

```
Plotting alpha diversity.
```{r}
alpha_size_plot_for_diversity <- ggplot(alpha_diversity_25_meta, aes(x =Size, y = diversity_shannon))+
  geom_boxplot(aes(group = Size), outlier.shape = NA)+
  geom_label(x=1.5, y=3.5, label = "p(wilcox test) < 0.0001", size = 3)+
  geom_point(aes(color = Spat_family, size = rare_reads), position=position_dodge(width = 0.5))+
  labs(title = "Microbiome alpha diversity in differentially sized spat",
       y = "Shannon diversity index",
       x = "Spat size",
       size = "Rarefied read depth",
       color = "Spat family",
       caption = "Taxa present in at least 25% of samples\nRarefied data (read depth capped at 15,000 per sample)")+
  scale_color_manual(values = mkePalette_10)

ggsave(filename = "alpha_size_plot_for_diversity.png", plot = alpha_size_plot_for_diversity, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/", scale = 2.5, width = 75, height = 50, units = "mm")
# Shannon index just to have it
summary(lmer(diversity_shannon ~ Size + (1|Family), alpha_size)) # p < 0.0001
t.test(diversity_shannon ~ Size, data = alpha_size) # p < 0.0005
alpha_size_plot_shannon <- ggplot(alpha_diversity_25_meta, aes(x =Size, y = diversity_shannon))+
  geom_boxplot(aes(group = Size))+
  geom_label(x=1.5, y=60, label = "p(t.test)<0.0005", size = 3)+
  geom_point(aes(color = Family), position=position_dodge(width = 0.5), show.legend = FALSE)+
  labs(title = "Microbiome alpha diversity in differentially sized spat",
       y = "Shannon diversity index",
       x = "Spat size",
       caption = "Taxa representing >0.01% of all reads\nRarefied data (read depth capped at 15,000 per sample)")+
  scale_color_manual(values = mkePalette_10)
ggsave(filename = "alpha_size_plot_shannon.png", plot = alpha_size_plot_shannon, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/", scale = 2.5, width = 75, height = 50, units = "mm")
```
Beta diversity time 

March 15 2022: starting to look at a core microbiome.
Read in phyloseq object so we can subset it.
```{r}
taxa_size5000_with_asvs <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/taxa_size_5000_reduced_with_asvs.csv", sep = ",", header = T, row.names = 1)
spat_size_meta_reduced_5000 <- read.csv(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_meta_reduced_5000.csv", sep = ",", header = T, row.names = 1)
spat_size_meta_reduced_5000$Spat_family <- as.factor(spat_size_meta_reduced_5000$Spat_family)
spat_size_meta_reduced_5000$Size <- as.factor(spat_size_meta_reduced_5000$Size)
spat_size_meta_reduced_5000$order_for_heat_2 <- as.numeric(spat_size_meta_reduced_5000$order_for_heat_2)
spat_size_meta_reduced_5000$order_for_heat_3 <- as.character(spat_size_meta_reduced_5000$order_for_heat_3)
spat_5000_tree <- ape::read.tree(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqs_5000_strings.filter.midroot.tre", )
size_5000_fasta_filtered <- readDNAStringSet("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/seqs_5000_strings.filter.fasta")
spat_size_phylo_rare_5000 <- phyloseq(otu_table(asvs_size5000_with_asvs_renamed, taxa_are_rows = TRUE), 
                                    sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
# transform it
```
Now subset for ASVs that are present in all members of a family except 1
```{r}
namelist<- unique(spat_size_meta_reduced_5000$Spat_family)
spat_size_phylo_rare_5000 <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
for (name in namelist){    
   # assign(paste0("subset_", name), name) # assigning internal variable called subset_name
#    print(paste("Spat family: ", get("name")))
    subset_name <- subset_samples(spat_size_phylo_rare_5000, Spat_family == name)
    sub_small <- paste0(name, "_small")
    subset_small <- subset_samples(subset_name, Size == "Small")
    subset_large <- subset_samples(subset_name, Size == "Large")
    taxa_family_small_name <- filter_taxa(subset_small, function(x) sum(x > 0) >= ( nsamples(subset_small)-1), prune=FALSE)
    taxa_family_large_name <- filter_taxa(subset_large, function(x) sum(x > 0) >= ( nsamples(subset_large)-1), prune=FALSE)
    phylo_small_name <- subset_taxa(taxa_family_small_name, physeq = subset_small)
    phylo_large_name <- subset_taxa(taxa_family_large_name, physeq = subset_large)
    # announce small
    print(paste("Spat family: ", get("name"), "small"), quote = FALSE)
    print(paste0("Number of asvs: ", dim(tax_table(phylo_small_name))[1]), quote = FALSE)
    print(paste0("Number of replicates: ", dim(sample_data(phylo_small_name))[1]), quote = FALSE)
    # announce large
    #writeLines("\n")
    print(paste("Spat family: ", get("name"), "large"), quote = FALSE)
    print(paste0("Number of asvs: ", dim(tax_table(phylo_large_name))[1]), quote = FALSE)
    print(paste0("Number of replicates: ", dim(sample_data(phylo_large_name))[1]), quote = FALSE)
    #  print.noquote(tax_table(phylo_large_name))
    writeLines("\n")
    #write "small" csv
    write.csv(tax_table(phylo_small_name), file =
                paste0("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/", 
                       name, "_small_taxa.csv", collapse = ""), quote = FALSE)
    write.csv(otu_table(phylo_small_name), file =
                paste0("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/", 
                       name, "_small_asvs.csv", collapse = ""), quote = FALSE)
    # write "large" csv
    write.csv(tax_table(phylo_large_name), file =
                paste0("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/", 
                       name, "_large_taxa.csv", collapse = ""), quote = FALSE)
    write.csv(otu_table(phylo_large_name), file =
                paste0("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/", 
                       name, "_large_asvs.csv", collapse = ""), quote = FALSE)
    # write combo file
    write.csv(cbind(otu_table(phylo_large_name), tax_table(phylo_large_name)), file =
                paste0("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/", 
                       name, "_combos.csv", collapse = ""), quote = FALSE)
    #write tree file
    # write asv file
    # write string set
}
# ok but now going to do 75%, to account for 3/4
for (name in namelist){    
   # assign(paste0("subset_", name), name) # assigning internal variable called subset_name
#    print(paste("Spat family: ", get("name")))
    subset_name <- subset_samples(spat_size_phylo_rare_5000, Spat_family == name)
    sub_small <- paste0(name, "_small")
    subset_small <- subset_samples(subset_name, Size == "Small")
    subset_large <- subset_samples(subset_name, Size == "Large")
    taxa_family_small_name <- filter_taxa(subset_small, function(x) sum(x > 0) >= ( 0.75*nsamples(subset_small)), prune=FALSE)
    taxa_family_large_name <- filter_taxa(subset_large, function(x) sum(x > 0) >= (0.75* nsamples(subset_large)), prune=FALSE)
    phylo_small_name <- subset_taxa(taxa_family_small_name, physeq = subset_small)
    phylo_large_name <- subset_taxa(taxa_family_large_name, physeq = subset_large)
    # announce small
    print(paste("Spat family: ", get("name"), "small"), quote = FALSE)
    print(paste0("Number of asvs: ", dim(tax_table(phylo_small_name))[1]), quote = FALSE)
    print(paste0("Number of replicates: ", dim(sample_data(phylo_small_name))[1]), quote = FALSE)
    # announce large
    #writeLines("\n")
    print(paste("Spat family: ", get("name"), "large"), quote = FALSE)
    print(paste0("Number of asvs: ", dim(tax_table(phylo_large_name))[1]), quote = FALSE)
    print(paste0("Number of replicates: ", dim(sample_data(phylo_large_name))[1]), quote = FALSE)
}
```

Look into the number of subsetted ASVs in each family. Does this correlate with alpha div? What about number of replicates in the family?
```{r}
namelist<- as.list(unique(spat_size_meta_reduced_5000$Spat_family))
subset_small <- subset_samples(spat_size_phylo_rare_5000, Size == "Small")
subset_large <- subset_samples(spat_size_phylo_rare_5000, Size == "Large")

for (name in namelist) {
  assign(paste0("subset_", name), name)
  subset_small <- 
  print(paste0(name, "_small"), quote = FALSE)
  subset_small_name <- subset_samples(subset_small, Spat_family == name)
 # small_filter <- filter_taxa(subset_small_name, function(x) sum(x >= 1) >= (0.85* nsamples(subset_small_name)), prune=FALSE)
 # phylo_small_name <- subset_taxa(small_filter, physeq = subset_small_name)
  print(paste("Spat family: ", get("name"), "small"), quote = FALSE)
    #print(paste0("Number of asvs: ", dim(tax_table(phylo_small_name))[1]), quote = FALSE)
    #print(paste0("Number of replicates: ", dim(sample_data(phylo_small_name))[1]), quote = FALSE)
   # return(phylo_small_name)
}
```

```{r}
subset_small <- function(name, physeq){
  sm.physeq <- physeq
  subset_name <- subset_samples(physeq, Spat_family == name)
  #sub_small <- paste0(name, "_small")
  sub_small <- subset_samples(subset_name, Size == "Small")
  taxa_family_small_name <- filter_taxa(sub_small, function(x) sum(x >= 1) >= 0.85* nsamples(sub_small), prune=FALSE)
  phylo_small_name <- subset_taxa(taxa_family_small_name, physeq = sub_small)

  return (sm.physeq)
}

subset_large <- function(name, physeq){
  lg.physeq <- physeq
  subset_name <- subset_samples(physeq, Spat_family == name)
  subset_large <- subset_samples(subset_name, Size == "Large")
  taxa_family_large_name <- filter_taxa(subset_large, function(x) sum(x >= 1) >= 0.85* nsamples(subset_large), prune=FALSE)
  phylo_large_name <- subset_taxa(taxa_family_large_name, physeq = subset_large)

  return (lg.physeq)
}

ps.list <- lapply(namelist, function(name){
  subset_small(name, spat_size_phylo_rare_5000)
  subset_large(name, spat_size_phylo_rare_5000)
})
```

```{r}
small_samp_3 <- subset_samples(spat_size_phylo_rare_5000, Spat_family == "3" & Size == "Small")
small_subset_3 <- filter_taxa(small_samp_3, function(x) sum(x > 0) >= (nsamples(small_samp_3)-1), prune = TRUE)

large_samp_3 <- subset_samples(spat_size_phylo_rare_5000, Spat_family == "3" & Size == "Large")
large_subset_3 <- filter_taxa(large_samp_3, function(x) sum(x > 0) >= (nsamples(large_samp_3)-1), prune = TRUE)




  subset_name <- subset_samples(spat_size_phylo_rare_5000, Spat_family == name)
  sub_small <- paste0(name, "_small")
  subset_small <- subset_samples(subset_name, Size == "Small")
  subset_large <- subset_samples(subset_name, Size == "Large")
  taxa_family_small_name <- filter_taxa(subset_small, function(x) sum(x > 0) >= ( nsamples(subset_small)-1), prune=FALSE)
namelist_large
phylo_subset <- function(namelist, physeq){
  subset_name <- subset_samples(physeq, Spat_family == namelist)
  subset_name_large <- subset_samples(subset_name, Size == "Large")
  taxa_family_large_name <- filter_taxa(subset_name_large, function(x) sum(x > 0) >= (nsamples(subset_name_large)-1), prune=FALSE)
  phylo_large_name <- subset_taxa(taxa_family_large_name, physeq = subset_name_large)
}
phylo_subset(namelist, spat_size_phylo_rare_5000)
```
Can I make a large spat microbiome?
```{r}
subset_large <- subset_samples(spat_size_phylo_rare_5000, Size == "Large")
taxa_large <- filter_taxa(subset_large, function(x) sum(x > 0) >= (0.9* nsamples(subset_large)), prune=FALSE)
phylo_large_filtered <- subset_taxa(taxa_large, physeq = subset_large)

large_core_barplot <- 
  plot_bar(phylo_large_filtered, x = "Replicate", fill = "Family")+facet_wrap(.~Spat_family, nrow = 2)+
  labs(title = "Large spat core microbiome: taxa found in 90% of samples", y = "Relative abundance in sample", x = "Replicate")+
    ylim(0,1)+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "none")+
  viridis::scale_fill_viridis(option = "C", discrete = TRUE)

ggsave(filename = "large_core_barplot.png", plot = large_core_barplot, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000", scale = 2, width = 85, height = 50, units = "mm")
# Large subset, but 75%, not 90
spat_size_phylo_rare_5000 <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
subset_large <- subset_samples(spat_size_phylo_rare_5000, Size == "Large")
taxa_large_75 <- filter_taxa(subset_large, function(x) sum(x > 0) >= (0.75* nsamples(subset_large)), prune=FALSE)
phylo_large_filtered_75 <- subset_taxa(taxa_large_75, physeq = subset_large)

large_core_75_barplot <- plot_bar(phylo_large_filtered_75, x = "Replicate", fill = "Family")+facet_wrap(.~Spat_family, nrow = 2)+
  labs(title = "Large spat core microbiome: taxa found in 75% of samples", y = "Relative abundance in sample", x = "Replicate")+
    ylim(0,1)+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right")+
  viridis::scale_fill_viridis(option = "C", discrete = TRUE)
ggsave(filename = "large_core_75_barplot.png", plot = large_core_75_barplot, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000", scale = 2, width = 85, height = 50, units = "mm")
```
Small:
```{r}
subset_small <- subset_samples(spat_size_phylo_rare_5000, Size == "Small")
taxa_small <- filter_taxa(subset_small, function(x) sum(x > 0) >= (0.9* nsamples(subset_small)), prune=FALSE)
phylo_small_filtered <- subset_taxa(taxa_small, physeq = subset_small)

small_core_barplot <- 
  plot_bar(phylo_small_filtered, x = "Replicate", fill = "Family")+facet_wrap(.~Spat_family, nrow = 2)+
  labs(title = "Small spat core microbiome: taxa found in 90% of samples", y = "Relative abundance in sample", x = "Replicate")+
    ylim(0,1)+
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "none")+
  viridis::scale_fill_viridis(option = "C", discrete = TRUE, begin = 0.5)
ggsave(filename = "small_core_barplot.png", plot = small_core_barplot, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000", scale = 2, width = 85, height = 50, units = "mm")

# Small at 75%
subset_small <- subset_samples(spat_size_phylo_rare_5000, Size == "Small")
taxa_small_75 <- filter_taxa(subset_small, function(x) sum(x > 0) >= (0.75* nsamples(subset_small)), prune=FALSE)
phylo_small_filtered_75 <- subset_taxa(taxa_small_75, physeq = subset_small)
small_core_75_barplot <- plot_bar(phylo_small_filtered_75, x = "Replicate", fill = "Family")+facet_wrap(.~Spat_family, nrow = 2)+
  labs(title = "Small spat core microbiome: taxa found in 75% of samples", y = "Relative abundance in sample", x = "Replicate")+
    ylim(0,1)+
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right")+
  viridis::scale_fill_viridis(option = "B", discrete = TRUE, begin = 0.1)
ggsave(filename = "small_core_75_barplot.png", plot = small_core_75_barplot, path = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000", scale = 2, width = 85, height = 50, units = "mm")
```
What if I try the core microbiome definition from Ainsworth et al 2015?:
"The core microbiome of each coral species was determined by plotting OTU abundance in the core at 2% intervals (from 0% to 100% of samples). Phylotypes not consistently present in at least 30% of samples, of each preparation type, were considered to represent the individual variability of colonies. In the current study, a phylotype presence in at least 30% of samples was chosen as a conservative representation of the core microbiome. In selecting a 30% representation in the current study as the minimum representation, we determined the minimum percentage of samples at which net change in core OTU abundance from the previous 2% was found to be zero (that is, a rate of change of zero, was found to occur at 30% representation). A 30% core microbiome was found to be the lowest percentage at which core OTU abundance was stable across core microbiomes (Supplementary Figure S2). The coral core microbiome at 50% sample representation and 75% sample representation were also annotated and recorded. Phylotype conservation at 50% and 75% are consistent with previous research on core microbiome annotations. Therefore we also refer to the 50% coral coral microbiome and 75% coral core microbiome"
Function that doesn't work
```{r}
# doesn't work
subset_large <- subset_samples(spat_size_phylo_rare_5000, Size == "Large") # 97
asv_count <- function(subset){
  count = 0
  while (count <= 1)
    count = count + 0.02
#    taxa <- filter_taxa(subset, function(x) sum(x > 0) >= (count*nsamples(subset)), prune = FALSE)
#    assign("taxa", taxa)
    phylo <- subset_taxa(taxa, physeq = subset)
#    assign("phylo", phylo)
    taxa <- filter_taxa(subset, function(x) sum(x > 0) >= (count*nsamples(subset)), prune = FALSE)
    phylo <- subset_taxa(filter_taxa(subset, function(x) sum(x > 0) >= (count*nsamples(subset)), prune = FALSE), physeq = subset)
    print(count)
   # print(dim(otu_table(subset_taxa(filter_taxa(subset, function(x) sum(x > 0) >= (count*nsamples(subset)), prune = FALSE), physeq = subset)))[1])
}

asv_count(subset_large)
```
Big ugly list because I can't get the function to work:
```{r}
taxa_large <- filter_taxa(subset_large, function(x) sum(x > 0) >= (0*nsamples(subset_large)), prune=FALSE)
phylo_large <- subset_taxa(taxa_large, physeq = subset_large)
dim(otu_table(phylo_large))[1]
```
Small:
```{r}
taxa_small <- filter_taxa(subset_small, function(x) sum(x > 0) >= (0*nsamples(subset_small)), prune=FALSE)
phylo_small <- subset_taxa(taxa_small, physeq = subset_small)
dim(otu_table(phylo_small))[1]
```
Graphing them after importing CSV:
```{r}
core_interval_plot <- ggplot(core_intervals, aes(x = Percent_samples, y = Number_of_ASVs))+
  geom_point(mapping = aes(color = Size))+
  geom_line(mapping = aes(color = Size))+
  labs(x = "Proportion of samples containing ASVs",
       y = "Number of ASVs present in X samples",
       title = "Defining a core microbiome", 
       caption = "Method: Ainsworth et al., 2015")+
  scale_color_manual(values = c("forestgreen", "steelblue4"))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right",
        legend.key = element_rect(fill = NA))
ggsave(plot = core_interval_plot, filename = "core_interval_plot.png", path = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000", scale = 2, width = 100, height = 75, units = "mm")
```
These core microbiomes never really level off. I'm going to do the same thing, but with the "diversity" phyloseq object -- which doesn't filter at 0.01% abundance.

Fancy function Michael made for me:
```{r}
subset_small <- subset_samples(diversity_size, Size == "Small")
subset_large <- subset_samples(diversity_size, Size == "Large")
# Create Function to work on subsetted phyloseq objects
asv_count <- function(ps.obj, 
                      sub.name
                      ){
              # Makes a vector of values between 0 and 1, increasing by 0.02 for each successive index
              cutoff.lvls = seq(from = 0, to = 1, by = 0.01)
              
              # Count cumulative sums for each cutoff level
              # lapply iterates through each cutoff level in "cutoff.lvls"
              tmp.sums <- lapply(cutoff.lvls, function(cutoff){
                              # Returns true/false if a particular ASV is present in cutoff_level*number_samples(ps.obj)
                              #   eg. cutoff = 0.02, is an asv present in 2% of the samples? If yes, return True.
                              tmp.isPresent <- filter_taxa(ps.obj, function(x) sum(x > 0) >= (cutoff*nsamples(ps.obj)), prune = FALSE)
                              
                              # What is the sum of ASVs present at each cutoff level.
                              tmp.sumPresent <- sum(tmp.isPresent)
                            })
              
              # Make a dataframe with cutoff levels, sum values, and size group
              tmp.df <- data.frame(Percent_samples = cutoff.lvls,
                                   Number_of_ASVs = unlist(tmp.sums, recursive = F),
                                   Size = sub.name)
  
  return(tmp.df)
}

# Run function for each size group
df.large <- asv_count(subset_large, "Large")
df.small <- asv_count(subset_small, "Small")


# Combine dataframes
df.all <- rbind(df.large, df.small)
write.csv(df.all, file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/core_intervals_diversity.csv")
# Check the dataframe looks good
View(df.all)
```
Determine when rate of change = 0
```{r}
df.all_edited <- read.csv("~/Documents/Oyster_Project/size_microbiome/for_diversity/core_intervals_diversity.csv", sep = ",", header = T, row.names = 1)
df.all_edited$Change <- round(df.all_edited$Change, digits = 2)
```

Now re-graph:
```{r}
core_intervals_diversity <- df.all
core_intervals_diversity$Percent_samples_2 <- core_intervals_diversity$Percent_samples*100
core_interval_diversity_plot <- ggplot(core_intervals_diversity, aes(x = Percent_samples_2, y = Number_of_ASVs))+
  geom_point(mapping = aes(color = Size), size = 1)+
  geom_smooth(mapping = aes(color = Size), se = FALSE, span = 0.1, size = 0.75, fullrange = TRUE)+
#  geom_line(mapping = aes(color = Size), )+
  labs(x = "Percent of samples",
       y = "Number of ASVs present in samples",
       title = "Defining a core microbiome", 
       caption = "Using all taxa (not cut at 0.01% abundance)\nMethod: Modified from Ainsworth et al., 2015")+
  scale_color_manual(values = c("forestgreen", "steelblue4"))+
  scale_x_continuous(limits = c(0,100), expand = c(.01,0))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right",
        legend.key = element_rect(fill = NA))
ggsave(plot = core_interval_diversity_plot, filename = "core_interval_diversity_plot.png", path = "~/Documents/Oyster_Project/size_microbiome/for_diversity", scale = 2, width = 100, height = 75, units = "mm")
```
Same graph as above, but with vertical lines at 0.52 (green) and 0.56 (blue)
```{r}
core_interval_diversity_plot_lines <- ggplot(core_intervals_diversity, aes(x = Percent_samples_2, y = Number_of_ASVs))+
  geom_point(mapping = aes(fill = Size), size = 1.5, shape = 21, color = "black", stroke = 0.25)+
  geom_smooth(mapping = aes(color = Size), se = FALSE, span = 0.1, size = 0.75, fullrange = TRUE, show.legend = FALSE)+
#  geom_line(mapping = aes(color = Size), )+
  labs(x = "Percent of samples",
       y = "Number of ASVs present in samples",
       title = "Defining the spat core microbiome")+
       #caption = "Using all taxa (not cut at 0.01% abundance)\nMethod: Modified from Ainsworth et al., 2015")+
  scale_fill_manual(values = c("#FF6600", "steelblue3"))+
  scale_color_manual(values = c("#FF6600", "steelblue3"))+
  scale_x_continuous(limits = c(0,100), expand = c(.01,0))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right",
        legend.key = element_rect(fill = NA),
        legend.text = element_text(size = 10),
        plot.title = element_text(hjust = 0.5),
        legend.box.spacing = unit(1, "mm"))+
  geom_segment(x = 52, y = 100, xend = 52, yend = 500, color = "#FF6600", linetype = 1, size = 1, arrow = arrow(angle = 30, ends = "first", length = unit(10, "pt")))+
  geom_segment(x = 56, y = 100, xend = 56, yend = 500, color = "steelblue3", linetype = 1, size = 1, arrow = arrow(angle = 30, ends = "first", length = unit(10, "pt")))+
  guides(fill = guide_legend(override.aes = list(size = 4)))
#  geom_segment(x = 52, y = 700, xend = 100, yend = 700, color = "#FF6600", arrow = arrow(angle = 30, ends = "last", length = unit(10, "pt")))+
#  geom_segment(x = 56, y = 500, xend = 100, yend = 500, color = "steelblue3", arrow = arrow(angle = 30, ends = "last", length = unit(10, "pt")))
ggsave(plot = core_interval_diversity_plot_lines, filename = "core_interval_diversity_plot_lines.png", path = "~/Documents/Oyster_Project/size_microbiome/for_diversity", scale = 2, width = 100, height = 75, units = "mm")
```

for large microbiome, setting threshold at 0.52. For small, 0.56
```{r}
spat_size_phylo_rare_5000_core_trans <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
subset_large <- subset_samples(spat_size_phylo_rare_5000_core_trans, Size == "Large")
taxa_large <- filter_taxa(subset_large, function(x) sum(x > 0) >= (0.52*nsamples(subset_large)), prune=FALSE)
phylo_large <- subset_taxa(taxa_large, physeq = subset_large)

large_core <- plot_bar(phylo_large, x = "Replicate", fill = "Class")+facet_wrap(.~Spat_family, nrow = 2)+
  labs(title = "Large spat core microbiome: taxa found in 52% of samples", y = "Relative abundance in sample", x = "Replicate")+
    ylim(0,1)+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right",
        legend.key.height = unit(3,"mm"),
        legend.spacing.y = unit(5, "mm"),
        legend.text = element_text(size = 10))+
  guides(fill = guide_legend(ncol = 1))+
  geom_bar(aes(fill = Class), stat = "identity", position = "stack")+
  scale_fill_manual(values = c("Alphaproteobacteria" = "#238A8DFF",
                               "Bacteroidia" = "#2D708EFF",
                                "Campylobacteria" = "#39568CFF",
                                "Deltaproteobacteria" = "#453781FF",
                                "Gammaproteobacteria" = "#481567FF",
                                "Ignavibacteria" = "royalblue",
                                "Mollicutes" = "#FDE725FF",
                                "Spirochaetia" = "#B8DE29FF"))
#  viridis::scale_fill_viridis(option = "C", discrete = TRUE)
ggsave(plot = large_core, filename = "large_core.png", path = "~/Documents/Oyster_Project/size_microbiome/for_diversity", scale = 2, width = 150, height = 100, units = "mm")
```
Now for small:
```{r}
spat_size_phylo_rare_5000_core_trans <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
subset_small <- subset_samples(spat_size_phylo_rare_5000_core_trans, Size == "Small")
taxa_small <- filter_taxa(subset_small, function(x) sum(x > 0) >= (0.56*nsamples(subset_small)), prune=FALSE)
phylo_small <- subset_taxa(taxa_small, physeq = subset_small)
otu_table(phylo_small)
tax_table(phylo_small)
small_core <- plot_bar(phylo_small, x = "Replicate", fill = "Class")+facet_wrap(.~Spat_family, nrow = 2)+
  labs(title = "Small spat core microbiome: taxa found in 56% of samples", y = "Relative abundance in sample", x = "Replicate")+
    ylim(0,1)+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right",
        legend.key.height = unit(3,"mm"),
        legend.spacing.y = unit(5, "mm"),
        legend.text = element_text(size = 10))+
  guides(fill = guide_legend(ncol = 1))+
  geom_bar(aes(fill = Class), stat = "identity", position = "stack")+
  scale_fill_manual(values = c("Acidimicrobiia" = "skyblue",
                                "Alphaproteobacteria" = "#238A8DFF",
                                "Bacteroidia" = "#2D708EFF",
                                "Campylobacteria" = "#39568CFF",
                                "Deltaproteobacteria" = "#453781FF",
                                "Gammaproteobacteria" = "#481567FF",
                                "Ignavibacteria" = "royalblue",
                                "Mollicutes" = "#FDE725FF",
                                "Planctomycetacia" = "darkblue",
                                "Spirochaetia" = "#B8DE29FF",
                                "Verrucomicrobiae" = "#3CBB75FF",
                                "NA" = "gray45"))
  #viridis::scale_fill_viridis(option = "B", discrete = TRUE, begin = 0.2)
ggsave(plot = small_core, filename = "small_core.png", path = "~/Documents/Oyster_Project/size_microbiome/for_diversity", scale = 2, width = 150, height = 100, units = "mm")
```
New plot idea: need to average these values such that each family & size combo has one value per ASV. There will be 20 "bars" (and 8 or whatever ASV averages per bar). Averages will be down to taxonomic rank of Class.
```{r}
asv_phylo <- data.frame(otu_table(spat_size_phylo_rare_5000_trans), check.names = FALSE)
asv_phylo$asv <- rownames(asv_phylo)
taxa_phylo <- data.frame(tax_table(spat_size_phylo_rare_5000))
taxa_phylo$asv <- rownames(taxa_phylo)
meta_phylo <- data.frame(sample_data(spat_size_phylo_rare_5000))
col_order <- c("asv","Kingdom", "Phylum",  "Class",   "Order",   "Family",  "Genus")
taxa_phylo <- taxa_phylo[, col_order]
asv_taxa <- merge(asv_phylo, taxa_phylo, by = "asv")
rownames(meta_phylo) == colnames(asv_taxa)[2:129]
list <- c("sample", rownames(meta_phylo), rep("NA", times = 6))
asv_taxa$sample <- NULL
asv_taxa$sample <- NA
asv_taxa <- t(asv_taxa)
asv_taxa$sample <- list
summary(small_core)
test <- merge_taxa(asv_phylo, "")
```
Trying with Grace's method!
```{r}
# small and large core mbiome separately
spat_size_phylo_rare_5000_core_trans <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
subset_small <- subset_samples(spat_size_phylo_rare_5000_core_trans, Size == "Small")
taxa_small <- filter_taxa(subset_small, function(x) sum(x > 0) >= (0.56*nsamples(subset_small)), prune=FALSE)
phylo_small <- subset_taxa(taxa_small, physeq = subset_small)

small_df <- data.frame(rownames(tax_table(phylo_small)))
colnames(small_df) <- "ASVs_S"
small_df <- small_df %>% add_column(Size = "Small")
small_df <- small_df[order(small_df$ASVs_S),]
subset_large <- subset_samples(spat_size_phylo_rare_5000_core_trans, Size == "Large")
taxa_large <- filter_taxa(subset_large, function(x) sum(x > 0) >= (0.52*nsamples(subset_large)), prune=FALSE)
phylo_large <- subset_taxa(taxa_large, physeq = subset_large)
large_df <- data.frame(rownames(tax_table(phylo_large)))
colnames(large_df) <- "ASVs_L"

large_df <- large_df %>% add_column(Size = "Large")
large_df <- large_df[order(large_df$ASVs_L),]

comb_subset <- subset(otu_table(spat_size_phylo_rare_5000), rownames(otu_table(spat_size_phylo_rare_5000)) %in% comb_list)

spat_size_phylo_rare_5000_core <- merge_phyloseq(comb_subset, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))

phylo_class <- merge_phyloseq(phylo_large, phylo_small)

phylo_class_large <- phylo_large %>%
  tax_glom(taxrank = "Class") %>%
  psmelt()%>%
  group_by(Spat_family, Class)%>%
  summarise(mean(Abundance)*100)
phylo_class_large$Size <- "Large"
phylo_class_small <- phylo_small %>%
  tax_glom(taxrank = "Class", ) %>%
  psmelt()%>%
  group_by(Spat_family, Class)%>%
  summarise(mean(Abundance)*100)
phylo_class_small$Size <- "Small"

abundances <- data.frame(rbind(phylo_class_large, phylo_class_small))
colnames(abundances)[3] <- "Mean_group_abn"
write.csv(phylo_class, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/abundance_groups.csv")
```
a plot!
```{r}
#abundances <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/abundance_groups.csv")
core_plot <- ggplot(abundances, aes(x = Spat_family, y = Mean_group_abn))+
    geom_bar(aes(fill = Class), stat = "identity", position = "stack")+
    # scale_fill_manual(values = c("Acidimicrobiia" = "skyblue",
    #                             "Alphaproteobacteria" = "#238A8DFF",
    #                             "Bacteroidia" = "#2D708EFF",
    #                             "Campylobacteria" = "#39568CFF",
    #                             "Deltaproteobacteria" = "#453781FF",
    #                             "Gammaproteobacteria" = "#481567FF",
    #                             "Ignavibacteria" = "royalblue",
    #                             "Mollicutes" = "#FDE725FF",
    #                             "Planctomycetacia" = "darkblue",
    #                             "Spirochaetia" = "#B8DE29FF",
    #                             "Verrucomicrobiae" = "3CBB75FF",
    #                             "NA" = "gray45"))+
  scale_fill_manual(values = c("Acidimicrobiia" = "deepskyblue",
                    "Alphaproteobacteria" = "#E41A1C",
                    "Bacteroidia" = "#377EB8",
                    "Campylobacteria" = "#4DAF4A",
                    "Deltaproteobacteria" = "#984EA3",
                    "Gammaproteobacteria" = "#FF7F00",
                    "Ignavibacteria" = "royalblue",
                    "Mollicutes" = "#FDE725FF",
                    "Planctomycetacia" = "darkblue",
                    "Spirochaetia" = "gray80",
                    "Verrucomicrobiae" = "black",
                    "NA" = "gray45"))+
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major.y = element_line(color = "gray90"),
        panel.grid.minor.y = element_line(color = "gray90"),
        panel.grid.major.x = element_line(size = 0),
        panel.grid.minor.x = element_line(size = 0),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot", 
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  labs(title = "Spat core microbiome by size", x = "Spat family", y = "Mean relative class abundance (% reads)")+
  scale_y_continuous(limits = c(0,100), expand = expansion(add = c(1,0)))+
  facet_wrap(.~Size)

ggsave(plot = core_plot, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/core_plot.tiff", scale = 1, width = 180, height = 150, units = "mm")
```


How different are these microbiomes? Measure with beta diversity. First, subset the phyloseq object to include only taxa included in the core microbiome for small, and the core microbiome for large. 
```{r}
spat_size_phylo_rare_5000 <- phyloseq(otu_table(asvs_size5000_with_asvs_renamed, taxa_are_rows = TRUE), 
                                    sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
saveRDS(spat_size_phylo_rare_5000, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_phylo_rare_5000.rds")
subset_small <- subset_samples(spat_size_phylo_rare_5000, Size == "Small")
taxa_small_75 <- filter_taxa(subset_small, function(x) sum(x > 0) >= (0.75* nsamples(subset_small)), prune=FALSE)
phylo_small_filtered_75 <- subset_taxa(taxa_small_75, physeq = subset_small)
subset_large <- subset_samples(spat_size_phylo_rare_5000, Size == "Large")
taxa_large_75 <- filter_taxa(subset_large, function(x) sum(x > 0) >= (0.75* nsamples(subset_large)), prune=FALSE)
phylo_large_filtered_75 <- subset_taxa(taxa_large_75, physeq = subset_large)
spat_size_phylo_rare_5000_trans <- transform_sample_counts(spat_size_phylo_rare_5000, function(x) x / sum(x))
```
Wait -- merging the two will not work. I need to filter based on a combined list of asvs that are found in one and another found in the other.
```{r}
large_df <- data.frame(rownames(tax_table(phylo_large)))
colnames(large_df) <- "ASVs_L"
#large_df <- large_df %>% add_row(ASVs_L = rep(NA,9))
large_df <- large_df %>% add_column(Size = "Large")
large_df <- large_df[order(large_df$ASVs_L),]

small_df <- data.frame(rownames(tax_table(phylo_small)))
colnames(small_df) <- "ASVs_S"
small_df <- small_df %>% add_column(Size = "Small")
small_df <- small_df[order(small_df$ASVs_S),]
#comb_df <- cbind(large_df,small_df)

comb_subset <- subset(otu_table(spat_size_phylo_rare_5000), rownames(otu_table(spat_size_phylo_rare_5000)) %in% comb_list)
spat_size_phylo_rare_5000_core <- merge_phyloseq(comb_subset, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
# Can't do a bar plot because of relative abundance

```
Beta diversity
```{r}
spat_size_phylo_rare_5000_core_wuni <- ordinate(spat_size_phylo_rare_5000_core, "PCoA", "unifrac", weighted=TRUE)

spat_size_phylo_rare_5000_core_uni <- ordinate(spat_size_phylo_rare_5000_core, "PCoA", "unifrac", weighted=FALSE)  
spat_size_phylo_rare_5000_core_wunidist <- UniFrac(spat_size_phylo_rare_5000_core, weighted=TRUE, normalized = TRUE, parallel=TRUE, fast=TRUE)
spat_size_phylo_rare_5000_core_wunidist_adonis_size <- adonis2(spat_size_phylo_rare_5000_core_wunidist ~ Size, data = spat_size_meta_reduced_5000)
spat_size_phylo_rare_5000_core_wunidist_adonis_size # p = 0.008
# Basic stats, trying to run adonis2 on a mixed effects model with the "perm" option.
perm <- how(nperm = 999)
setBlocks(perm) <- with(spat_size_meta_reduced_5000, Spat_family)
adonis2(spat_size_phylo_rare_5000_core_wunidist ~ Size, data = spat_size_meta_reduced_5000, permutations = perm) # p = .011def
plot_all <- plot_bar(spat_size_phylo_rare_5000_core_trans, x = "Replicate", fill = "Family")
#plot_final <- 
  plot_all +
  facet_grid(Size~Spat_family, scales = "free")+
  labs(title = "Spat microbiome by family and size")+
  theme(legend.position = "right", legend.spacing.y = unit(1,"mm"), legend.key.height = unit(1,"mm"), 
        axis.text.x = element_text(angle = 0))+
  guides(fill = guide_legend(ncol = 1))
```
Venn diagram - this is pretty useless
```{r}
x <- list(Small = small_df$ASVs_S, Large = large_df$ASVs_L)
ggVennDiagram(x, label = "count", label_geom = "label")#+
 # geom_label(mapping = aes(label = ""))

```
Now i'd like to subset a phyloseq object containing all taxa above.
```{r}
# list of taxa
x <- list(Small = small_df$ASVs_S, Large = large_df$ASVs_L)
comb_list <- unique(append(small_df$ASVs_S, large_df$ASVs_L)) # 42 taxa
comb_subset <- subset(otu_table(spat_size_phylo_rare_5000), rownames(otu_table(spat_size_phylo_rare_5000)) %in% comb_list)
spat_size_phylo_rare_5000_core <- merge_phyloseq(comb_subset, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
spat_size_phylo_rare_5000_core_trans <- transform_sample_counts(spat_size_phylo_rare_5000_core, function(x) x / sum(x))
# Beta div
spat_size_phylo_rare_5000_core_wuni <- ordinate(spat_size_phylo_rare_5000_core, "PCoA", "unifrac", weighted=TRUE)

spat_size_phylo_rare_5000_core_uni <- ordinate(spat_size_phylo_rare_5000_core, "PCoA", "unifrac", weighted=FALSE)  
spat_size_phylo_rare_5000_core_wunidist <- UniFrac(spat_size_phylo_rare_5000_core, weighted=TRUE, normalized = TRUE, parallel=TRUE, fast=TRUE)
spat_size_phylo_rare_5000_core_wunidist_adonis_size <- adonis2(spat_size_phylo_rare_5000_core_wunidist ~ Size, data = spat_size_meta_reduced_5000)
spat_size_phylo_rare_5000_core_wunidist_adonis_size # p = 0.012
# Basic stats, trying to run adonis2 on a mixed effects model with the "perm" option.
perm <- how(nperm = 999)
setBlocks(perm) <- with(spat_size_meta_reduced_5000, Spat_family)
adonis2(spat_size_phylo_rare_5000_core_wunidist ~ Size, data = spat_size_meta_reduced_5000, permutations = perm) # p = .015
```
Graphics
```{r}
spat_size_phylo_rare_5000_core_wuni_plot <- 
  plot_ordination(spat_size_phylo_rare_5000_core, spat_size_phylo_rare_5000_core_wuni, color = "Size")+
  geom_point(mapping = aes(size = rare_reads, fill = Size), pch = 21, color = "black", show.legend = TRUE)+
  scale_color_manual(values = c("forestgreen", "steelblue4"))+
  scale_fill_manual(values = c("forestgreen", "steelblue4"))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot",
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  stat_ellipse(type = "norm", linetype = 2, show.legend = FALSE)+
    geom_label(x=0.28, y=-0.3, label = "p(dispersion ~ Size (Spat family = random)) = 0.015", size = 3, color = "black", fill = "white")+
  labs(title = expression(paste("Beta diversity in ", bold("core microbiomes"), " of different-sized spat")),
       subtitle = "Weighted UniFrac",
       caption = "Rarefied data (read depth 5,000 - 15,000 per sample)",
       size = "Rarefied read depth")+
    guides(color = guide_legend(override.aes = list(size = 5)))
ggsave(plot = spat_size_phylo_rare_5000_core_wuni_plot, filename = "~/Documents/Oyster_Project/size_microbiome/for_diversity/spat_size_phylo_rare_5000_core_wuni_plot.png", scale = 1.25, width = 250, height = 100, units = "mm")
```
What if I do a beta diversity graph for spat_size_phylo_rare_5000, but remove family 94 and plot size as dry weight?
```{r}
spat_size_phylo_rare_5000 <- phyloseq(otu_table(asvs_size5000_with_asvs, taxa_are_rows = TRUE), 
                                   sample_data(spat_size_meta_reduced_5000),
                                   tax_table(as.matrix(taxa_size5000_with_asvs)),
                                   phy_tree(spat_5000_tree),
                                   refseq(size_5000_fasta_filtered))
spat_size_phylo_rare_5000_no94 <- subset_samples(spat_size_phylo_rare_5000, Spat_family != "94")
spat_size_phylo_rare_5000_no94_wuni <- ordinate(spat_size_phylo_rare_5000_no94, "PCoA", "unifrac", weighted=TRUE)

plot_ordination(spat_size_phylo_rare_5000_no94, spat_size_phylo_rare_5000_no94_wuni, color = "Size")+
  geom_point(mapping = aes(size = Avg_dry_weight, fill = Size), pch = 21, color = "black", show.legend = TRUE)+
  scale_color_manual(values = c("forestgreen", "steelblue4"))+
  scale_fill_manual(values = c("forestgreen", "steelblue4"))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot",
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  stat_ellipse(type = "norm", linetype = 2, show.legend = FALSE)+
    geom_label(x=0.28, y=-0.3, label = "p(dispersion ~ Size (Spat family = random)) = 0.015", size = 3, color = "black", fill = "white")+
  labs(title = expression(paste("Beta diversity in ", bold("core microbiomes"), " of different-sized spat")),
       subtitle = "Weighted UniFrac",
       caption = "Rarefied data (read depth 5,000 - 15,000 per sample)",
       size = "Average dry weight (g)")+
    guides(color = guide_legend(override.aes = list(size = 5)))
spat_size_meta_reduced_5000_no94 <- subset(spat_size_meta_reduced_5000, Spat_family != "94")
spat_size_phylo_rare_5000_no94_wunidist <- UniFrac(spat_size_phylo_rare_5000_no94, weighted=TRUE, normalized = TRUE, parallel=TRUE, fast=TRUE)
spat_size_phylo_rare_5000_no94_wunidist_adonis_size <- adonis2(spat_size_phylo_rare_5000_no94_wunidist ~ Size, data = spat_size_meta_reduced_5000_no94)
spat_size_phylo_rare_5000_no94_wunidist_adonis_size # p = 0.13
# Basic stats, trying to run adonis2 on a mixed effects model with the "perm" option.
perm <- how(nperm = 999)
setBlocks(perm) <- with(spat_size_meta_reduced_5000_no94, Spat_family)
adonis2(spat_size_phylo_rare_5000_no94_wunidist ~ Size, data = spat_size_meta_reduced_5000_no94, permutations = perm) # p = .129
```


```{r}
plot_bar(phylo_small, x = "Replicate", fill = "Family")
```
ANCOMBC - core
```{r}
ancom_out <- ancombc(phyloseq = spat_size_phylo_rare_5000_core, formula = "Size", 
              p_adj_method = "BH", zero_cut = 0.90,
              group = "Size", struc_zero = FALSE, neg_lb = FALSE, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = FALSE)
ancom_res_df <- data.frame(
  Species = row.names(ancom_out$res$beta),
  beta = unlist(ancom_out$res$beta),
  se = unlist(ancom_out$res$se),
  W = unlist(ancom_out$res$W),
  p_val = unlist(ancom_out$res$p_val),
  q_val = unlist(ancom_out$res$q_val),
  diff_abn = unlist(ancom_out$res$diff_abn))
fdr_ancom <- ancom_res_df %>%
  dplyr::filter(q_val < 0.05)
dim(fdr_ancom) # 26 x 7
#####
# interpretation test; asv11 is more abundant in large, and has a negative beta value
asv11_subset <- subset_taxa(spat_size_phylo_rare_5000_trans, Genus == "Psychromonas")
plot_bar(asv11_subset, fill = "Genus", x = "Replicate", facet_grid = Size~Spat_family)
# interpretation test; asv10 is more abundant in large, and has a negative beta value
asv10_subset <- subset_taxa(spat_size_phylo_rare_5000_trans, Family == "Rhodospirillaceae")
plot_bar(asv10_subset, fill = "Family", x = "Replicate", facet_grid = Size~Spat_family)
# interpretation test; asv60 is more abundant in small, and has a positive beta value
asv60_subset <- subset_taxa(spat_size_phylo_rare_5000_trans, Genus == "Sulfitobacter")
plot_bar(asv60_subset, fill = "Genus", x = "Replicate", facet_grid = Size~Spat_family)
# interpretation test; asv35 is more abundant in small, and has a positive beta value
asv35_subset <- subset_taxa(spat_size_phylo_rare_5000_trans, Order == "MBAE14")
plot_bar(asv35_subset, fill = "Order", x = "Replicate", facet_grid = Size~Spat_family)

# if beta < 0, it's more abundant in the Large.
# if beta > 0, it's more abundant in the Small.
#####
fdr_ancom$Species %in% comb_list # all true

write.csv(fdr_ancom, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/ancombc.csv")
ancombc_subset <- subset(otu_table(spat_size_phylo_rare_5000_trans), rownames(otu_table(spat_size_phylo_rare_5000_trans)) %in% fdr_ancom$Species)
spat_size_phylo_rare_5000_ancombc <- merge_phyloseq(ancombc_subset, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
# Row sums by size for rank abundance plot
spat_size_phylo_rare_5000_ancombc_L <- subset_samples(spat_size_phylo_rare_5000_ancombc, Size == "Large")
spat_size_phylo_rare_5000_ancombc_S <- subset_samples(spat_size_phylo_rare_5000_ancombc, Size == "Small")
temp_df <- cbind.data.frame(rownames(otu_table(spat_size_phylo_rare_5000_ancombc_L)), (rowMeans(otu_table(spat_size_phylo_rare_5000_ancombc_L))*100), (rowMeans(otu_table(spat_size_phylo_rare_5000_ancombc_S))*100), (rowMeans(otu_table(spat_size_phylo_rare_5000_ancombc))*100))
colnames(temp_df) <- c("ASV", "Avg_Rel_abund_L", "Avg_rel_abund_S", "Avg_rel_abund_comb")
write.csv(temp_df, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/rankabund_both.csv")
#plot_bar(spat_size_phylo_rare_5000_ancombc, fill = "Order", x = "Replicate", facet_grid = Size~Spat_family)
# Rank abundance plot
ancombc_size_meta <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/ancombc_taxa.csv", sep = ",", header = T)
a
# reorder by abundance in 
# Relative abundance: overall (no fold change)

rankabundplot <- ggplot(ancombc_size_meta, aes(x = Species, y = Avg_rel_abund, color = Order))+
  geom_point(aes(fill = Order), pch = 21, color = "black", size = 4)+
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major.x = element_line(color = "gray90"),
        panel.grid.minor.x = element_line(color = "gray90"),
        panel.grid.minor.y = element_line(color = "gray90"),
        panel.grid.major.y = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot",
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  labs(y = "Average relative abundance of sample by size class",title = "Differentially abundant taxa in spat microbiome", x = NULL)+
  viridis::scale_fill_viridis(discrete = TRUE, option = "B", begin = 0.2, na.value = "black")+
  facet_grid(Size~.)
ggsave(plot = rankabundplot, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/rankabundplot.png", scale = 1, width = 200, height = 200, units = "mm")
```
Ancom, non-core
```{r}
ancom_out_nocore <- ancombc(phyloseq = spat_size_phylo_rare_5000, formula = "Size", 
              p_adj_method = "BH", zero_cut = 0.90,
              group = "Size", struc_zero = FALSE, neg_lb = FALSE, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = FALSE)
ancom_res_df_nocore <- data.frame(
  Species = row.names(ancom_out_nocore$res$beta),
  beta = unlist(ancom_out_nocore$res$beta),
  se = unlist(ancom_out_nocore$res$se),
  W = unlist(ancom_out_nocore$res$W),
  p_val = unlist(ancom_out_nocore$res$p_val),
  q_val = unlist(ancom_out_nocore$res$q_val),
  diff_abn = unlist(ancom_out_nocore$res$diff_abn))
fdr_ancom_nocore <- ancom_res_df_nocore %>%
  dplyr::filter(q_val < 0.05)
dim(fdr_ancom_nocore) # 33 x 7
fdr_ancom_nocore$Species
```
TEST 5/20
```{r}
ancom_out_nocore_redo <- ancombc(phyloseq = spat_size_phylo_rare_5000, formula = "Size", 
              p_adj_method = "BH", zero_cut = 0.90,
              group = "Size", struc_zero = FALSE, neg_lb = FALSE, tol = 1e-5, 
              max_iter = 100, conserve = FALSE, alpha = 0.05, global = FALSE)
ancom_res_df_nocore_redo <- data.frame(
  Species = row.names(ancom_out_nocore_redo$res$beta),
  beta = unlist(ancom_out_nocore_redo$res$beta),
  se = unlist(ancom_out_nocore_redo$res$se),
  W = unlist(ancom_out_nocore_redo$res$W),
  p_val = unlist(ancom_out_nocore_redo$res$p_val),
  q_val = unlist(ancom_out_nocore_redo$res$q_val),
  diff_abn = unlist(ancom_out_nocore_redo$res$diff_abn))
fdr_ancom_nocore_redo <- ancom_res_df_nocore_redo %>%
  dplyr::filter(q_val < 0.05)
dim(fdr_ancom_nocore_redo) # 41 x 7
fdr_ancom_nocore_redo$Species
ancomtax_redo <- data.frame(tax_table(as.matrix(taxa_size5000_with_asvs)))
ancomtax_redo$Species <- rownames(ancomtax_redo)
fdr_ancom_nocore_taxa_redo <- merge(fdr_ancom_nocore_redo, ancomtax_redo, by = "Species")
write.csv(fdr_ancom_nocore_taxa_redo, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/ancombc_nocore_taxa_redo.csv")
```


Alright - I think I need to redo the ancom plot but with all taxa, not just the core.
```{r}
ancomtax <- data.frame(tax_table(as.matrix(taxa_size5000_with_asvs)))
ancomtax$Species <- rownames(ancomtax)
fdr_ancom_nocore_taxa <- merge(fdr_ancom_nocore, ancomtax, by = "Species")
write.csv(fdr_ancom_nocore_taxa, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/ancombc_nocore_taxa.csv")
# add an identifier column; need to edit by hand
fdr_ancom_nocore_taxa_edit <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/ancombc_nocore_taxa_edits.csv", sep = ",", header = T)
# Need to somehow add relative abundance of each in large and small
ancombc_nocore_subset <- subset(otu_table(spat_size_phylo_rare_5000_trans), rownames(otu_table(spat_size_phylo_rare_5000_trans)) %in% fdr_ancom_nocore_taxa$Species)
spat_size_phylo_rare_5000_ancombc_nocore <- merge_phyloseq(ancombc_nocore_subset, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
```


Now a log-fold graph
```{r}
taxtable <- data.frame(tax_table(spat_size_phylo_rare_5000_ancombc))
taxtable$ASV <- rownames(taxtable)

spat_size_phylo_rare_5000_ancombc <- merge_phyloseq(ancombc_subset, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxtable)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
```

Plotting

```{r}
c25 <- c(
  "dodgerblue2", "#E31A1C", # red
 # "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
 # "black", 
  "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3",
  "darkorange4", "brown"
)
effectsizeplot <- ggplot(ancombc_size_meta, aes(x = reorder(Species.1, -W), y = -W))+
  geom_rect(xmin = 0, xmax = Inf, ymin = -6, ymax = 0, fill = "dodgerblue", alpha = 0.004)+
  geom_rect(xmin = 0, xmax = Inf, ymin = 0, ymax = 8, fill = "red", alpha = 0.005)+
  geom_point(aes(fill = Order), pch = 21, color = "black", size = 5)+
  geom_hline(yintercept = 0)+
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major.y = element_line(color = "gray90"),
        panel.grid.minor.y = element_line(color = "gray90"),
        panel.grid.major.x = element_line(color = "gray90"),
        panel.grid.minor.x = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
 #       plot.title = element_text(hjust = 0.5),
#        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot",
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  labs(x = "ASV", y = "-W (standardized effect size)",title = "ANCOM-BC: Differentially abundant taxa in spat microbiome",
       caption = "Positive values indicate higher relative abundance in large")+
#  viridis::scale_color_viridis(option = "D", discrete = TRUE, na.value = "#000000")+
#  scale_color_manual(values = c("Alteromonadales" = "#1E90FF", # dodgerblue
         #                       "Gammaproteobacteria_Incertae_Sedis" = "#32CD32", #lime green
         #                       "Flavobacteriales" = "#FFD700", # gold
          #                      "Rhizobiales" = "#00008B")) # dark blue
  scale_color_manual(values = c25, na.value = "#000000")+
  scale_fill_manual(values = c25, na.value = "#000000")
 # viridis::scale_fill_viridis(option = "D", discrete = TRUE, na.value = "#000000")

ggsave(plot = effectsizeplot, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/effectsizeplot.png", scale = 1.5, width = 200, height = 125, units = "mm")


tree_with_meta <- ggtree(spat_size_phylo_rare_5000_ancombc, branch.length = "none", ) %<+% spat_size_meta_reduced_5000
annotated_tree <- tree_with_meta + geom_label(aes(label = ASV), size = 2, hjust = -0.75, nudge_y = 0.25) + #geom_tiplab(hjust = 1, vjust = -.9, size = 3, aes(label = Taxa))+
  geom_tippoint(aes(color = Order), size = 2)+geom_tippoint(shape = 1, size = 2, color = "black", show.legend = TRUE)+
  viridis::scale_color_viridis(discrete = TRUE, option = "C", end = 0.9, na.value = "black")

ggsave(plot = annotated_tree, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/ancomtree.png", scale = 1.5, width = 150, height = 100, units = "mm")


```

Log-fold plot for non-core data (correct data)
```{r}
fdr_ancom_nocore_taxa_edit_with_pct <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/fdr_ancom_nocore_taxa_edit_with_pct.csv", header = T)
fdr_ancom_nocore_taxa_edit
colorBlindBlack8  <- c("#E69F00", "#56B4E9", "#009E73", 
                       "#F0E442", "#0072B2", "#D55E00", "#CC79A7","white")
#fdr_ancom_nocore_taxa_edit_with_pct$rounded <- ceiling(fdr_ancom_nocore_taxa_edit_with_pct$avg_pct)

#fdr_ancom_nocore_taxa_edit_with_pct$rounded <- ceiling(fdr_ancom_nocore_taxa_edit_with_pct$avg_pct<1)
#contrast_8 <- c("#67000D","#CB181D", "#FB6A4A", "#FCBBA1","#FFF5F0", "#FDE725FF", "#08306B", "#2171B5")
#mkePalette_8_redo <- c("#440154FF", "#553781FF", "#39568CFF", "#287D8EFF", "20A387FF", "#FDE725FF", "#B8DE29FF", "#55C667FF")
#whatever_pal <- c("#0072B2", "#E69F00", "white", "black", "#CC79A7", "#FDE725FF", "#D55E00", "#56B4E9")
#rainbow <- RColorBrewer::brewer.pal(8, name = "Set1")
rainbow <- c("#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00", "#FDE725FF", "gray80", "black")
# rename asv09
fdr_ancom_nocore_taxa_edit_with_pct[9,16] <- "asv09_Vibrionaceae"
effectsizeplot_nocore <-
  ggplot(fdr_ancom_nocore_taxa_edit_with_pct, aes(x = reorder(ID, -W), y = -W))+
  geom_rect(xmin = 0, xmax = Inf, ymin = -Inf, ymax = 0, fill = "steelblue3", alpha = 0.02)+ # blue box
  geom_rect(xmin = 0, xmax = Inf, ymin = 0, ymax = Inf, fill = "#FF6600", alpha = 0.02)+ # orange box
  geom_point(aes(fill = Class, size = avg_pct), pch = 21, color = "black")+ # dots
  geom_hline(yintercept = 0)+ #dividing line
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major.y = element_line(color = "gray90"),
        panel.grid.minor.y = element_line(color = "gray90"),
        panel.grid.major.x = element_line(color = "gray90"),
    #    panel.grid.minor.x = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.caption.position = "plot", 
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 12),
        legend.box.spacing = unit(1, "mm"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
 # scale_x_continuous(breaks = seq(1,32,2))+
  labs(x = "ASV", y = "-W (test statistic) ", title = "Differentially abundant ASVs by spat size", size = "Avg. pct. of reads in size", fill = "Class")+
  guides(fill = guide_legend(override.aes = list(size = 5), order = 2), size = guide_legend(order = 1))+
  scale_color_manual(values = rainbow, na.value = "gray45")+
  scale_fill_manual(values = rainbow, na.value = "gray45")+
  scale_size_continuous(limits = c(0,26), breaks = c(0.1,1, 5,10,25), labels = c("0.1","1","5","10","25"), range = c(2,10))

ggsave(plot = effectsizeplot_nocore, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/effectsizeplot_nocore.jpg", scale = 1.5, width = 180, height = 100, units = "mm")
```


Corncob:
Wald vs LRT:
It's important to note that although the likelihood ratio test and the Wald test are used by researchers to accomplish the same empirical goal(s), they are testing different hypotheses. The likelihood ratio test evaluates whether the data were likely to have come from a more complex model, vs. a more simple model. Put another way, does the addition of a particular effect allow the model to account for more information. The Wald test, conversely, evaluates whether it is likely that the estimated effect could be zero. It's a nuanced difference, to be sure, but an important conceptual difference nonetheless.
```{r}
# with non-core object
#data(spat_size_phylo_rare_5000)
corncob <- differentialTest(formula = ~ Size,
                            phi.formula = ~ Size,
                            formula_null = ~ 1,
                            phi.formula_null = ~ Size,
                            test = "Wald", boot = FALSE,
                            data = spat_size_phylo_rare_5000,
                            fdr_cutoff = 0.05)
both_list <- intersect(fdr_ancom_nocore$Species, corncob$significant_taxa)
```
Choose taxa significant in both tests:
```{r}
both_subset <- subset(otu_table(spat_size_phylo_rare_5000), rownames(otu_table(spat_size_phylo_rare_5000)) %in% both_list)
both_subset_trans <- subset(otu_table(spat_size_phylo_rare_5000_trans), rownames(otu_table(spat_size_phylo_rare_5000_trans)) %in% both_list)
spat_size_phylo_rare_5000_da <- merge_phyloseq(both_subset, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
spat_size_phylo_rare_5000_da_trans <- merge_phyloseq(both_subset_trans, sample_data(spat_size_meta_reduced_5000),
                                    tax_table(as.matrix(taxa_size5000_with_asvs)),
                                    phy_tree(spat_5000_tree),
                                    refseq(size_5000_fasta_filtered))
```


Upset plot
```{r}
spat_size_phylo_rare_5000_trans_100 <- otu_table(spat_size_phylo_rare_5000_trans)*100
phylo_for_upset <- merge_phyloseq(otu_table(spat_size_phylo_rare_5000_trans_100, taxa_are_rows = TRUE),
                                  sample_data(spat_size_meta_reduced_5000),
                                  tax_table(as.matrix(taxa_size5000_with_asvs)), 
                                  phy_tree(spat_5000_tree), refseq(size_5000_fasta_filtered))
saveRDS(phylo_for_upset, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/phylo_for_upset.rds")

samp <- sample_data(diversity_size)
# all taxa, by size
ps2.venn <- merge_samples(diversity_size, 'Size', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
samp$Size <- as.factor(samp$Size)
ncolors <- length(levels(samp$Size))
primary_color_list <- c("forestgreen", "steelblue4")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 2, sets = rev(upset_order), mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per size", keep.order = T, order.by = "freq", sets.bar.color = rev(primary_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.9)
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/diversity_upset.png", width = 2000, height = 1200, res = 200)
shared_ASV_plot
dev.off()
# all taxa, by size+family
samp <- sample_data(diversity_size)

ps2.venn <- merge_samples(diversity_size, 'ID', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
samp$ID <- as.factor(samp$ID)
ncolors <- length(levels(samp$ID))
# colors for 10 families()
primary_color_list <- RColorBrewer::brewer.pal(length(levels(samp$ID))/2, name = "Paired")
paired_color_list <- c("chartreuse4", "chartreuse2", # light greens
  "cadetblue4", "cadetblue3", # light/tealy blues
  "darkorange3", "darkorange", # orange
  "darkorchid4", "darkorchid1", # purple
  "darkred", "firebrick2", # red
  "gold3", "gold1", # yellow
  "darkblue", "blue", # blue
  "deeppink3", "deeppink1", #pink
  "tan4", "tan3",
  "palegreen3", "palegreen"
)
primary_color_list <- viridis::viridis(ncolors)
#primary_color_list <- c("forestgreen", "steelblue4")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 10, sets = rev(upset_order), mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per size and family", keep.order = T, order.by = "freq", sets.bar.color = rev(paired_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.5)
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/diversity_upset.png", width = 2000, height = 1400, res = 200)
shared_ASV_plot
dev.off()

# with trimmed 0.01 object, by size
samp <- sample_data(spat_size_phylo_rare_5000)
ps2.venn <- merge_samples(spat_size_phylo_rare_5000, 'Size', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
samp$Size <- as.factor(samp$Size)
ncolors <- length(levels(samp$Size))
primary_color_list <- c("forestgreen", "steelblue4")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 10, sets = rev(upset_order), mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per size", keep.order = T, order.by = "freq", sets.bar.color = rev(primary_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.9)
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/trimmed_upset.png", width = 2000, height = 1200, res = 200)
shared_ASV_plot
dev.off()

# with core, by size+family
samp <- sample_data(spat_size_phylo_rare_5000_core)
ps2.venn <- merge_samples(spat_size_phylo_rare_5000_core, 'ID', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
samp$ID <- as.factor(samp$ID)
ncolors <- length(levels(samp$ID))
primary_color_list <- c("forestgreen", "steelblue4")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 20, sets = rev(upset_order), #empty.intersections = "on",
                         mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per size and family", keep.order = T, order.by = "freq", sets.bar.color = rev(paired_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.9)
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/core_upset.png", width = 2000, height = 1400, res = 200)
shared_ASV_plot
dev.off()

# core, size only
samp <- sample_data(spat_size_phylo_rare_5000_core)
ps2.venn <- merge_samples(spat_size_phylo_rare_5000_core, 'Size', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
#samp$Spat_family <- as.factor(samp$Spat_family)
#ncolors <- length(levels(samp$Spat_family))
primary_color_list <- c("forestgreen", "steelblue4")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 2, sets = rev(upset_order), #empty.intersections = "on",
                         mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per size", keep.order = T, order.by = "freq", sets.bar.color = rev(primary_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.9)
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/core_upset_size.png", width = 2000, height = 1400, res = 200)
shared_ASV_plot
dev.off()
# by family only
samp <- sample_data(diversity_size)
ps2.venn <- merge_samples(diversity_size, 'Spat_family', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
samp$Spat_family <- as.factor(samp$Spat_family)
ncolors <- length(levels(samp$Spat_family))
primary_color_list <- viridis::viridis(ncolors)
primary_color_list <- c((RColorBrewer::brewer.pal(n = 9, name = "Set1")),"deepskyblue")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 10, empty.intersections = "on", sets = rev(upset_order), mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per spat family", keep.order = T, order.by = "freq", sets.bar.color = rev(primary_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.9)
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/family_upset.png", width = 2000, height = 1400, res = 200)
shared_ASV_plot
dev.off()
# with trimmed 0.01 object, family
samp <- sample_data(spat_size_phylo_rare_5000)
ps2.venn <- merge_samples(spat_size_phylo_rare_5000, 'Spat_family', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
samp$Spat_family <- as.factor(samp$Spat_family)
ncolors <- length(levels(samp$Spat_family))
primary_color_list <- viridis::viridis(ncolors)
#primary_color_list <- c("forestgreen", "steelblue4")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 10, sets = rev(upset_order), mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per family", keep.order = T, order.by = "freq", sets.bar.color = rev(primary_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.9, empty.intersections = "on")
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/trimmed_upset_family.png", width = 2000, height = 1200, res = 200)
shared_ASV_plot
dev.off()
# core microbiome, family
samp <- sample_data(spat_size_phylo_rare_5000_core)
ps2.venn <- merge_samples(spat_size_phylo_rare_5000_core, 'Spat_family', fun = sum)
venn_obj <- as.data.frame(t(otu_table(ps2.venn)))
venn_obj_binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0), USE.NAMES = T)
rownames(venn_obj_binary) <- rownames(venn_obj)
venn_obj_binary <- as.data.frame(venn_obj_binary)
upset_order <- colnames(venn_obj_binary)
samp$Spat_family <- as.factor(samp$Spat_family)
ncolors <- length(levels(samp$Spat_family))
#primary_color_list <- c("forestgreen", "steelblue4")
shared_ASV_plot <- upset(venn_obj_binary, nsets = 10, sets = rev(upset_order), empty.intersections = "on",
                         mainbar.y.label = "Number of shared ASVs", sets.x.label = "ASVs per family", keep.order = T, order.by = "freq", sets.bar.color = rev(primary_color_list), matrix.color = c("black"), shade.color = "gray70", main.bar.color = "black", matrix.dot.alpha = 0.9)
png(file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/core_upset_family.png", width = 2000, height = 1400, res = 200)
shared_ASV_plot
dev.off()
```


Going to make a core microbiome for small and large combined, to see how it looks and if it looks different from small and large.
```{r}
# Create Function to work on phyloseq objects
asv_count <- function(ps.obj 
                      ){
              cutoff.lvls = seq(from = 0, to = 1, by = 0.01)
              tmp.sums <- lapply(cutoff.lvls, function(cutoff){
                              tmp.isPresent <- filter_taxa(ps.obj, function(x) sum(x > 0) >= (cutoff*nsamples(ps.obj)), prune = FALSE)
                              tmp.sumPresent <- sum(tmp.isPresent)
                            })
              tmp.df <- data.frame(Percent_samples = cutoff.lvls,
                                   Number_of_ASVs = unlist(tmp.sums, recursive = F))#,
                                  # Size = sub.name)
  return(tmp.df)
}
# Run function
df_combo <- asv_count(diversity_size)
View(df_combo)
write.csv(df_combo, file =  "~/Documents/Oyster_Project/size_microbiome/for_diversity/combo_intervals.csv")

df_combo$Pct_samples_2 <- df_combo$Percent_samples*100
#core_combo_interval_diversity_plot <- 
ggplot(df_combo, aes(x = Pct_samples_2, y = Number_of_ASVs))+
  geom_point(size = 1)+
  geom_smooth(se = FALSE, span = 0.1, size = 0.75, fullrange = TRUE, color = "black")+
  labs(x = "Percent of samples",
       y = "Number of ASVs present in samples",
       title = "Defining a core microbiome", 
       subtitle = "All samples (not separated by size)",
       caption = "Using all taxa (not cut at 0.01% abundance)\nMethod: Modified from Ainsworth et al., 2015")+
  scale_x_continuous(limits = c(0,100), expand = c(.01,0))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right",
        legend.key = element_rect(fill = NA))
ggsave(plot = core_interval_diversity_plot, filename = "core_interval_diversity_plot.png", path = "~/Documents/Oyster_Project/size_microbiome/for_diversity", scale = 2, width = 100, height = 75, units = "mm")
# Changes at 0.61 (19 ASVs)
diversity_size_trans <- transform_sample_counts(diversity_size, function(x) x / sum(x))
taxa_combo <- filter_taxa(diversity_size_trans, function(x) sum(x > 0) >= (0.61*nsamples(diversity_size_trans)), prune=FALSE)
phylo_combo <- subset_taxa(taxa_combo, physeq = diversity_size_trans)

phylo_combo_plot <- plot_bar(phylo_combo, x = "Fam_rep", fill = "Order")+facet_wrap(.~Spat_family, nrow = 2)+
  labs(title = "Oyster core microbiome: taxa found in 61% of samples", subtitle = "All samples (not separated by size)", y = "Relative abundance in sample", x = "Replicate")+
    ylim(0,1)+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"),
        legend.position = "right",
        legend.key.height = unit(1,"mm"),
        legend.text = element_text(size = 10))+
  guides(fill = guide_legend(ncol = 1))+
  viridis::scale_fill_viridis(option = "D", discrete = TRUE)
ggsave(plot = phylo_combo_plot, filename = "phylo_combo_plot.png", path = "~/Documents/Oyster_Project/size_microbiome/for_diversity", scale = 2, width = 200, height = 100, units = "mm")
```

Beta diversity stats on the combo core
```{r}
phylo_combo_wuni <- ordinate(phylo_combo, "PCoA", "unifrac", weighted=TRUE)

phylo_combo_wunidist <- UniFrac(phylo_combo, weighted=TRUE, normalized = TRUE, parallel=TRUE, fast=TRUE)
phylo_combo_wunidist_adonis_size <- adonis2(phylo_combo_wunidist ~ Size, data = spat_size_meta_reduced_5000) # p = 0.02
spat_size_phylo_rare_5000_core_wunidist_adonis_size # p = 0.012
# Basic stats, trying to run adonis2 on a mixed effects model with the "perm" option.
perm <- how(nperm = 999)
setBlocks(perm) <- with(spat_size_meta_reduced_5000, Spat_family)
adonis2(phylo_combo_wunidist ~ Size, data = spat_size_meta_reduced_5000, permutations = perm) # p = .018
```

basic stats for the manuscript about community composition.
```{r}
sums <- data.frame(rowSums(data.frame(otu_table(diversity_size_trans))))
sums["asv0001",] # 26.8%
sums2 <- data.frame(rowSums(data.frame(otu_table(spat_size_phylo_rare_5000_trans))))
sums2["asv01",] # 31.33%
sums2["asv02",] # 8.96
sums2["asv06",] # 3.5%
sums2["asv09",] # 4%
sums2["asv10",] # 3.5
sums2["asv11",] # 3.15%
sums2["asv13",] # 2.5
sums2["asv18",] # 1.39%
sums2["asv20",] # 1.2% 
sums2["asv22",] # 1.19%
sums2["asv35",] # MBAE14; 0.73%
sums2["asv89",] # 0.3%
sums2["asv81",]
fdr_ancom_nocore_taxa_edit$Species
sums2[fdr_ancom_nocore_taxa_edit$Species,]
subset_large_trans <- subset_samples(spat_size_phylo_rare_5000_trans, Size == "Large")
means_large <- data.frame(rowMeans(data.frame(otu_table(subset_large_trans))))
means_large[fdr_ancom_nocore_taxa_edit$Species,]
fdr_ancom_nocore_taxa_edit$large_pct <- 100*means_large[fdr_ancom_nocore_taxa_edit$Species,]
subset_small_trans <- subset_samples(spat_size_phylo_rare_5000_trans, Size == "Small")
means_small <- data.frame(rowMeans(data.frame(otu_table(subset_small_trans))))
means_small[fdr_ancom_nocore_taxa_edit$Species,]
fdr_ancom_nocore_taxa_edit$small_pct <- 100*means_small[fdr_ancom_nocore_taxa_edit$Species,]

write.csv(fdr_ancom_nocore_taxa_edit, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/fdr_ancom_nocore_taxa_edit_with_pct.csv")
subset(fdr_ancom_nocore_taxa_edit, (large_pct >= 0.1 | small_pct >= 0.1))
```



Going to try normalized alpha diversity, using Michael's code.
Setup:
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}
for (pkg in c(#"devtools", "dada2", "phyloseq", 
              "ALDEx2")) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    BiocManager::install(pkg)
  }
}
devtools::install_github("ggloor/CoDaSeq/CoDaSeq")
devtools::install_github("kstagaman/phyloseqCompanion")

## General Purpose
library(knitr)  # For knitting documents to HTML or PDF formats
library(tinytex)  # LaTex stuff for Rmarkdown
library(data.table)  # Handle data tables
library(tidyverse)  # Making your code look pretty and tidy
library(reshape2)  # for reshaping data using melt()
library(rcompanion)  # various functions, transformTukey()
## Figures/Tables
library(ggplot2)  # For plotting pretty graphs
library(CoDaSeq)  # For CLR transformations, PCOA plots
library(gridExtra)  # use marrangeGrob, for combining plots
library(ggbeeswarm)  # Pretty dots on box plots
## Microbiome Analysis
library(phyloseq)  # For microbiome analysis and plotting functions
library(phyloseqCompanion)  # helper functions for manipulating phyloseq objects
library(nortest)  # Allows us to run ad.test()
library(picante)  # Allows us to use pd() to calc phylogenetic diversity
```

```{r func-calc-alpha-scores}
# Calculate Alpha Scores --------------------------------------------------
#   Description: Generates alpha diversity scores from a list of alpha methods
#   Input: phyloseq object, list of alpha div. methods, 
#   Output: dataframe of alpha-diversity scores
alpha_base <- function(
  physeq,  # Phyloseq object
  methods,  # List of alpha methods (e.g., c(Shannon, Simpson, Observed) )
  smpl.col.name = "Sample",  # Default is "Sample" but you can change it to whatever when you call the function 
  phylo.div = F  # Only set to true if you have phylogenetic information attached to your phyloseq object
  ){
  
  # Calculates alpha scores
  tmp.dt <- phyloseq::estimate_richness(
    physeq = physeq,  # Physeq object
    measures = methods[1:(length(methods)-1)]
  ) %>% as.data.table(keep.rownames = smpl.col.name) %>% setkeyv(smpl.col.name)  # Sets sample column name to whatever you've called it (e.g., "Sample" or "Sample.ID")
  tmp.dt[, se.chao1 := NULL] # No idea what this does, but it's from Keatons code and I think it's important if you calculate se.chao1 scores
  
  # If you have phylogenetic information in your phyloseq object you'll want to set phylo.div to true
  if(isTRUE(phylo.div)){
    
    # Calculate the sum of the total phylogenetic branch length for one or multiple samples. See ?picante::pd() for more info
    #   - Returns a dataframe of the PD and species richness (SR) values for all samples
    phy.dt <- picante::pd(samp = otu.matrix(physeq), tree = phyloseq::phy_tree(physeq)) %>%
      as.data.table(keep.rownames = smpl.col.name) %>% setkeyv(smpl.col.name)  # set col name
    
    # Names the columns by alpha method
    # names(phy.dt)[(length(phy.dt)-1):length(phy.dt)] <- methods[length(methods)]  #methods[(length(methods)-1):length(methods)]
    names(phy.dt)[2:3] <- methods[(length(methods)-1):length(methods)]
    tmp.dt[phy.dt] # adds the columns to the other dataframe with the previously calculated alpha scores
    
    # return to sender
    return (tmp.dt[phy.dt])
  }
  
  # Returns alpha scores datatable
  return (tmp.dt)
}
```


## Running Function

Set alpha methods: 
```{r set-alpha-methods}
## phyloseq objects with phylogenetic data
methods.alpha <- c("Shannon", "Simpson", "Phylogenetic", "Richness") %>%
                  purrr::set_names()  # Set's names list elements in "alpha.methods"
```

Calculate alpha scores: 

```{r calc-alpha-scores}
# Calculate raw alpha scores
#   Note: if you don't include se.chao1, it will throw a warning. No biggie
dt.alphaScores.all <- alpha_base(physeq = spat_size_phylo_rare_5000,  # Phyloseq object
                                 methods = methods.alpha,  # List of alpha methods
                                 smpl.col.name = "Sample",  # Default is "Sample" but you can change it to name of sample column
                                 phylo.div = T  # Set to true if your physeq obj has phylogenetic information
                                 )
# Check that scores were calculated
head(dt.alphaScores.all)
```

# Normalize Alpha Scores

## Save Function

```{r function-norm-alpha-scores}
# Normalize Alpha Scores --------------------------------------------------
#   Description: normalizes alpha diversity scores based on their distributions
#   Input: dataframe of alpha diversity scores, metadata table
#   Output: normalized datatable of alpha scores (0 to 1)
norm_alpha_score <- function(
  alpha.base, 
  sample.df, 
  methods,
  smpl.col.name = "Sample"
  ){
  
  # Makes a copy of the dataframe you input and adds a column for your sample IDs
  model.data.base <- copy(alpha.base[alpha.base[[smpl.col.name]] %in% 
                                       row.names(sample.df)])
  
  # Loops through the different alpha methods
  for (alpha in methods) {
    
    # ad.test(): Performs the Anderson-Darling test for the composite hypothesis of normality
    #   - Basically checking to see if the alpha score distribution that was calculated previously follows a normal distribution or not
    #   - Check this out for more info: ?ad.test()
    if (nortest::ad.test(model.data.base[[alpha]])$p.value <= 0.05) {
      
      # If the alpha scores do not follow a normal distribution, then you transform it using Tukey's power of ladders
      #   - This will transform your data as closely to a normal distribution
      
      # Sub-function to transform data that isn't normally distributed using Tukey's  power of ladders
      #   - Check this out for more info: ?transformTukey()
      trans <- rcompanion::transformTukey(model.data.base[[alpha]], plotit = F, quiet = F, statistic = 2)
      trans <- (trans-min(trans))/(max(trans)-min(trans))   # Fixes normalization 0 to 1
      
      # Runs ad.test again to see if data returns higher than 0.05 p-value, if true then it transforms your data along tukey's power of ladders
      if (nortest::ad.test(trans)$p.value > 0.05) {
        model.data.base[[alpha]] <- trans  # Transform data with transformTukey() above
        print(paste0("Finished: ", alpha))  # Letting you know what it's working on
        
        # If your data is now normally distributed it will return < 0.05 p.val, and then it uses max/min values to distribute the scores along a 0 and 1 scale.
      } else {
        model.data.base[[alpha]] <- (model.data.base[[alpha]] - min(model.data.base[[alpha]] ))/(max(model.data.base[[alpha]] )-min(model.data.base[[alpha]] ))  # Fixes normalization 0 to 1
        print(paste0("Finished: ", alpha))  # Letting you know what it's working on
      } 
      
    # If your data is already normally distributed, then it uses max/min values to distribute the scores along a 0 and 1 scale.
    } else {
      model.data.base[[alpha]] <- (model.data.base[[alpha]] - min(model.data.base[[alpha]] ))/(max(model.data.base[[alpha]] )-min(model.data.base[[alpha]] ))  # Fixes normalization 0 to 1
      print(paste0("Finished: ", alpha))  # Letting you know what it's working on
    }
  }
  
  # Sends your data back normalized from 0 to 1
  return(model.data.base)
}
```


## Run Function

Normalizing scores from 0 to 1 for easier comparison across metrics.

Under the hood, we use the functions `descdist` and `fitdist` (`fitdistrplus` package) to determine that the best distribution for the alpha-diversity metric scores were almost always the beta distribution. This distribution is not directly supported by the `glm` function (used in later data analysis) but is approximated by the `quasibinomial` family. These distributions only take values from 0 to 1, so we divide all alpha-diversity scores by the max score for each metric.

```{r norm-alpha-scores}

sample_data(spat_size_phylo_rare_5000) <- sample_data(spat_size_phylo_rare_5000)[, c(-1)]  # [rows, cols]
# Rename sample column one at a time by name
# colnames(sample_data(ps.all))[colnames(sample_data(ps.all)) == "X.SampleID"] <- "Sample"
# Rename all columns
# colnames(sample_data(ps.all)) <- c()  # c("colName1", "colName2", ...)
# Check that renaming worked
# view(sample.data.frame(ps.all))
# Load Sample Data
df.all <- sample.data.frame(spat_size_phylo_rare_5000)  # Dataframe
dt.all <- sample.data.table(spat_size_phylo_rare_5000)
# Normalize scores from 0 to 1 for easier comparison across metrics
#   - The test will out put some statistical information about which transformations were done
dt.alphaScores.norm.all <- norm_alpha_score(alpha.base = dt.alphaScores.all,  # Unnormalized alpha scores
                                            sample.df = df.all,  # sample data frame
                                            methods = methods.alpha,  # list of alpha methods
                                            smpl.col.name = "Sample"  # Default is "Sample" but you can change it to name of sample column
                                            )
# View 
head(dt.alphaScores.norm.all)
```


# Prepare for Plotting

This creates a combined datatable with your metadata and alpha diversity scores

```{r data-table-metadata-alpha-scores}
# Create a datatable of alpha scores for melting
dt.alphaPlus.all <- dt.all[dt.alphaScores.norm.all, on = "Sample"] %>% setkeyv("Sample")
# View
head(dt.alphaPlus.all)
```

## Save Function

```{r function-save-melt-datatable}
# Melt Data Table ---------------------------------------------------------
#   Description: Melts sample data table for easy plotting
#   Input: normalized alpha scores, alpha methods, variable names
#   Output: a melted datatable
melt_to_datatable_2 <- function(datatable1, datatable2, vars, var.name, samp.name = "Sample", val.name = "Score"){
  
  comb.data <- datatable1[datatable2, on = (samp.name)] %>% setkeyv(samp.name)
  
  return( melt(
    comb.data,  # combined datatables from above
    measure.vars = vars, # Measure variables for melting. Can be missing, vector, list, or pattern-based.
    variable.name = var.name, # Name for the measured variable names column.
    value.name = val.name  # Name for the molten data values column(s).
  )
  )
}
```

## Run Function

```{r melt-alpha-data-table}
# Melt data table for easy plotting and statistical analysis
dt.alphaPlus.all.melt <- melt_to_datatable_2(datatable1 = dt.all, 
                                         datatable2 = dt.alphaScores.norm.all, 
                                         vars = methods.alpha, 
                                         var.name = "Alpha.Metric",
                                         samp.name = "Sample",
                                         val.name = "Alpha.Score")
# View
head(dt.alphaPlus.all.melt)
```


DONE! 

Now if you want to go on to plotting, here are some examples.

# Plotting

## Assign data

I like to assign temporary data variables for each plot/statistical analysis chunk, because in my experience it keeps the code nimble and flexible. 

This avoids situations where you have dozens of variables for each figure, table, etc. Instead, you can add a function at the end of the chunks to export your variables, figures, and tables with unique names, if you want.

```{r plot-setup}
# Assign a temporary data variable
data <- dt.alphaPlus.all.melt
```

### Plot

```{r plot-alphaScore-Size}
alpha_data_with_edits$Alpha.Metric <- as.factor(alpha_data_with_edits$Alpha.Metric)
alpha_data_with_edits$Alpha.Metric <- factor(alpha_data_with_edits$Alpha.Metric, levels = c("Richness", "Phylogenetic", "Simpson", "Shannon"))
# order: shannon, simpson, phylo, rich
rich <- rep("p = 0.0001", times = 128)
phy <- rep("p = 0.002", times = 128)
simp <- rep("p = 0.53", times = 128)
shan <- rep("p = 0.007", times = 128)
alpha_data_with_edits$pval <- c(shan, simp, phy, rich)
# now add means
data_for_editing <- write.csv(data, "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/alpha_data.csv")
rich_mean_large <- mean(rich_large$Alpha.Score)
rich_mean_small <- mean(rich_small$Alpha.Score)
phy_mean_large <- mean(phylo_large$Alpha.Score)
phy_mean_small <- mean(phylo_small$Alpha.Score)
simp_mean_large <- mean(simp_large$Alpha.Score)
simp_mean_small <- mean(simp_small$Alpha.Score)
shan_mean_large <- mean(shan_large$Alpha.Score)
shan_mean_small <- mean(shan_small$Alpha.Score)
# added in manually in excel
alpha_data_with_edits <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/alpha_data_with_edits.csv", header = T)
alpha_plot <- ggplot(alpha_data_with_edits, aes(x = Size, y=Alpha.Score)) +
  geom_boxplot(aes(fill = Size), outlier.shape = NA) +
  scale_fill_manual(values = c("#FF6600", "steelblue3"))+
#  ggbeeswarm::geom_quasirandom(mapping = aes(color = Spat_family), groupOnX = TRUE, alpha = 0.25, show.legend = TRUE) +  # spaces the dots out nicely
  ggbeeswarm::geom_quasirandom(groupOnX = TRUE, alpha = 0.25)+
  scale_color_manual(values = mkePalette_10)+
  facet_grid(.~Alpha.Metric) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 33, hjust = 1, vjust=1)) +
  labs(title = "Spat microbiome alpha diversity",
    y = "Normalized diversity score",
    x = "Size")+
  geom_text(label = alpha_data_with_edits$pval, x = 1.5, y = 1, size = 3, stat = "identity", check_overlap = TRUE)+
  geom_text(label = "*", x = 1, size = 7, y = alpha_data_with_edits$Mean_L, color = "black")+
  geom_text(label = "*", x = 2, size = 7, y = alpha_data_with_edits$Mean_S, color = "black")+
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major = element_line(size = 0),
        panel.grid.minor = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot", 
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))
ggsave(plot = alpha_plot, filename = "alpha_plot.jpg", path =  "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000", scale = 2, width = 85, height = 50, units = "mm")
```
Separate plot showing rarefaction depth's effect:
- summary(aov(Alpha.Score ~ rare_reads, data = shan_vals_merge)) # p = 0.354
- summary(glm(Alpha.Score ~ rare_reads, family = "quasibinomial", data = simpson_vals_merge)) # p = 0.4478
- summary(lm(Alpha.Score ~ rare_reads, data = phylo_vals_merge)) # p < 0.0001
- summary(lm(Alpha.Score ~ rare_reads, data = rich_vals_merge)) # p < 0.0001

Maybe using geom_point instead, with darkness of color indicating read depth
```{r}
shan_1 <- rep("p = 0.354", times = 128)
simp_1 <- rep("p = 0.448", times = 128)
rich_1 <- rep("p < 0.0001", times = 128)
phy_1 <- rep("p < 0.0001", times = 128)
alpha_data_with_edits$rare_pval <- c(rich_1, phy_1, simp_1, shan_1)

alpha_data_with_edits$Alpha.Metric <- factor(alpha_data_with_edits$Alpha.Metric, levels = c("Richness", "Phylogenetic", "Simpson", "Shannon"))
alpha_plot_rare <- 
  ggplot(alpha_data_with_edits, aes(x = Size, y=Alpha.Score))+
    ggbeeswarm::geom_quasirandom(groupOnX = TRUE, mapping = aes(fill = rare_reads), show.legend = TRUE, pch = 21, color = "black", size = 2.5, width = 0.5)+
    theme(legend.position = "right",
          axis.text.x = element_text(angle = 33, hjust = 1, vjust=1)) +
    labs(title = "Spat microbiome alpha diversity by sequencing depth",
      y = "Normalized diversity score",
      x = "Size", 
      fill = "Rarefied sampling depth")+
    theme(panel.background = element_rect(fill = "white"),
          panel.border = element_rect(color = "black", fill = NA),
          panel.grid.major = element_line(color = "gray90"),
          panel.grid.minor = element_line(color = "gray90"),
          axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
          axis.text.y = element_text(color = "black"),
          legend.key = element_rect(fill = NA),
          legend.spacing.y = unit(5, "mm"), # insert space between legend title and keys
          plot.title = element_text(hjust = 0.5),
          plot.subtitle = element_text(hjust = 0.5),
          plot.caption.position = "plot", 
          strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  guides(fill = guide_colourbar(ticks = FALSE))+
    geom_text(label = alpha_data_with_edits$rare_pval, x = 0.66, y = 0.04, size = 3, stat = "identity", check_overlap = TRUE)+
  scale_fill_continuous(limits = c(5000,15000), 
                        breaks = c(5000, 10000, 15000), 
                        labels = c("5000", "10000", "15000"), 
                        low = "lightgreen", high = "springgreen4")+
    facet_wrap(.~Alpha.Metric, nrow = 2)
  
 
 # geom_text(label = "*", x = 1, size = 7, y = alpha_data_with_edits$Mean_L, color = "black")+
 # geom_text(label = "*", x = 2, size = 7, y = alpha_data_with_edits$Mean_S, color = "black")+
ggsave(plot = alpha_plot_rare, filename = "alpha_plot_rare.png", path =  "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000", scale = 1.25, width = 180, height = 140, units = "mm")
```


Running some quick stats: shannon
```{r}
shan_vals <- subset(data, Alpha.Metric == "Shannon")
qqnorm(resid(lm(Alpha.Score ~ Size, data = shan_vals))) # normally distributed
t.test(Alpha.Score ~ Size, data = shan_vals)
descdist(shan_vals$Alpha.Score) # normal

shan_large <- subset(shan_vals, Size == "Large")
sd(shan_large$Alpha.Score) # 0.169
shan_small <- subset(shan_vals, Size == "Small")
sd(shan_small$Alpha.Score) # 0.243

# summary(lm(Alpha.Score ~ Size*Spat_family, data = shan_vals)) # p = 0.00774
# summary(lmer(Alpha.Score ~ Size + (1|Spat_family), data = shan_vals)) # p = 0.00774
# summary(glm(Alpha.Score ~ Size, family = "quasibinomial", data = shan_vals)) # p = 0.008


alpha_size_table_with_weight$Sample <- rownames(alpha_size_table_with_weight)
shan_vals_merge <- merge(shan_vals, alpha_size_table_with_weight, by = "Sample")
descdist(shan_vals_merge$Alpha.Score) # normal
# lm because normal distribution + continuous variables
summary(aov(Alpha.Score ~ rare_reads, data = shan_vals_merge)) # p = 0.354
summary(aov(Alpha.Score ~ Spat_family, data = shan_vals_merge)) # 0.7480
```
Running some quick stats: simpson
```{r}
simpson_vals <- subset(data, Alpha.Metric == "Simpson")

summary(lm(Alpha.Score ~ Size, data = simpson_vals)) # p = 0.529
summary(lmer(Alpha.Score ~ Size + (1|Spat_family), data = simpson_vals)) # p = 0.516
descdist(simpson_vals$Alpha.Score) # solidly beta
summary(glm(Alpha.Score ~ Size, family = "quasibinomial", data = simpson_vals)) # p = 0.53
simp_large <- subset(simpson_vals, Size == "Large")
mean(simp_large$Alpha.Score) # 0.724
sd(simp_large$Alpha.Score) # 0.201
simp_small <- subset(simpson_vals, Size == "Small")
mean(simp_small$Alpha.Score) # 0.749
sd(simp_small$Alpha.Score) # 0.237
as.numeric(alpha_rare_size_meta$Rare_reads)
# add read depth variables
alpha_size_table_with_weight$Sample <- rownames(alpha_size_table_with_weight)
simpson_vals_merge <- merge(simpson_vals, alpha_size_table_with_weight, by = "Sample")
descdist(simpson_vals_merge$Alpha.Score) # beta
summary(glm(Alpha.Score ~ rare_reads, family = "quasibinomial", data = simpson_vals_merge)) # p = 0.4478
summary(glm(Alpha.Score ~ Spat_family, family = "quasibinomial", data = simpson_vals_merge)) # none
#summary(glm(Alpha.Score ~ Spat_family, family = binomial(link = "logit"), data = simpson_vals_merge))
summary(aov(Alpha.Score ~ Spat_family, data = simpson_vals_merge))
summary(aov(Alpha.Score ~ rare_reads, data = simpson_vals_merge))
```

Phylogenetic
```{r}
phylo_vals <- subset(data, Alpha.Metric == "Phylogenetic")
descdist(phylo_vals$Alpha.Score) # normal
phylo_large <- subset(phylo_vals, Size == "Large")
t.test(Alpha.Score ~ Size, data = phylo_vals)
sd(phylo_large$Alpha.Score)
phylo_small <- subset(phylo_vals, Size == "Small")
sd(phylo_small$Alpha.Score)
phylo_vals_merge <- merge(phylo_vals, alpha_size_table_with_weight, by = "Sample")
descdist(phylo_vals_merge$Alpha.Score) # normal
summary(lm(Alpha.Score ~ rare_reads, data = phylo_vals_merge)) # p < 0.0001
summary(aov(Alpha.Score ~ rare_reads, data = phylo_vals_merge))
summary(glm(Alpha.Score ~ Spat_family, family = "quasibinomial", data = phylo_vals_merge)) # family 22 = 0.003
summary(aov(Alpha.Score ~ Spat_family, data = phylo_vals_merge))
# View as scatterplot
ggplot(data = data, aes(x = rare_reads, y = Alpha.Score))+
  geom_point(mapping = aes(color = Alpha.Metric))+
  facet_wrap(.~Alpha.Metric)+
  scale_color_viridis(discrete = TRUE, end = 0.8)
```
Richness
```{r}
rich_vals <- subset(data, Alpha.Metric == "Richness")
descdist(rich_vals$Alpha.Score) # normal
t.test(Alpha.Score ~ Size, data = rich_vals)
#summary(glm(Alpha.Score ~ Size, family = "quasibinomial", data = rich_vals)) # p = 0.000145
rich_large <- subset(rich_vals, Size == "Large")
mean(rich_large$Alpha.Score) # 0.359
sd(rich_large$Alpha.Score) # 0.16
t.test(Alpha.Score ~ Size, data = rich_vals)
rich_small <- subset(rich_vals, Size == "Small")
mean(rich_small$Alpha.Score)
sd(rich_small$Alpha.Score) # 0.209
rich_vals_merge <- merge(rich_vals, alpha_size_table_with_weight, by = "Sample")
descdist(rich_vals_merge$Alpha.Score) # normal
summary(lm(Alpha.Score ~ rare_reads, data = rich_vals_merge)) # p < 0.0001
summary(aov(Alpha.Score ~ rare_reads, data = rich_vals_merge))
# spat family
summary(lm(Alpha.Score ~ Spat_family, data = rich_vals_merge)) # Family 22 p = 0.03
summary(aov(Alpha.Score ~ Spat_family, data = rich_vals_merge))

```


Doing my own alpha diversity.
TRIMMED for 5-15,000 reads
Want to make a an alpha diversity figure with 4 metrics: 1. richness, 2. phylogenetic distance, 3. simpson, 4. shannon
```{r}
alpha_size_table_with_weight <- read.csv("~/Documents/Oyster_Project/size_microbiome/alpha_size_table_with_weight.csv", sep = ",", header = T, row.names = 1)
samp_dat <- matrix(sample_data(spat_size_phylo_rare_5000))
tree <- phy_tree(spat_5000_tree)
# phylogenetic distance
phy_dist <- picante::pd(samp = matrix(t(otu_table(spat_size_phylo_rare_5000))), tree = phyloseq::phy_tree(spat_size_phylo_rare_5000), include.root = TRUE)# %>%
      as.data.table(keep.rownames = smpl.col.name) %>% setkeyv(smpl.col.name)
```

Need to look at weight data.
```{r}
weight_data <- read.csv("~/Documents/Oyster_Project/size_microbiome/weight_meta/Spat_weights.csv", sep = ",", header = T)
weight_data$Family <- as.factor(weight_data$Family)
weight_large <- subset(weight_data, Size == "Large")
mean(weight_large$Avg_dry_weight) # 0.356g
sd(weight_large$Avg_dry_weight) # 0.137
weight_small <- subset(weight_data, Size == "Small")
mean(weight_small$Avg_dry_weight) # 0.029g
sd(weight_small$Avg_dry_weight) # 0.018
spat_weight_plot <- ggplot(weight_data, aes(x = Family, y = Dry_weight, group = Size))+
  geom_bar(stat = "summary", position = position_dodge(width = 0.9), mapping = aes(fill = Size), color = "black", size = 0.25)+
    geom_errorbar(aes(ymin = Avg_dry_weight-Sterr_drywt, ymax = Avg_dry_weight+Sterr_drywt), 
                width = 0.25, position = position_dodge(width = 0.9))+
  scale_color_manual(values = c("#FF6600", "steelblue3"))+
  scale_fill_manual(values = c("#FF6600", "steelblue3"))+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major.y = element_line(color = "gray90"),
        panel.grid.minor.y = element_line(color = "gray90"),
        panel.grid.major.x = element_line(size = 0),
        panel.grid.minor.x = element_line(size = 0),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot",
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  scale_y_continuous(limits = c(0,0.7), expand = c(0,0))+
 #   geom_label(x=0.28, y=-0.34, label = "p(dispersion ~ Size (Spat family = random)) = 0.03", size = 3, color = "black", fill = "white")+
  labs(title = "Dry weight by spat family and size",
   #    caption = "Family 94 not included",
       x = "Spat family",
       y = "Spat dry weight (g)")
ggsave(plot = spat_weight_plot, filename = "spat_weight_plot.png", path = "~/Documents/Oyster_Project/size_microbiome/weight_meta/", scale = 2, width = 100, height = 75, units = "mm")
# boxplot of same thing
spat_weight_boxplot <- 
  ggplot(weight_data, aes(y = Dry_weight, x = Size))+
  geom_boxplot(mapping = aes(fill = Size), show.legend = FALSE, outlier.shape = NA)+
  scale_color_manual(values = c("#FF6600", "steelblue3"))+
  scale_fill_manual(values = c("#FF6600", "steelblue3"))+
  ggbeeswarm::geom_quasirandom(groupOnX = TRUE, alpha = 0.25)+
 # geom_segment(aes(x = 0.9, xend = 1.1, y = 0.55, yend = 0.55),  linetype = "dashed")+
    geom_text(label = "*", x = 1, y = 0.356, size = 7)+
    geom_text(label = "*", x = 2, y = 0.029, size = 7)+
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major.y = element_line(color = "gray90"),
        panel.grid.minor.y = element_line(color = "gray90"),
        panel.grid.major.x = element_line(size = 0),
        panel.grid.minor.x = element_line(size = 0),
        axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        plot.caption.position = "plot",
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
  geom_text(label = "p < 0.0001", x = 1.5, y = 0.55, size = 3, stat = "identity", check_overlap = TRUE)+
  scale_y_continuous(limits = c(0,0.7), expand = c(0,0))+
  labs(title = "Spat dry weight",
       x = "Size",
       y = "Dry weight (g)")
ggsave(plot = spat_weight_boxplot, filename = "spat_weight_boxplot.jpg", path = "~/Documents/Oyster_Project/size_microbiome/weight_meta/", scale = 1.75, width = 50, height = 75, units = "mm")
```
Stats proving size differences
```{r}
descdist(weight_data$Dry_weight) # to decide which model; if beta, use glm (quasi).
summary(glm(Dry_weight ~ Size, family = "quasibinomial", data =  weight_data)) # p < 0.0001
summary(glm(Dry_weight ~ Family*Size, family = "quasibinomial", data =  weight_data))
large_sub <- subset(weight_data, Size == "Large")
shapiro.test(large_sub$Dry_weight) # p = 0.129
shapiro.test(log(large_sub$Dry_weight)) # p = 0.22
small_sub <- subset(weight_data, Size == "Small")
shapiro.test(small_sub$Dry_weight) # p = 0.0067
shapiro.test(log(small_sub$Dry_weight)) # p = 0.1475
shapiro.test(weight_data$Dry_weight) # p < 0.0001
shapiro.test(log(weight_data$Dry_weight)) # p = 0.002
# Looks like I might have to log-transform the weight
t.test(Dry_weight ~ Size, data = weight_data) # p < 0.0001
lm <- summary(lm(Dry_weight ~ Size * Family, data = weight_data))
qqnorm(resid(lmer(Dry_weight ~ Size + (1|Family), weight_data)))
class(weight_data$Family)
summary(aov(Dry_weight ~ Family, data = weight_data))
weight_data$Size <- as.factor(weight_data$Size)
lm$coefficients[,4]
```
T tests for each family (for figure S2)
```{r}
weight_data <- read.csv("~/Documents/Oyster_Project/size_microbiome/weight_meta/Spat_weights.csv", sep = ",", header = T)
weight_data$Family <- as.factor(weight_data$Family)
summary(lm(weight_data$Dry_weight ~ weight_data$Family+weight_data$Size))
```


GLM looking at core microbiome taxa percentage as a predictor of dry weight
```{r}
write.csv(t(otu_table(phylo_combo)), file = "~/Documents/Oyster_Project/size_microbiome/for_diversity/asv_core.csv")
# Added in metadata
asv_core_with_meta <- read.csv("~/Documents/Oyster_Project/size_microbiome/for_diversity/asv_core_with_meta.csv", sep = ",", header = T)
lm_p_val_asv0013 <- summary(lm(asv0013~Avg_dry_weight, data = asv_core_with_meta))$coefficients[8]
summary(lm(asv0002~Avg_dry_weight, data = asv_core_with_meta))
ttest_p_asv0013 <- t.test(asv0013 ~ Size, data = asv_core_with_meta)
```

Looking into Mollicutes, since it's important in other studies and shows up as higher in large.
```{r}
otu_data <- otu_table(spat_size_phylo_rare_5000_trans)
meta <- data.frame(sample_data(spat_size_phylo_rare_5000_trans))
rownames(otu_data)
asv01 <- subset(otu_data, rownames(otu_data) %in% "asv01")
asv01 <- data.frame(t(asv01))
asv01 <- asv01 %>% add_column(.before = "asv01", ID = NA)
asv01$ID <- rownames(asv01)
colnames(asv01) <- c("identifier", "asv01")
meta01 <- merge(asv01, meta, by = "identifier")
# stats test for relative abundance by size
t.test(log(meta01$asv01) ~ meta01$Size) # p = 0.5194; with log it's .08
ggplot(data = meta01)+
  geom_point(aes(x = Size, y = asv01*100, color = Size))
descdist(meta01$asv01) # beta -- use glm
qqnorm(resid(lm((log(asv01)) ~ Size, data = meta01))) # log-transformed looks more normal
summary(glm(asv01 ~ Size, data = meta01, family = "quasibinomial"))
```
Looking at asv10
```{r}
rhodo_tree <- ape::read.nexus("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/rhodo_tree.nexus")
plot(rhodo_tree)

```
From Hannah:
```{bash}
# convert fasta to qza 
qiime tools import --input-path /home/micro/englimar/mk/spatmicro/seq.fa --output-path /home/micro/englimar/mk/spatmicro/seqs.qza --type 'FeatureData[Sequence]'

# convert newick to qza
qiime tools import   --input-path /home/micro/englimar/mk/spatmicro/rhodo_tree.newick --output-path /home/micro/englimar/mk/spatmicro/rhodo_tree.qza --type 'Phylogeny[Rooted]'
```

Mycoplasma
```{r}
myco_tree <- read.tree(file = "/Users/marykenglish/Desktop/myco.ntree")
```
Table exploring dominant taxa 1) overall, 2) in large and 3) in small
```{r}
counts <- data.frame(otu_table(spat_size_phylo_rare_5000), check.names = FALSE)
all_sum_pct <- data.frame((rowSums(counts)/(sum(rowSums(counts)))*100)) # total percentage of reads dedicated to given ASV in both size classes
colSums(all_pct) # verify that it adds to 100
counts <- t(counts)
all_avg_pct <- (data.frame(colMeans(counts / rowSums(counts)) * 100)) # average percentage of reads dedicated to given ASV in both size classes
counts <- t(counts)
# all
# large only
subset_large <- subset_samples(spat_size_phylo_rare_5000, Size == "Large")
large <- data.frame(otu_table(subset_large), check.names = FALSE)
large_sum_pct <- data.frame((rowSums(large)/sum(rowSums(large)))*100) # total percentage of reads dedicated to given ASV in large
large <- t(large) # have to transpose to get it to work
max(large / rowSums(large) * 100)
large_avg_pct <- (data.frame(colMeans((large / rowSums(large)) * 100))) # average percentage of reads dedicated to given ASV in large
large <- t(large)
rownames(large) == rownames(large_sum_pct)
# small only
subset_small <- subset_samples(spat_size_phylo_rare_5000, Size == "Small")
small <- data.frame(otu_table(subset_small), check.names = FALSE)
small_sum_pct <- data.frame((rowSums(small)/sum(rowSums(small)))*100) # total percentage of reads dedicated to given ASV in small
small <- t(small) # have to transpose to get it to work
max(small / rowSums(small) * 100)
small_avg_pct <- (data.frame(colMeans((small / rowSums(small)) * 100))) # average percentage of reads dedicated to given ASV in small
small <- t(small)
rownames(small) == rownames(small_sum_pct)
# verified that row names are same among all 3 data frames
asv_meta <- cbind(rownames(counts), 
                  round(all_sum_pct, digits = 4), 
                  round(all_avg_pct, digits = 4),
                  round(large_sum_pct, digits = 4), 
                  round(large_avg_pct, digits = 4),
                  round(small_sum_pct, digits = 4),
                  round(small_avg_pct, digits = 4))
colnames(asv_meta) <- c("ASV", 
                        "Total percent of reads in all", 
                        "Average percent of reads in all",
                        "Total percent of reads in large", 
                        "Avg. percent of reads in large",
                        "Total percent of reads in small",
                        "Avg. percent of reads in small")
rownames(asv_meta) <- NULL
sum(asv_meta[,4])
colSums(asv_meta[,2:4])
write.csv(asv_meta, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/asv_meta.csv")
```
Pulling out the ASVs that are DA in large, but < 1% rel abund
```{r}
large_lowabund <- subset(fdr_ancom_nocore_taxa_edit_with_pct, beta < 0 & large_pct < 1)
write.csv(large_lowabund, file = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/large_lowabund.csv")
print(large_lowabund)
```


Pedigree with spat cohort # 29
```{r}
pedigree <- read.csv("~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/pedigree.csv")
matrix_full <- Amatrix(pedigree, ploidy = 2)
pedigree_29 <- pedigree[grep("^29.*", pedigree$Family),]
matrix_sub <- matrix_full[grep("^29.*", matrix_full), ]
individuals <- c(29.003,
29.022,
29.024,
29.025,
29.029,
29.04,
29.069,
29.074,
29.094,
29.095)
male1 <- c(27.011, 28.076, 27.016, 28.029, 28.037, 28.053, 27.046, 26.014, 27.017, 28.016)
female1 <- c(27.028, 27.071, 27.046, 27.012, 27.012, 27.016, 24.019, 28.007, 27.028, 27.009)
male2 <- c(25.048,
27.028,
23.001,
27.041,
27.011,
27.044,
26.025,
23.01,
26.029,
27.052)
female2 <- c(25.015,
26.024,
26.021,
26.014,
25.014,
22.039,
25.023,
24.072,
26.008,
22.02)
male3 <- c(26.023,
24.019,
26.025,
26.021,
26.021,
23.001,
"P4.003",
27.053,
26.023,
26.029)
female3 <- c(26.014,
22.002,
25.023,
26.024,
26.024,
26.021,
"P4.014",
25.015,
26.014,
24.062)
combined <- c(individuals, male1, female1, male2, female2, male3, female3)
matrix <- Amatrix(pedigree, ploidy = 2)
#pedigree_29 <- pedigree[grep(combined, pedigree$Family),]
#matrix_sub <- matrix_full[grep("^29.*", matrix_full), ]
matrix_sub <- subset(matrix, rownames(matrix) %in% combined)
matrix_sub <- t(matrix_sub)
matrix_sub <- subset(matrix_sub, rownames(matrix_sub) %in% individuals)
matrix_sub <- t(matrix_sub)
matrix_sub <- subset(matrix_sub, rownames(matrix_sub) %in% individuals)
class(matrix_sub)
colnames(matrix_sub) <- c(3, 22, 24, 25, 29, 40, 69, 74, 94, 95)
matrix_sub <- t(matrix_sub)
my_colors <- colorRampPalette(c("yellow", "blue4")) 
heatmap(matrix_sub, col = my_colors(50))
ggplot(matrix_sub)+
  geom_tile()
temp_matrix_cor <- cor(temp_matrix_num)
matrix_sub_round <- round(matrix_sub, 3)
  get_upper_tri <- function(matrix_sub_round){
    matrix_sub_round[lower.tri(matrix_sub_round)]<- NA
    return(matrix_sub_round)
  }
upper_matrix <- get_upper_tri(matrix_sub_round)
melt_upper_matrix <- melt(upper_matrix, na.rm = TRUE)
get_lower_tri<-function(matrix_sub_round){
    matrix_sub_round[upper.tri(matrix_sub_round)] <- NA
    return(matrix_sub_round)
  }
upper_matrix <- get_upper_tri(matrix_sub_round)
melt_upper_matrix <- melt(upper_matrix, na.rm = TRUE)
lower_matrix <- get_lower_tri(matrix_sub_round)
melt_lower_matrix <- reshape2::melt(lower_matrix, na.rm = TRUE)

ggplot(data = melt_lower_matrix, aes(x=as.character(Var1), y=as.character(Var2), fill = value))+
 geom_tile(color = "white")+
 # geom_text(aes(Var1, Var2, label = sprintf(fmt = '%.03f', value)), color = "white", size = 3)+
 scale_fill_gradient2(low = "skyblue1", high = "navyblue", mid = "royalblue1", 
  midpoint = 0.534912, limit = c(0,1.07), space = "Lab")+#, 
#  name="Pearson\nCorrelation\nCoefficient")#+
#  theme(panel.background = element_rect(fill = "white"), panel.grid.major = element_line(color = "gray90"), axis.ticks.x = element_line(color = "gray90"),
#        axis.ticks.y = element_line(color = "gray90"), axis.line.x = element_line(color = "gray90"), axis.line.y = element_line(color = "gray90"))+
   labs(x = NULL, y = NULL) 
 #      title = "Correlation matrix: antibiotic experiment",
 #      subtitle = "Light intensity data by tank",
  #     caption = "Summer 2021")#+ coord_fixed()
```
Looking for the sample with the most Mycoplasma.
```{r}
mycos <- data.frame(otu_table(spat_size_phylo_rare_5000_trans)[c("asv01","asv02")], check.names = FALSE)
mycos <- t(mycos)
```
Sample 94L7 has the highest percentage of ASV01, at 79%. But how much asv02 does it have
```{r}
mycos2 <- data.frame(otu_table(spat_size_phylo_rare_5000_trans)["asv02",])
mycos2 <- t(mycos2)
mycos2["X94L7",]
min((data.frame(otu_table(spat_size_phylo_rare_5000_trans)))["asv01",]) # 0.00103
min((data.frame(otu_table(spat_size_phylo_rare_5000)))["asv01",]) # 12
grep("0", (data.frame(otu_table(spat_size_phylo_rare_5000)["asv02",])), invert = TRUE)
```
0.28%. Way less, sounds good to me.
How many samples have ASV02 in them?
```{r}
asv_table <- data.frame(otu_table(spat_size_phylo_rare_5000), check.names = FALSE)
asv_table <- t(asv_table)
asv02_sub <- data.frame(asv_table[,"asv02"])
colnames(asv02_sub) <- "asv02count"
asv02_no0 <- subset(asv02_sub, asv02count > 0)
dim(asv02_no0)
nrow(asv02_no0)/nrow(asv02_sub) # 85.9%

```

Plot of ASV01 abundance with bodyweight
```{r}
spat_size_phylo_rare_5000_trans
weight_data
```
are the differentially abundant taxa part of the core microbiomes?
```{r}
lg <- subset(fdr_ancom_nocore_taxa_edit_with_pct, W < 0)
pl_sort <- rownames(otu_table(phylo_large)) %>% sort()
lg_sort <- lg$Species %>% sort()
sum(lg$avg_pct) # 55%
sum(sums2[c("asv01", "asv06", "asv09", "asv10", "asv11", "asv13", "asv18", "asv20", "asv22", "asv29", "asv41", "asv44", "asv48", "asv61", "asv62", "asv67", "asv72", "asv81", "asv82", "asv86", "asv89"),]) # 56%
mean(sums2[c("asv01", "asv06", "asv09", "asv10", "asv11", "asv13", "asv18", "asv20", "asv22", "asv29", "asv41", "asv44", "asv48", "asv61", "asv62", "asv67", "asv72", "asv81", "asv82", "asv86", "asv89"),]) # 2.7%
# Present in core but not diff abundant: asv72, asv44, asv48, 61, 81, 86 = 6/21
# differentially abundant and in core: 15/21

sm <- subset(fdr_ancom_nocore_taxa_edit_with_pct, W > 0)
ps_sort <- rownames(otu_table(phylo_small)) %>% sort()
sm_sort <- sm$Species %>% sort()
sum(sm$avg_pct) # 7.9%
# differentially abundant and in core: 10/12 # asv45, 77.

sum(sums2[c("asv19", "asv30", "asv35", "asv45", "asv50", "asv53", "asv54", "asv57", "asv60", "asv76", "asv77", "asv93"),]) # 6.5
```

EXPLORING MYCOPLASMA MAG IN SAMPLE 94L7
```{bash}
gunzip *
mv lane1-s276-index--CGAGGCTG-AGAGTAGA-94L7_S276_R1_001.fastq meta_94L7_R1.fastq
mv lane1-s276-index--CGAGGCTG-AGAGTAGA-94L7_S276_R2_001.fastq meta_94L7_R2.fastq
/local/cluster/bin/trim_galore --paired /home/micro/englimar/mk/rol_abx/meta_94L7_R1.fastq /home/micro/englimar/mk/rol_abx/meta_94L7_R2.fastq
```
Using Nextera adapter (CTGTCTCTTATA)

RUN STATISTICS FOR INPUT FILE: /home/micro/englimar/mk/rol_abx/meta_94L7_R1.fastq
=============================================
5406752 sequences processed in total

RUN STATISTICS FOR INPUT FILE: /home/micro/englimar/mk/rol_abx/meta_94L7_R2.fastq
=============================================
5406752 sequences processed in total

Assemble reads.
```{bash}
/local/cluster/bin/megahit -1 meta_94L7_R1_val_1.fq -2 meta_94L7_R2_val_2.fq -o megahit_assembly --min-count 2 --k-list 21,41,61,81 --merge-level 20,0.95 --prune-depth 2 --min-contig-len 500
get_assembly_stats.py final.contigs.fa
```
210869 contigs, 
total 183072191 bp, 
min 500 bp, 
max 16914 bp, 
avg 868 bp, 
N50 868 bp

Make a database of bacterial, archaeal, and viral genomes:
```{bash}
/local/cluster/bin/kaiju-makedb -s refseq
```
Make a database of the C. gigas genome:
```{bash}
bin/kaiju-mkbwt -a DNA -o kaiju_oyster c_gigas_genome.fasta
```

REVIEWER COMMENTS
Add rarefaction curve.
```{r}
all_seqs <- read.csv("~/Documents/Oyster_Project/size_microbiome/non_reduced/seqtab_with_filter.csv")
```
Adding centroids

```{r}
df <- plot_ordination(spat_size_phylo_rare_5000, ordinate(spat_size_phylo_rare_5000, method="PCoA", 
                                distance="unifrac", 
                                weighted=TRUE), type="samples", 
                  color="Size", justDF=TRUE)
a <- mean(df$Axis.1[df$Size=="Large"]) 
b <- mean(df$Axis.2[df$Size=="Large"])
c <- mean(df$Axis.1[df$Size=="Small"])
d <- mean(df$Axis.2[df$Size=="Small"])
spat_size_phylo_rare_5000_wuni_cent <-
plot_ordination(spat_size_phylo_rare_5000, spat_size_phylo_rare_5000_wuni, color = "Size")+
geom_point(aes(x=a,y=b), fill = "#FF6600", color = "black", pch = 22, size = 5)+
geom_point(aes(x=c,y=d), fill = "steelblue3", color = "black", pch = 22, size = 5)+
geom_point(mapping = aes(fill = Size), size = 4, pch = 21, color = "black", show.legend = TRUE)+
scale_color_manual(values = c("steelblue3","#FF6600"))+
scale_fill_manual(values = c("steelblue3","#FF6600"))+
theme(panel.background = element_rect(fill = "white"),
panel.border = element_rect(color = "black", fill = NA),
panel.grid.major = element_line(color = "gray90", linewidth = 0.25),
panel.grid.minor = element_line(size = 0),
axis.text.x = element_text(color = "black", angle = 0, hjust = 0.5),
axis.text.y = element_text(color = "black"),
legend.key = element_rect(fill = NA),
plot.title = element_text(hjust = 0.5),
plot.subtitle = element_text(hjust = 0.5),
plot.caption.position = "plot",
legend.text = element_text(size = 10),
legend.box.spacing = unit(1, "mm"),
strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
stat_ellipse(type = "norm", linetype = 2, show.legend = FALSE)+
labs(title = "Spat microbiome by size (Weighted UniFrac)",
x = "Axis 1 (38.4%)",
y = "Axis 2 (18.4%)")+
guides(color = guide_legend(override.aes = list(size = 5)))+
geom_text(label = "p < 0.05", x = -0.47, y = -0.34, stat = "identity", check_overlap = TRUE, size = 3, show.legend = FALSE, color = "black")
ggsave(plot = spat_size_phylo_rare_5000_wuni_cent, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/spat_size_phylo_rare_5000_wuni_cent.jpg", scale = 1.25, width = 200, height = 100, units = "mm")

```
Ancombc2
```{r}
# Size: group, with reference = Large
# Spat_family: fixed effect
# The following is a simple model, the one we want to use in the manuscript.
ancom_out_nocore_size <- ancombc2(spat_size_phylo_rare_5000, group = "Size",
fix_formula = "Size+Spat_family",
p_adj_method = "BH", neg_lb = TRUE # for sample sizes over 30 per group
)
library(ANCOMBC)
# Size: group, with reference = Large
# Spat_family: fixed effect
# The following is a simple model, the one we want to use in the manuscript.
ancom_out_nocore_size <- ancombc2(spat_size_phylo_rare_5000, group = "Size",
fix_formula = "Size+Spat_family",
p_adj_method = "BH", neg_lb = TRUE # for sample sizes over 30 per group
)
```
This got erased, but the "new" effect size plot uisng ANCOMBC2 can be made by tweaking the following code.
```{r}
effectsizeplot_nocore <-
  ggplot(fdr_ancom_nocore_taxa_edit_with_pct, aes(x = reorder(ID, -W), y = -W))+
  geom_rect(xmin = 0, xmax = Inf, ymin = -Inf, ymax = 0, fill = "steelblue3", alpha = 0.02)+ # blue box
  geom_rect(xmin = 0, xmax = Inf, ymin = 0, ymax = Inf, fill = "#FF6600", alpha = 0.02)+ # orange box
  geom_point(aes(fill = Class, size = avg_pct), pch = 21, color = "black")+ # dots
  geom_hline(yintercept = 0)+ #dividing line
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(color = "black", fill = NA),
        panel.grid.major.y = element_line(color = "gray90"),
        panel.grid.minor.y = element_line(color = "gray90"),
        panel.grid.major.x = element_line(color = "gray90"),
    #    panel.grid.minor.x = element_line(color = "gray90"),
        axis.text.x = element_text(color = "black", angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(color = "black"),
        legend.key = element_rect(fill = NA),
        plot.title = element_text(hjust = 0.5),
        plot.caption.position = "plot", 
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 12),
        legend.box.spacing = unit(1, "mm"),
        strip.background = element_rect(fill = "white", color = "black", linetype = "solid"))+
 # scale_x_continuous(breaks = seq(1,32,2))+
  labs(x = "ASV", y = "-W (test statistic) ", title = "Differentially abundant ASVs by spat size", size = "Avg. pct. of reads in size", fill = "Class")+
  guides(fill = guide_legend(override.aes = list(size = 5), order = 2), size = guide_legend(order = 1))+
  scale_color_manual(values = rainbow, na.value = "gray45")+
  scale_fill_manual(values = rainbow, na.value = "gray45")+
  scale_size_continuous(limits = c(0,26), breaks = c(0.1,1, 5,10,25), labels = c("0.1","1","5","10","25"), range = c(2,10))

ggsave(plot = effectsizeplot_nocore, filename = "~/Documents/Oyster_Project/size_microbiome/rarefied/rare5000/effectsizeplot_nocore.jpg", scale = 1.5, width = 180, height = 100, units = "mm")
```

Beta dispersion
```{r}
permutest(betadisper(spat_size_phylo_rare_5000_wunidist, group = spat_size_meta_reduced_5000$Size))
```
Weight stats
```{r}
ggplot(weight_data, aes(x = Size, y = Avg_dry_weight))+
  geom_point(aes(color = Size), position = "jitter")
large_weight <- subset(weight_data, Size == "Large")
small_weight <- subset(weight_data, Size == "Small")
sd(small_weight$Avg_dry_weight) # 0.01778
mean(small_weight$Avg_dry_weight) # 0.029
min(large_weight$Avg_dry_weight) # 0.163
max(small_weight$Avg_dry_weight) # 0.063
sum(max(small_weight$Avg_dry_weight), (3*sd(small_weight$Avg_dry_weight))) # 0.116
mean(small_weight$Avg_dry_weight)+(3*sd(small_weight$Avg_dry_weight)) # 0.083
mean(large_weight$Avg_dry_weight)-(3*sd(large_weight$Avg_dry_weight)) # -0.05
min(large_weight$Avg_dry_weight)/max(small_weight$Avg_dry_weight)
```
Reviewer is wondering if we rarefy all reads to 5000, if there will be no difference in richness between small and large.
```{r}
library(microbiome)
phylo_rarefied <- rarefy_even_depth(spat_size_phylo_rare_5000, sample.size = 4999)
alpha_check <- microbiome::alpha(phylo_rarefied, index = c("diversity_shannon", "observed"))
alpha_check_rich <- data.frame(alpha_check[,"observed"])
rownames(alpha_check_rich) <- rownames(alpha_check)
alpha_check_rich$identifier <- rownames(alpha_check_rich)
colnames(alpha_check_rich) <- c("observed", "identifier")
alpha_rich <- dplyr::left_join(x = alpha_check_rich, y = spat_size_meta_reduced_5000 %>% dplyr::select("identifier", "Spat_family", "Size"), by = "identifier")
library(fitdistrplus)
descdist(alpha_rich$observed) # normal, can use t.test
t.test(observed ~ Size, data = alpha_rich)
ggplot(data = alpha_rich, aes(x = Size, y = observed))+
  geom_point(aes(color = Size))
```
Yes, the difference in richness with size are still significant even when all samples are rarefied to 5000 reads.
